<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2022-07-01" />

<title>CellDecon_WEHI</title>

<script src="site_libs/header-attrs-2.23/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.28/datatables.js"></script>
<link href="site_libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script>
<link href="site_libs/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet" />
<script src="site_libs/nouislider-7.0.10/jquery.nouislider.min.js"></script>
<link href="site_libs/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet" />
<script src="site_libs/selectize-0.12.0/selectize.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">20131906_HickeyT_JC_NormalBreast</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="description.html">Description</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    QC
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="qc_raw.html">Raw Data</a>
    </li>
    <li>
      <a href="qc_trimmed.html">Trimmed Data</a>
    </li>
    <li>
      <a href="qc_aligned.html">Aligned Data</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="dge_analysis.html">Differential Gene Expression</a>
    </li>
    <li>
      <a href="IDC_ILC.html">IDC/ILC classification</a>
    </li>
    <li>
      <a href="CellDecon_WEHI.html">Cell Type Deconvolution</a>
    </li>
    <li>
      <a href="sSNAPPY.html">sSNAPPY</a>
    </li>
    <li>
      <a href="E2_signal.html">E2 response</a>
    </li>
    <li>
      <a href="DHT_signal.html">DHT response</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">CellDecon_WEHI</h1>
<h4 class="date">2022-07-01</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-08-01
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>WenjunLiu_Thesis_Chapter4/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity itâ€™s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200930code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20200930)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200930code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200930)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongnoversioncontrol">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>Repository version:</strong> no
version control </a>
</p>
</div>
<div id="strongRepositoryversionstrongnoversioncontrol"
class="panel-collapse collapse">
<div class="panel-body">
<p>Tracking code development and connecting the code version to the
results is critical for reproducibility. To start using Git, open the
Terminal and type <code>git init</code> in your project directory.</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
This project is not being versioned with Git. To obtain the full
reproducibility benefits of using workflowr, please see
<code>?wflow_start</code>.
</p>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tidyverse)
library(yaml)
library(scales)
library(pander)
library(glue)
library(edgeR)
library(AnnotationHub)
library(ensembldb)
library(cowplot)
library(ggfortify)
library(magrittr)
library(cqn)
library(ggrepel)
library(DT)
library(celldex)
library(dtangle)
# library(loomR)
library(Seurat)
library(Matrix)
library(corrplot)
# library(MuSiC)
library(plotly)
library(colorspace)
library(ggpubr)
library(randomcoloR)
library(ggforce)
library(pheatmap)
source(here::here(&quot;analysis/smallFunctions/DWLS_funs.R&quot;))</code></pre>
<pre class="r"><code>panderOptions(&quot;table.split.table&quot;, Inf)
panderOptions(&quot;big.mark&quot;, &quot;,&quot;)
theme_set(theme_bw())</code></pre>
<pre class="r"><code>config &lt;- here::here(&quot;config/config.yml&quot;) %&gt;%
  read_yaml()
suffix &lt;- paste0(config$tag)
sp &lt;- config$ref$species %&gt;%
  str_replace(&quot;(^[a-z])[a-z]*_([a-z]+)&quot;, &quot;\\1\\2&quot;) %&gt;%
  str_to_title()</code></pre>
<div id="setup" class="section level1">
<h1>Setup</h1>
<div id="gene-annotations" class="section level2">
<h2>Gene Annotations</h2>
<pre class="r"><code>ah &lt;- AnnotationHub() %&gt;%
  subset(rdataclass == &quot;EnsDb&quot;) %&gt;%
  subset(str_detect(description, as.character(config$ref$release))) %&gt;%
  subset(genome == config$ref$build)
stopifnot(length(ah) == 1)</code></pre>
<pre class="r"><code>ensDb &lt;- ah[[1]]
genesGR &lt;- read_rds(here::here(&quot;output/genesGR.rds&quot;))</code></pre>
</div>
<div id="sample-metadata-and-merged-counts" class="section level2">
<h2>Sample metadata and merged counts</h2>
<p>Sample metadata and merged counts were read in and filtered as in the
dge_analysis.Rmd. DGElists were formed for normal and tumor samples and
<code>cqn</code> was applied for biases introduced by systematic
artefacts.</p>
<pre class="r"><code>init_cellType &lt;- read_delim(here::here(&quot;config/sample_meta.txt&quot;), delim = &quot;\t&quot;)
init_cellType &lt;- init_cellType %&gt;%
    mutate(Stroma = ifelse(str_detect(Dominant_cell_type, regex(&quot;Stroma&quot;, ignore_case = T)), TRUE, FALSE), 
           Epithelial = ifelse(str_detect(Dominant_cell_type, regex(&quot;Epithelial&quot;, ignore_case = T)), TRUE, FALSE), 
           Ducts = ifelse(str_detect(Dominant_cell_type, regex(&quot;ducts&quot;, ignore_case = T)), TRUE, FALSE), 
           Fat = ifelse(str_detect(Dominant_cell_type, regex(&quot;fat&quot;, ignore_case = T)), TRUE, FALSE), 
           patient = str_replace(patient, &quot;TH&quot;, &quot;TH-&quot;)) %&gt;%
    pivot_longer(c(&quot;Stroma&quot;, &quot;Epithelial&quot;, &quot;Ducts&quot;, &quot;Fat&quot;),
                 names_to = &quot;cell_type&quot;, 
                 values_to = &quot;TF&quot;) %&gt;%
    mutate(cell_type = ifelse(TF, cell_type, NA)) %&gt;%
    dplyr::select(-c(&quot;TF&quot;, &quot;Dominant_cell_type&quot;)) %&gt;%
    .[!is.na(.$cell_type),] %&gt;%
    chop(&quot;cell_type&quot;) %&gt;%
    mutate(cell_type = vapply(.$cell_type, function(x){
        paste(x,collapse = &quot;;&quot;)
    }, character(1)))</code></pre>
<pre class="r"><code>samples &lt;- config$samples %&gt;%
  here::here() %&gt;%
  read_tsv() %&gt;%
    left_join(init_cellType) %&gt;%
  mutate(
    Filename = paste0(sample, &quot;.r_1&quot;), 
    condition = ifelse(Tumor, 
                       paste(&quot;Tumor&quot;, treat, sep = &quot;_&quot;), 
                       paste(&quot;Normal&quot;, treat, sep = &quot;_&quot;)), 
    patient = vapply(.$patient, function(x){str_split(x, &quot;-&quot;)[[1]][2]}, character(1)), 
    patient = ifelse(Tumor, 
                       paste(&quot;Tumor&quot;, patient, sep = &quot;-&quot;), 
                       paste(&quot;Normal&quot;,patient, sep = &quot;-&quot;)), 
    desc = paste(patient, treat, sep = &quot; &quot;)
  ) %&gt;%
  dplyr::select(-c(&quot;name&quot;, &quot;sample&quot;)) %&gt;%
  dplyr::rename(name = desc) %&gt;%
  mutate_if(
    function(x){length(unique(x)) &lt; length(x)},
    as.factor
  ) %&gt;%
  mutate(
    treat = relevel(treat, ref = &quot;Veh&quot;)
  )</code></pre>
<pre class="r"><code>mergedSamples &lt;- samples %&gt;%
  group_by(name, patient, treat, Tumor, cell_type, Tissue_type, Age, Diagnosis) %&gt;%
  tally()
tumor_sample &lt;- mergedSamples %&gt;%
  dplyr::filter(Tumor == TRUE) %&gt;%
    droplevels()</code></pre>
<p>Tumour samples were classified into a IDC-like and a ILC-like group
bassing on the analysis in <a href="IDC_ILC.html">ILC/IDC
classification</a>:</p>
<pre class="r"><code>tumor_sample &lt;- tumor_sample %&gt;%
    dplyr::mutate(
        Diagnosis = ifelse(
            str_detect(Diagnosis, &quot;invasive&quot;), 
            &quot;ILC-like&quot;, &quot;IDC-like&quot;
        ),
        Diagnosis = ifelse(
            patient == &quot;Tumor-8&quot;, 
            &quot;IDC-like&quot;, 
            Diagnosis
        ), 
        Diagnosis = as.factor(Diagnosis)
    )</code></pre>
<pre class="r"><code>tumor_sample %&gt;%
    ungroup() %&gt;%
    dplyr::select(patient, Diagnosis) %&gt;%
    unique() %&gt;%
    pander()</code></pre>
<table style="width:31%;">
<colgroup>
<col width="13%" />
<col width="16%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">patient</th>
<th align="center">Diagnosis</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">Tumor-1</td>
<td align="center">ILC-like</td>
</tr>
<tr class="even">
<td align="center">Tumor-2</td>
<td align="center">IDC-like</td>
</tr>
<tr class="odd">
<td align="center">Tumor-3</td>
<td align="center">IDC-like</td>
</tr>
<tr class="even">
<td align="center">Tumor-4</td>
<td align="center">ILC-like</td>
</tr>
<tr class="odd">
<td align="center">Tumor-5</td>
<td align="center">IDC-like</td>
</tr>
<tr class="even">
<td align="center">Tumor-6</td>
<td align="center">ILC-like</td>
</tr>
<tr class="odd">
<td align="center">Tumor-7</td>
<td align="center">ILC-like</td>
</tr>
<tr class="even">
<td align="center">Tumor-8</td>
<td align="center">IDC-like</td>
</tr>
</tbody>
</table>
<pre class="r"><code>treat_cols &lt;- c(
  Veh = rgb(0.7, 0.7, 0.7),
  DHT = rgb(0.8, 0.2, 0.2),
  E2 = rgb(0.2, 0.2, 0.8),
  `E2+DHT` = rgb(1, 0.4, 1)
)
tumor_cols &lt;- hcl.colors(
  n = length(unique(samples$Tumor)), 
  palette = &quot;Zissou 1&quot;
  ) %&gt;%
  setNames(unique(samples$Tumor))
patient_cols &lt;- hcl.colors(
  n = length(levels(mergedSamples$patient)), 
  palette = &quot;Spectral&quot;
  ) %&gt;%
  setNames(levels(mergedSamples$patient))
diag_cols &lt;- readRDS(&quot;~/GSE800098/output/diag_cols.rds&quot;)</code></pre>
<pre class="r"><code>treat_shapes &lt;- c(
  Veh = 1,
  DHT = 19,
  E2 = 15,
  `E2+DHT` = 17
)</code></pre>
<pre class="r"><code># counts &lt;- here::here(&quot;data/aligned/counts/counts.out.gz&quot;) %&gt;%
#   gzfile() %&gt;%
#   read_tsv(comment = &quot;#&quot;) %&gt;%
#   dplyr::select(Geneid, ends_with(&quot;bam&quot;)) %&gt;%
#   rename_at(vars(ends_with(&quot;bam&quot;)), dirname) %&gt;%
#   rename_all(basename) %&gt;%
#   column_to_rownames(&quot;Geneid&quot;)</code></pre>
<pre class="r"><code># mergedCounts &lt;- counts %&gt;%
#   rownames_to_column(&quot;gene_id&quot;) %&gt;%
#   pivot_longer(
#     cols = -gene_id,
#     names_to = &quot;Filename&quot;,
#     values_to = &quot;counts&quot;
#   ) %&gt;%
#   left_join(samples, by = &quot;Filename&quot;) %&gt;%
#   group_by(
#     gene_id, name, patient, treat, Tumor) %&gt;%
#   summarise(counts = sum(counts), .groups = &quot;drop&quot;) %&gt;%
#   pivot_wider(
#     id_cols = gene_id,
#     values_from = counts,
#     names_from = name
#   ) %&gt;%
#   column_to_rownames(&quot;gene_id&quot;)
# saveRDS(mergedCounts, here::here(&quot;data/mergedCounts.rds&quot;))
mergedCounts &lt;- readRDS(here::here(&quot;data/mergedCounts.rds&quot;))</code></pre>
<pre><code>           used  (Mb) gc trigger   (Mb) limit (Mb) max used   (Mb)
Ncells 12419681 663.3   22151544 1183.1         NA 20207713 1079.3
Vcells 28890979 220.5   46391808  354.0     204800 38538054  294.1</code></pre>
<p>The filtered and normalised <code>DGEList</code> was loaded in.</p>
<pre class="r"><code>dge_tumor &lt;- readRDS(here::here(&quot;output/dge_tumor.rds&quot;))</code></pre>
<pre class="r"><code>cqNorma_tumor &lt;- with(
  dge_tumor,
  cqn(
    counts= counts,
    x = genes$gc_content,
    lengths = genes$ave_tx_len
  )
)
dge_tumor$offset &lt;- cqNorma_tumor$glm.offset
logCPM_tumor &lt;- cqNorma_tumor$y + cqNorma_tumor$offset</code></pre>
<pre class="r"><code>pcaPost_tumor &lt;- logCPM_tumor %&gt;%
  t() %&gt;%
  prcomp() </code></pre>
</div>
<div id="reference-data" class="section level2">
<h2>Reference data</h2>
<p>Annotaed scRNA-seq data was retrieved from <a
href="https://www.nature.com/articles/s41597-022-01236-2"><em>R code and
downstream analysis objects for the scRNA-seq atlas of normal and
tumorigenic human breast tissue</em></a>.</p>
<p>Firstly, two separate <code>Seruat</code> objects, both containing
total cells were downloaded, one from 13 reduction mammoplasty tissues
(8 pre-menopausal &amp; 3 post-menopausal), and the other one from 13
naive ER+ tumor..</p>
<p>Information about each tissue, including if total or only epithelial
cells were sequenced, the condition, diagnosis (only for ER+ tummor),
gender, menopausal status, and age was read in as a
<code>data.frame</code>.</p>
<p>The full tissue metadata table is displayed below:</p>
<pre class="r"><code>Seruat_ERTotal &lt;- readRDS(here::here(&quot;data/scRNA_ref/WEHI/SeuratObject_ERTotal.rds&quot;))
Seruat_NormTotal &lt;- readRDS(here::here(&quot;data/scRNA_ref/WEHI/SeuratObject_NormTotal.rds&quot;))
tissueInfo &lt;- readxl::read_xlsx(here::here(&quot;data/scRNA_ref/WEHI/sampleInfo.xlsx&quot;), sheet = 1) %&gt;%
    mutate(
        Specimen_ID = vapply(.$`Sample Name`, function(x){
            str_remove_all(x, &quot;-epi|-total&quot;)
        }, character(1))
    ) %&gt;%
    dplyr::select(-`Sample Name`) %&gt;%
    left_join(read_table(here::here(&quot;data/scRNA_ref/WEHI/patientAge.txt&quot;))) %&gt;%
    relocate(
        Specimen_ID, .before = Patient
    )</code></pre>
<pre class="r"><code>tissueInfo %&gt;%
    mutate_if(is.character, as.factor) %&gt;%
    datatable(
        filter = &quot;top&quot;, 
        caption = htmltools::tags$caption(
                  htmltools::em(
                      &quot;Tissue metadata of all 69 breast tissue sequenced in Pal(2021)SingleCellRNA&quot;)
              )
    )</code></pre>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-1a018cce38817de8328d" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-1a018cce38817de8328d">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td><\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;B1-0023&quot;,&quot;B1-0033&quot;,&quot;B1-0090&quot;,&quot;B1-0894&quot;,&quot;ER-0001&quot;,&quot;ER-0025&quot;,&quot;ER-0029-7C&quot;,&quot;ER-0029-9C&quot;,&quot;ER-0032&quot;,&quot;ER-0040-LN&quot;,&quot;ER-0040-T&quot;,&quot;ER-0042&quot;,&quot;ER-0043-LN&quot;,&quot;ER-0043-T&quot;,&quot;ER-0056-LN&quot;,&quot;ER-0056-T&quot;,&quot;ER-0064-LN&quot;,&quot;ER-0064-T&quot;,&quot;ER-0114-T3&quot;,&quot;ER-0125&quot;,&quot;ER-0151&quot;,&quot;ER-0163&quot;,&quot;ER-0167-LN&quot;,&quot;ER-0167-T&quot;,&quot;ER-0173-LN&quot;,&quot;ER-0173-T&quot;,&quot;ER-0319&quot;,&quot;ER-0360&quot;,&quot;HER2-0031&quot;,&quot;HER2-0069&quot;,&quot;HER2-0161&quot;,&quot;HER2-0176&quot;,&quot;HER2-0308&quot;,&quot;HER2-0337&quot;,&quot;mER-0068-LN&quot;,&quot;mER-0068-T&quot;,&quot;mER-0178&quot;,&quot;N-0019&quot;,&quot;N-0021&quot;,&quot;N-0064&quot;,&quot;N-0092&quot;,&quot;N-0093&quot;,&quot;N-0123&quot;,&quot;N-0169&quot;,&quot;N-0230.16&quot;,&quot;N-0230.17&quot;,&quot;N-0233&quot;,&quot;N-0275&quot;,&quot;N-0280&quot;,&quot;N-0288&quot;,&quot;N-0342&quot;,&quot;N-0372&quot;,&quot;N-0408&quot;,&quot;N-1105&quot;,&quot;N-1469&quot;,&quot;TN-0106&quot;,&quot;TN-0114-T2&quot;,&quot;TN-0126&quot;,&quot;TN-0135&quot;,&quot;TN-B1-0131&quot;,&quot;TN-B1-0177&quot;,&quot;TN-B1-0554&quot;,&quot;TN-B1-4031&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;0001&quot;,&quot;0019&quot;,&quot;0021&quot;,&quot;0023&quot;,&quot;0025&quot;,&quot;0029&quot;,&quot;0031&quot;,&quot;0032&quot;,&quot;0033&quot;,&quot;0040&quot;,&quot;0042&quot;,&quot;0043&quot;,&quot;0056&quot;,&quot;0064&quot;,&quot;0068&quot;,&quot;0069&quot;,&quot;0090&quot;,&quot;0092&quot;,&quot;0093&quot;,&quot;0106&quot;,&quot;0114&quot;,&quot;0125&quot;,&quot;0126&quot;,&quot;0131&quot;,&quot;0135&quot;,&quot;0151&quot;,&quot;0161&quot;,&quot;0163&quot;,&quot;0167&quot;,&quot;0169&quot;,&quot;0173&quot;,&quot;0176&quot;,&quot;0177&quot;,&quot;0178&quot;,&quot;0230.16&quot;,&quot;0230.17&quot;,&quot;0233&quot;,&quot;0275&quot;,&quot;0280&quot;,&quot;0288&quot;,&quot;0308&quot;,&quot;0319&quot;,&quot;0337&quot;,&quot;0342&quot;,&quot;0360&quot;,&quot;0372&quot;,&quot;0408&quot;,&quot;0554&quot;,&quot;0894&quot;,&quot;1105&quot;,&quot;1469&quot;,&quot;4031&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;ER+ tumor, total cells&quot;,&quot;HER2+ tumor, total cells&quot;,&quot;Lymph node, total cells&quot;,&quot;Normal BRCA1+/- pre-neoplastic, total cells&quot;,&quot;Normal tissue, epithelial cells&quot;,&quot;Normal tissue, total cells&quot;,&quot;Triple negative tumor (BRCA1), total cells&quot;,&quot;Triple negative tumor, total cells&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;Epithelial&quot;,&quot;Lymph node&quot;,&quot;Total&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;ER+ tumor&quot;,&quot;HER2+ tumor&quot;,&quot;Involved LN&quot;,&quot;Normal&quot;,&quot;Normal BRCA1+/- pre-neoplastic&quot;,&quot;Triple negative (BRCA1) tumor&quot;,&quot;Triple negative tumor&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;NA&quot;,&quot;Post&quot;,&quot;Pre&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;NA&quot;,&quot;Nulliparous&quot;,&quot;Parous&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;Female&quot;,&quot;Male&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"618\" data-max=\"21130\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"456\" data-max=\"16685\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"12778\" data-max=\"19577\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"19\" data-max=\"84\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;-&quot;,&quot;Invasive&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n<\/tr>","caption":"<caption>\n  <em>Tissue metadata of all 69 breast tissue sequenced in Pal(2021)SingleCellRNA<\/em>\n<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69"],["N-0092","N-0019","N-0280","N-0093","N-0093","N-1469","N-0408","N-1105","N-0230.17","N-0064","N-0064","N-0230.16","N-0233","N-0169","N-0123","N-0123","N-0342","N-0342","N-0288","N-0021","N-0275","N-0275","N-0372","N-0372","B1-0894","B1-0033","B1-0023","B1-0090","TN-0126","TN-0135","TN-0106","TN-0114-T2","TN-B1-4031","TN-B1-0131","TN-B1-0554","TN-B1-0177","HER2-0308","HER2-0337","HER2-0031","HER2-0069","HER2-0161","HER2-0176","ER-0319","ER-0001","ER-0125","ER-0360","ER-0114-T3","ER-0032","ER-0042","ER-0025","ER-0151","ER-0163","ER-0029-7C","ER-0029-9C","ER-0040-T","ER-0040-LN","ER-0043-T","ER-0043-LN","ER-0056-T","ER-0056-LN","ER-0064-T","ER-0064-LN","ER-0167-T","ER-0167-LN","ER-0173-T","ER-0173-LN","mER-0178","mER-0068-T","mER-0068-LN"],["0092","0019","0280","0093","0093","1469","0408","1105","0230.17","0064","0064","0230.16","0233","0169","0023","0023","0342","0342","0288","0021","0275","0275","0372","0372","0894","0033","0023","0090","0126","0135","0106","0114","4031","0131","0554","0177","0308","0337","0031","0069","0161","0176","0319","0001","0125","0360","0114","0032","0042","0025","0151","0163","0029","0029","0040","0040","0043","0043","0056","0056","0064","0064","0167","0167","0173","0173","0178","0068","0068"],["Normal tissue, total cells","Normal tissue, total cells","Normal tissue, epithelial cells","Normal tissue, epithelial cells","Normal tissue, total cells","Normal tissue, epithelial cells","Normal tissue, epithelial cells","Normal tissue, epithelial cells","Normal tissue, total cells","Normal tissue, epithelial cells","Normal tissue, total cells","Normal tissue, epithelial cells","Normal tissue, total cells","Normal tissue, total cells","Normal tissue, epithelial cells","Normal tissue, total cells","Normal tissue, epithelial cells","Normal tissue, total cells","Normal tissue, total cells","Normal tissue, total cells","Normal tissue, epithelial cells","Normal tissue, total cells","Normal tissue, epithelial cells","Normal tissue, total cells","Normal BRCA1+/- pre-neoplastic, total cells","Normal BRCA1+/- pre-neoplastic, total cells","Normal BRCA1+/- pre-neoplastic, total cells","Normal BRCA1+/- pre-neoplastic, total cells","Triple negative tumor, total cells","Triple negative tumor, total cells","Triple negative tumor, total cells","Triple negative tumor, total cells","Triple negative tumor (BRCA1), total cells","Triple negative tumor (BRCA1), total cells","Triple negative tumor (BRCA1), total cells","Triple negative tumor (BRCA1), total cells","HER2+ tumor, total cells","HER2+ tumor, total cells","HER2+ tumor, total cells","HER2+ tumor, total cells","HER2+ tumor, total cells","HER2+ tumor, total cells","ER+ tumor, total cells","ER+ tumor, total cells","ER+ tumor, total cells","ER+ tumor, total cells","ER+ tumor, total cells","ER+ tumor, total cells","ER+ tumor, total cells","ER+ tumor, total cells","ER+ tumor, total cells","ER+ tumor, total cells","ER+ tumor, total cells","ER+ tumor, total cells","ER+ tumor, total cells","Lymph node, total cells","ER+ tumor, total cells","Lymph node, total cells","ER+ tumor, total cells","Lymph node, total cells","ER+ tumor, total cells","Lymph node, total cells","ER+ tumor, total cells","Lymph node, total cells","ER+ tumor, total cells","Lymph node, total cells","ER+ tumor, total cells","ER+ tumor, total cells","Lymph node, total cells"],["Total","Total","Epithelial","Epithelial","Total","Epithelial","Epithelial","Epithelial","Total","Epithelial","Total","Epithelial","Total","Total","Epithelial","Total","Epithelial","Total","Total","Total","Epithelial","Total","Epithelial","Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Total","Lymph node","Total","Lymph node","Total","Lymph node","Total","Lymph node","Total","Lymph node","Total","Total","Lymph node"],["Normal","Normal","Normal","Normal","Normal","Normal","Normal","Normal","Normal","Normal","Normal","Normal","Normal","Normal","Normal","Normal","Normal","Normal","Normal","Normal","Normal","Normal","Normal","Normal","Normal BRCA1+/- pre-neoplastic","Normal BRCA1+/- pre-neoplastic","Normal BRCA1+/- pre-neoplastic","Normal BRCA1+/- pre-neoplastic","Triple negative tumor","Triple negative tumor","Triple negative tumor","Triple negative tumor","Triple negative (BRCA1) tumor","Triple negative (BRCA1) tumor","Triple negative (BRCA1) tumor","Triple negative (BRCA1) tumor","HER2+ tumor","HER2+ tumor","HER2+ tumor","HER2+ tumor","HER2+ tumor","HER2+ tumor","ER+ tumor","ER+ tumor","ER+ tumor","ER+ tumor","ER+ tumor","ER+ tumor","ER+ tumor","ER+ tumor","ER+ tumor","ER+ tumor","ER+ tumor","ER+ tumor","ER+ tumor","Involved LN","ER+ tumor","Involved LN","ER+ tumor","Involved LN","ER+ tumor","Involved LN","ER+ tumor","Involved LN","ER+ tumor","Involved LN","ER+ tumor","ER+ tumor","Involved LN"],["Pre","Pre","Pre","Pre","Pre","Pre","Pre","Pre","Pre","Pre","Pre","Pre","Pre","Pre","Pre","Pre","Post","Post","Post","Post","Post","Post","Post","Post","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA"],["Nulliparous","Nulliparous","Nulliparous","Nulliparous","Nulliparous","Nulliparous","Nulliparous","Nulliparous","Nulliparous","Nulliparous","Nulliparous","Parous","Parous","Parous","Parous","Parous","Nulliparous","Nulliparous","Nulliparous","Parous","Parous","Parous","Parous","Parous","Nulliparous","Nulliparous","Parous","Parous","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA","NA"],["Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Female","Male","Male","Male"],[4966,7130,1198,9879,7412,3015,3089,5201,3443,4785,1678,2157,5665,4605,3037,7371,7856,10178,2320,2463,6401,1914,12212,4825,7850,6827,5932,5663,3666,15870,1065,2015,5581,6456,9593,21130,4779,12288,4737,618,6561,19027,3406,7393,7289,6676,8822,1445,5439,10592,3357,7334,3395,8012,6468,6196,5572,912,4140,7241,4037,2411,13848,8422,9872,3843,13510,1013,4854],[4443,6361,1109,8319,5130,2858,3004,4938,3292,4047,1336,1847,5199,4267,2831,6498,7505,9076,1904,1551,6096,1469,11162,3806,7330,5931,5019,4960,2766,15212,879,1575,5228,5688,6787,16685,4160,8252,4084,456,4863,10103,2846,5754,4226,3714,7723,1108,4529,8102,2527,5990,3102,7261,5904,5661,4185,745,2704,6611,3603,2328,7014,4392,7766,3257,7274,802,4720],[18128,18628,16321,18483,18651,16323,16143,17494,18121,15342,15540,15382,19187,18610,16163,17956,18439,18968,16950,15221,17268,16916,18453,18770,17860,17434,17537,17867,18090,18431,17052,17430,17743,17571,19462,19518,18316,18219,17757,15355,18353,19577,18231,18090,17479,17894,18626,17159,16860,18312,18616,17420,17421,17565,16944,17468,17711,14987,14422,15870,16689,15736,18355,18006,19316,19054,15887,12778,15223],[19,21,21,22,22,23,24,26,30,44,44,20,34,35,48,48,55,55,69,48,48,48,49,49,30,31,42,43,64,61,65,84,25,84,29,30,32,66,47,71,80,60,58,58,45,70,84,55,58,52,49,45,59,59,69,69,55,55,66,66,65,65,83,83,83,83,84,79,79],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"Invasive","Invasive","Invasive","Invasive","Invasive","Invasive","Invasive","Invasive","Invasive","Invasive","Invasive","Invasive","Invasive","-","Invasive","-","Invasive","-","Invasive","-","Invasive","-","Invasive","-","Invasive","Invasive","-"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Specimen_ID<\/th>\n      <th>Patient<\/th>\n      <th>Description<\/th>\n      <th>Source<\/th>\n      <th>Condition<\/th>\n      <th>Menopause<\/th>\n      <th>Parity<\/th>\n      <th>Gender<\/th>\n      <th>Number of Cells<\/th>\n      <th>Number of Cells After Filtering<\/th>\n      <th>Number of Genes Detected<\/th>\n      <th>Age<\/th>\n      <th>Diagnosis<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[9,10,11,12]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>Seruat_ERsub &lt;- readRDS(here::here(&quot;data/scRNA_ref/WEHI/SeuratObject_ERTotalSub.rds&quot;))
Seruat_Normsub &lt;- readRDS(here::here(&quot;data/scRNA_ref/WEHI/SeuratObject_NormTotalSub.rds&quot;))</code></pre>
<p>Among normal tissues, 8 clusters were identified, cluster 1, 3, 4
were identified to be epithelial cell through ECPAM expression, where
cluster 4 was mature luminal (ML), cluster 1 was luminal progenitor and
cluster 3 was basal cells. The other EPCAM-negative non-epithelial cells
were then reclustered to study the tumor microenvironment, where 7
sub-clusters were identified. <code>Seurat</code> object for the
non-epithelail cells subclustering result was also readed in.
Visualisations of the two-step clustering results and the cluster
annotation can be found <a
href="https://www.embopress.org/cms/asset/e2ad3a16-4bd0-4cb5-9af0-17d5e77c9735/embj2020107333-fig-0002-m.jpg">in
this figure</a>. Note: when the non-epithelial cells were reclusterd, a
really small subset of cells (<code>r</code>nrow(<a
href="mailto:Seruat_Normsub@meta.data"
class="email">Seruat_Normsub@meta.data</a> %&gt;%
dplyr::filter(seurat_clusters == 5))` cells) were also labelled as
Epithelial cells. For simplicity, those cells were excluded and not used
in the reference dataset for deconvolution.</p>
<p>Among ER<sup>+</sup> tumor cells, 11 clusters were identified, where
1,5,6 and 7 were defined to be epithalial cells. Out of those four
epithelial clusters, cluster 1 and 5 were defined to be tumor epithelial
cells by inferring CNB, and 6 and 7 were defined to be normal
epithelial. The non-epithelial cells were also reclustered into 13
subcluster and annotated. Visualisations for this part of the analysis
are: <a
href="https://www.embopress.org/cms/asset/f035e7db-cd0f-46a6-a820-f96c055d3e24/embj2020107333-fig-0006-m.jpg">Figure
6</a> and <a
href="https://www.embopress.org/cms/asset/1a957da0-974c-4a0d-b230-6458d8bcf22d/embj2020107333-fig-0007-m.jpg">Figure
7</a>.</p>
<p>The <code>Seruat</code> object contained which cluster cells were
assigned to but not the cluster annotation. So I manually put the
annotations in.</p>
<pre class="r"><code>Seruat_NormTotal@meta.data &lt;- Seruat_NormTotal@meta.data %&gt;%
    mutate(
        seurat_clusters = as.numeric(as.character(seurat_clusters)) +1,
        cell_type = ifelse(seurat_clusters %in% c(1, 3, 4), &quot;Epithelial&quot;, &quot;Non-Epithelial&quot;)
    ) %&gt;%
    rownames_to_column(&quot;cell_id&quot;)
Seruat_Normsub@meta.data &lt;- Seruat_Normsub@meta.data %&gt;%
    mutate(
        seurat_clusters = as.numeric(as.character(seurat_clusters)) +1L
    ) %&gt;%
    rownames_to_column(&quot;cell_id&quot;) %&gt;%
    left_join(
        data_frame(
            seurat_clusters = c(1:7),
            cell_type = c(
                &quot;Fibroblasts&quot;, &quot;Endothelial_vasc&quot;, &quot;Pericytes&quot;, &quot;Myeloid&quot;,
                &quot;Lymphoid&quot;, &quot;Epi&quot;, &quot;Endothelial_lymph&quot;
            )
        )
    )  %&gt;%
    dplyr::filter(seurat_clusters != 6)
Seruat_ERTotal@meta.data &lt;- Seruat_ERTotal@meta.data %&gt;%
    mutate(
        seurat_clusters = as.numeric(as.character(seurat_clusters)) +1,
        cell_type = case_when(
            seurat_clusters %in% c(1,5) ~ &quot;Tumor_Epithelial&quot;,
            seurat_clusters %in% c(6,7) ~ &quot;Normal_Epithelial&quot;,
            !seurat_clusters %in% c(1,5,6,7) ~ &quot;Non-Epithelial&quot;)
    ) %&gt;%
    rownames_to_column(&quot;cell_id&quot;)
Seruat_ERsub@meta.data &lt;- Seruat_ERsub@meta.data %&gt;%
    mutate(
        seurat_clusters = as.numeric(as.character(seurat_clusters)) +1L
    ) %&gt;%
    rownames_to_column(&quot;cell_id&quot;) %&gt;%
    left_join(
        data_frame(
            seurat_clusters = c(1:13),
            cell_type = c(
                &quot;T cells&quot;, &quot;TAMs&quot;, &quot;CAFs&quot;, &quot;Pericytes&quot;, NA, &quot;Endothelial&quot;, &quot;TAMs&quot;,
                &quot;B cells&quot;, &quot;Myeloid&quot;, &quot;CAFs&quot;, &quot;Plasma cells&quot;, NA, NA
            )
        )
    )</code></pre>
<pre class="r"><code># rm(Seruat_ERsub, 
#    Seruat_ERTotal, 
#    Seruat_Normsub, 
#    Seruat_NormTotal)</code></pre>
<div id="subset-the-reference" class="section level3">
<h3>Subset the reference</h3>
<p>Since the full scRNA-seq data contains too many cells and is too big
to handle on my local machine, I randomly sampled 10% of each cell type
to form the reference datasets for deconvolution for both the ER+ tumor
and the normal tissues.</p>
<pre class="r"><code># set.seed(123)
# normalCell_meta &lt;- Seruat_NormTotal@meta.data %&gt;%
#     dplyr::filter(cell_type == &quot;Epithelial&quot;) %&gt;%
#     sample_frac(0.1) %&gt;%
#     rbind(
#         Seruat_Normsub@meta.data %&gt;%
#             group_by(cell_type) %&gt;%
#             sample_frac(0.1)
#     )
# normalCell_mat &lt;- GetAssayData(Seruat_NormTotal, slot = &#39;data&#39;) %&gt;%
#     .[ ,colnames(.) %in% normalCell_meta$cell_id] %&gt;%
#     .[,match(normalCell_meta$cell_id, colnames(.))] %&gt;%
#     as.matrix() 
# saveRDS(normalCell_mat, here::here(&quot;data/scRNA_ref/WEHI/normalCell_mat.rds&quot;))
# saveRDS(normalCell_meta, here::here(&quot;data/scRNA_ref/WEHI/normalCell_meta.rds&quot;))
normalCell_mat &lt;- readRDS(here::here(&quot;data/scRNA_ref/WEHI/normalCell_mat.rds&quot;))
normalCell_meta &lt;- readRDS(here::here(&quot;data/scRNA_ref/WEHI/normalCell_meta.rds&quot;))</code></pre>
<pre class="r"><code># set.seed(123)
# ERCell_meta &lt;- Seruat_ERTotal@meta.data %&gt;%
#     dplyr::filter(cell_type %in% c(&quot;Tumor_Epithelial&quot;, &quot;Normal_Epithelial&quot;)) %&gt;%
#     sample_frac(0.1) %&gt;%
#     rbind(
#         Seruat_ERsub@meta.data %&gt;%
#             drop_na() %&gt;%
#             group_by(cell_type) %&gt;%
#             sample_frac(0.1)
#     )
# temp &lt;- GetAssayData(Seruat_ERTotal, slot = &#39;data&#39;) %&gt;%
#     .[ ,colnames(.) %in% ERCell_meta$cell_id] %&gt;%
#     .[,match(ERCell_meta$cell_id, colnames(.))] %&gt;%
#     as.matrix() 
# saveRDS(temp, here::here(&quot;data/scRNA_ref/WEHI/ERCell_mat.rds&quot;))
# saveRDS(ERCell_meta, here::here(&quot;data/scRNA_ref/WEHI/ERCell_meta.rds&quot;))
ERCell_mat &lt;- readRDS(here::here(&quot;data/scRNA_ref/WEHI/ERCell_mat.rds&quot;))
ERCell_meta &lt;- readRDS(here::here(&quot;data/scRNA_ref/WEHI/ERCell_meta.rds&quot;))</code></pre>
<pre class="r"><code>rm(Seruat_ERsub, 
   Seruat_ERTotal, 
   Seruat_Normsub, 
   Seruat_NormTotal)</code></pre>
<pre class="r"><code>cell_cols_tot &lt;- hcl.colors(
    n = length(unique(normalCell_meta$cell_type)), 
    palette = &quot;Zissou 1&quot;) %&gt;%
    setNames(unique(normalCell_meta$cell_type) %&gt;%
                        str_replace_all(&quot;_&quot;, &quot; &quot;))
cell_cols_ER &lt;- distinctColorPalette(length(unique(ERCell_meta$cell_type))) %&gt;%
    setNames(unique(ERCell_meta$cell_type) %&gt;%
                        str_replace_all(&quot;_&quot;, &quot; &quot;))</code></pre>
</div>
</div>
</div>
<div id="dwls" class="section level1">
<h1>DWLS</h1>
<p>In DWLS, gene signatures of each cell type are firstly determined
through differential expression analysis, which could be undertaken
using either Seurat or MAST. Mean of expressions of those signatures
detected across cells assigned to be the same cell type will be taken
and those mean expressions are used to model the bulk RNA-seq expression
through a weighted least square approach.</p>
<p><code>DWLS</code> offers two options for the signature gene
selection, one using MAST while the other one uses Seruat. The MAST
options requires the raw scRNA-seq count as input, which was what was <a
href="CellDecon_garvan.html">available in the Garvan dataset</a>. But in
this study, only the log-transformed and scaled data were provided with
the <code>Seruat</code> object, so the
<code>buildSignatureMatrixUsingSeurat</code> option was chosen
instead.</p>
<div id="tumor" class="section level2">
<h2>Tumor</h2>
<pre class="r"><code>mapID_test &lt;- mapIds(ensDb, keys = rownames(logCPM_tumor), column = &quot;GENENAME&quot;, keytype = &quot;GENEID&quot;) %&gt;%
    .[!is.na(.)]
index_test &lt;- pmatch(rownames(logCPM_tumor), names(mapID_test))
logCPM_tumor_n &lt;- logCPM_tumor
rownames(logCPM_tumor_n) &lt;- mapID_test[index_test]
logCPM_tumor_n &lt;- logCPM_tumor_n[!is.na(rownames(logCPM_tumor_n)), ]
# saveRDS(logCPM_tumor_n, here::here(&quot;data/scRNA_ref/garvan/logCPM_tumor_n.rds&quot;))</code></pre>
<pre class="r"><code>#replace the spaec in cell type annotatoin to avoid error in DWLS
ERCell_meta &lt;- ERCell_meta %&gt;%
    mutate(
        cell_type = str_replace(cell_type, &quot; &quot;, &quot;_&quot;)
    ) </code></pre>
<pre class="r"><code># result &lt;- here::here(&quot;output/DWLS/WEHI_ER/signature&quot;)
# Sig_tumor &lt;- buildSignatureMatrixUsingSeurat(scdata = ERCell_mat,
#                                      id = ERCell_meta$cell_type,
#                                      path= result,
#                                      diff.cutoff=0.5,
#                                      pval.cutoff=0.01)
# tr_tumor &lt;- sapply(colnames(logCPM_tumor_n), function(x){
#   logCPM_tumor_n[,x] %&gt;%
#     set_names(rownames(logCPM_tumor_n)) %&gt;%
#     trimData(Sig_tumor, .)
# }, simplify = FALSE)
# solDWLS_tumor &lt;- sapply(colnames(logCPM_tumor_n), function(x){
#   solveDampenedWLS(tr_tumor[[x]]$sig, tr_tumor[[x]]$bulk)
# }, simplify = FALSE)
# write_rds(solDWLS_tumor,
#           here::here(
#               &quot;output/DWLS/WEHI_ER/results_tumor.rds&quot;
#           ))</code></pre>
<pre class="r"><code>solDWLS_tumor &lt;- readRDS(here::here(&quot;output/DWLS/WEHI_ER/results_tumor.rds&quot;))
solDWLS_tumor %&gt;%
    bind_rows() %&gt;%
    mutate(Sample = names(solDWLS_tumor) ) %&gt;%
    pivot_longer(col = -&quot;Sample&quot;,
                 names_to = &quot;Cell Type&quot;, values_to = &quot;Proportion&quot;) %&gt;%
    mutate(treatment = vapply(.$Sample, 
                              function(x){str_split(x, &quot; &quot;)[[1]][2]}, character(1)), 
           patient = vapply(.$Sample, 
                            function(x){str_split(x, &quot; &quot;)[[1]][1]}, character(1)), 
           `Cell Type` = str_replace_all(`Cell Type`, &quot;_&quot;, &quot; &quot;)) %&gt;%
    ggplot(aes(x = Proportion, y= treatment, 
               fill = factor(`Cell Type`, levels = rev(c(&quot;Tumor Epithelial&quot;,&quot;Normal Epithelial&quot;,&quot;B cells&quot;, &quot;T cells&quot;,
                                                     &quot;Endothelial&quot;,&quot;CAFs&quot;, &quot;TAMs&quot;, &quot;Myeloid&quot;, &quot;Pericytes&quot;,&quot;Plasma cells&quot;))))) + 
    geom_bar(position=&quot;stack&quot;, stat=&quot;identity&quot;) +
    facet_wrap(~patient) +
    scale_fill_manual(values = cell_cols_ER, 
            limits = force,           
           name = &quot;Cell Type&quot;)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/CellDecon_WEHI.Rmd/unnamed-chunk-6-1.png" alt="*Cell type proportion estimated derived from DWLS for each tumor samples. Epithelial cell was the dominating cell type in all samples. Overall, more normal epithelials than cancer epithelial cells were detected.*" width="1152" />
<p class="caption">
<em>Cell type proportion estimated derived from DWLS for each tumor
samples. Epithelial cell was the dominating cell type in all samples.
Overall, more normal epithelials than cancer epithelial cells were
detected.</em>
</p>
</div>
<pre class="r"><code>WEHI_tumor &lt;- solDWLS_tumor %&gt;%
    bind_rows() %&gt;%
    mutate(name = names(solDWLS_tumor)
           # , 
           # Myeloid = Myeloid + TAMs
           ) %&gt;%
    # dplyr::select(-TAMs) %&gt;%
    pivot_longer(col = -&quot;name&quot;,
                 names_to = &quot;Cell Type&quot;, values_to = &quot;WEHI_Prop&quot;) </code></pre>
<p>Differences in the predicted cell type proportions between the two
groups of treatment-naive tumours: IDC-like and ILC-like tumours were
visualised.</p>
<pre class="r"><code>new_diag_col &lt;- qualitative_hcl(2, palette = &quot;Dark 3&quot;) %&gt;%
    set_names(c(&quot;IDC-like&quot;, &quot;ILC-like&quot;))
temp &lt;- WEHI_tumor %&gt;%
    left_join(tumor_sample %&gt;%
                  ungroup() %&gt;%
                  dplyr::select(name, 
                                Treatment = treat, Age, Diagnosis, patient)) %&gt;%
    mutate(
        `Cell Type` = str_replace_all(`Cell Type`, &quot;_&quot;, &quot; &quot;), 
        `Cell Type` = as.factor(`Cell Type`)) %&gt;%
    dplyr::filter(Treatment == &quot;Veh&quot;) %&gt;%
    group_by(`Cell Type`) %&gt;%
    mutate(ave = mean(WEHI_Prop)) %&gt;%
    ungroup()
cellOrder &lt;- temp %&gt;%
    dplyr::select(`Cell Type`, ave) %&gt;%
    unique() %&gt;%
    arrange(desc(ave)) %&gt;%
    pull(`Cell Type`)
# stat &lt;- temp %&gt;%
#     split(f = .$`Cell Type`) %&gt;%
#     lapply(dplyr::select, Diagnosis, WEHI_Prop, name) %&gt;%
#     lapply(pivot_wider, names_from = Diagnosis, values_from = WEHI_Prop)  %&gt;%
#     lapply(function(x){
#             t.test(
#                 x$`ILC-like`, x$`IDC-like`)$p.value
#     })
sup2a &lt;- temp %&gt;%
    mutate(WEHI_Prop = WEHI_Prop*100) %&gt;%
    ggplot(
        aes(
            factor(`Cell Type`, levels = cellOrder), 
            WEHI_Prop, fill = Diagnosis) 
    ) +
    geom_boxplot(
    ) +
    scale_fill_manual(values = new_diag_col, 
                      name = &quot;Subtype&quot;) + 
    scale_x_discrete(labels = function(x) str_wrap(x, width = 10)) +
    theme(
        axis.ticks.x = element_blank(), 
        panel.grid = element_blank(), 
        text = element_text(size = 22), 
        legend.position = &quot;none&quot;) +
    labs(
        # x = &quot;Cell Type&quot;,
        x = &quot;&quot;,
        y = &quot;Proportion (%)&quot;
    )
sup2a</code></pre>
<p><img src="figure/CellDecon_WEHI.Rmd/unnamed-chunk-8-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>T-tests were performed with each cell type to compare the mean
predicted cell type proportions between the two treatment groups. Only
endothelial cell had a raw p-value that is smaller than 0.05.</p>
<pre class="r"><code>temp %&gt;%
    split(f = .$`Cell Type`) %&gt;%
    lapply(dplyr::select, Diagnosis, WEHI_Prop, name) %&gt;%
    lapply(pivot_wider, names_from = Diagnosis, values_from = WEHI_Prop)  %&gt;%
    lapply(function(x){
            t.test(
                x$`ILC-like`, x$`IDC-like`)$p.value
    }) %&gt;%
    unlist() %&gt;%
    enframe(
        name = &quot;Cell Type&quot;, 
        value = &quot;p-value&quot;
    ) %&gt;%
    pander()</code></pre>
<table style="width:42%;">
<colgroup>
<col width="27%" />
<col width="13%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">Cell Type</th>
<th align="center">p-value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">B cells</td>
<td align="center">0.8696</td>
</tr>
<tr class="even">
<td align="center">CAFs</td>
<td align="center">0.4381</td>
</tr>
<tr class="odd">
<td align="center">Endothelial</td>
<td align="center">0.0092</td>
</tr>
<tr class="even">
<td align="center">Myeloid</td>
<td align="center">0.2858</td>
</tr>
<tr class="odd">
<td align="center">Normal Epithelial</td>
<td align="center">0.8707</td>
</tr>
<tr class="even">
<td align="center">Pericytes</td>
<td align="center">0.246</td>
</tr>
<tr class="odd">
<td align="center">Plasma cells</td>
<td align="center">0.2455</td>
</tr>
<tr class="even">
<td align="center">T cells</td>
<td align="center">0.6207</td>
</tr>
<tr class="odd">
<td align="center">TAMs</td>
<td align="center">0.8332</td>
</tr>
<tr class="even">
<td align="center">Tumor Epithelial</td>
<td align="center">0.3788</td>
</tr>
</tbody>
</table>
<pre class="r"><code>library(ggpubr)
total_immune &lt;- solDWLS_tumor %&gt;%
    bind_rows() %&gt;%
    mutate(
        name = names(solDWLS_tumor), 
        total_immune = (B_cells + Myeloid + T_cells + TAMs)*100) %&gt;%
    dplyr::select(name, total_immune)  %&gt;%
    left_join(tumor_sample %&gt;%
                  ungroup() %&gt;%
                  dplyr::select(name,  treat, Diagnosis)) %&gt;%
    dplyr::filter(treat == &quot;Veh&quot;) 
temp &lt;- total_immune %&gt;%
    dplyr::select(-treat) %&gt;%
    pivot_wider(names_from = &quot;Diagnosis&quot;, values_from = &quot;total_immune&quot;) 
immune_stat &lt;- t.test(temp$`ILC-like`, temp$`IDC-like`)$p.value</code></pre>
<p>The total predicted immune proportion was derived by summing the
predicted proportions of B cells, myeloid, T cells and TAMs within each
sample. The difference in mean immune proportions between the two tumour
groups is not statistically significant (p-value = 0.8825345).</p>
<pre class="r"><code>sup2b &lt;- total_immune %&gt;%
    ggplot(
        aes(Diagnosis, total_immune, fill = Diagnosis)
    ) +
    geom_boxplot() +
    scale_fill_manual(values = new_diag_col, 
                      name = &quot;Subtype&quot;) +
    labs(
        # x = &quot;Cell Type&quot;,
        x = &quot;&quot;,
        y = &quot;Total Immune Proportion (%)&quot;
    )
sup2b</code></pre>
<p><img src="figure/CellDecon_WEHI.Rmd/unnamed-chunk-11-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="thesis-figure" class="section level1">
<h1>Thesis figure</h1>
<p>Code used to generate figures for the thesis.</p>
<pre class="r"><code># sup2 &lt;- (sup2a | sup2b) +
#     plot_annotation(tag_levels = &quot;A&quot;) +
#     plot_layout(
#         widths = c(0.8, 0.2), 
#         guides = &quot;collect&quot;
#     ) &amp;
#     theme(
#         plot.tag = element_text(size = 26, face = &quot;bold&quot; ), 
#         text = element_text(size = 24)
#     )
# # png(
# #     &quot;/Users/wenjunliu/PhD_thesis/Images/chapter_04/deconvolute.png&quot;,
# #     width = 500, height =150, units=&#39;mm&#39;, res = 300
# # )
# # sup2
# # dev.off()</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21)
Platform: x86_64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.3.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Australia/Adelaide
tzcode source: internal

attached base packages:
[1] splines   stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] MAST_1.26.0                 SingleCellExperiment_1.22.0
 [3] varhandle_2.0.5             ROCR_1.0-11                
 [5] e1071_1.7-13                reshape_0.8.9              
 [7] quadprog_1.5-8              pheatmap_1.0.12            
 [9] ggforce_0.4.1               randomcoloR_1.1.0.1        
[11] ggpubr_0.6.0                colorspace_2.1-0           
[13] plotly_4.10.2               corrplot_0.92              
[15] Matrix_1.6-0                SeuratObject_4.1.3         
[17] Seurat_4.3.0.1              dtangle_2.0.9              
[19] celldex_1.10.1              SummarizedExperiment_1.30.2
[21] MatrixGenerics_1.12.2       matrixStats_1.0.0          
[23] DT_0.28                     ggrepel_0.9.3              
[25] cqn_1.46.0                  quantreg_5.95              
[27] SparseM_1.81                preprocessCore_1.62.1      
[29] nor1mix_1.3-0               mclust_6.0.0               
[31] magrittr_2.0.3              ggfortify_0.4.16           
[33] cowplot_1.1.1               ensembldb_2.24.0           
[35] AnnotationFilter_1.24.0     GenomicFeatures_1.52.1     
[37] AnnotationDbi_1.62.2        Biobase_2.60.0             
[39] GenomicRanges_1.52.0        GenomeInfoDb_1.36.1        
[41] IRanges_2.34.1              S4Vectors_0.38.1           
[43] AnnotationHub_3.8.0         BiocFileCache_2.8.0        
[45] dbplyr_2.3.3                BiocGenerics_0.46.0        
[47] edgeR_3.42.4                limma_3.56.2               
[49] glue_1.6.2                  pander_0.6.5               
[51] scales_1.2.1                yaml_2.3.7                 
[53] lubridate_1.9.2             forcats_1.0.0              
[55] stringr_1.5.0               dplyr_1.1.2                
[57] purrr_1.0.1                 readr_2.1.4                
[59] tidyr_1.3.0                 tibble_3.2.1               
[61] ggplot2_3.4.2               tidyverse_2.0.0            

loaded via a namespace (and not attached):
  [1] fs_1.6.2                      ProtGenerics_1.32.0          
  [3] spatstat.sparse_3.0-2         bitops_1.0-7                 
  [5] httr_1.4.6                    RColorBrewer_1.1-3           
  [7] backports_1.4.1               tools_4.3.0                  
  [9] sctransform_0.3.5             utf8_1.2.3                   
 [11] R6_2.5.1                      lazyeval_0.2.2               
 [13] uwot_0.1.16                   withr_2.5.0                  
 [15] sp_2.0-0                      prettyunits_1.1.1            
 [17] gridExtra_2.3                 progressr_0.13.0             
 [19] cli_3.6.1                     spatstat.explore_3.2-1       
 [21] labeling_0.4.2                sass_0.4.6                   
 [23] spatstat.data_3.0-1           proxy_0.4-27                 
 [25] ggridges_0.5.4                pbapply_1.7-2                
 [27] Rsamtools_2.16.0              parallelly_1.36.0            
 [29] readxl_1.4.3                  rstudioapi_0.15.0            
 [31] RSQLite_2.3.1                 generics_0.1.3               
 [33] BiocIO_1.10.0                 crosstalk_1.2.0              
 [35] vroom_1.6.3                   ica_1.0-3                    
 [37] spatstat.random_3.1-5         car_3.1-2                    
 [39] fansi_1.0.4                   abind_1.4-5                  
 [41] lifecycle_1.0.3               carData_3.0-5                
 [43] Rtsne_0.16                    grid_4.3.0                   
 [45] blob_1.2.4                    promises_1.2.0.1             
 [47] ExperimentHub_2.8.0           crayon_1.5.2                 
 [49] miniUI_0.1.1.1                lattice_0.21-8               
 [51] KEGGREST_1.40.0               pillar_1.9.0                 
 [53] knitr_1.43                    rjson_0.2.21                 
 [55] future.apply_1.11.0           codetools_0.2-19             
 [57] leiden_0.4.3                  V8_4.3.2                     
 [59] data.table_1.14.8             vctrs_0.6.3                  
 [61] png_0.1-8                     cellranger_1.1.0             
 [63] gtable_0.3.3                  cachem_1.0.8                 
 [65] xfun_0.39                     S4Arrays_1.0.4               
 [67] mime_0.12                     survival_3.5-5               
 [69] interactiveDisplayBase_1.38.0 ellipsis_0.3.2               
 [71] fitdistrplus_1.1-11           nlme_3.1-162                 
 [73] bit64_4.0.5                   progress_1.2.2               
 [75] filelock_1.0.2                RcppAnnoy_0.0.21             
 [77] rprojroot_2.0.3               bslib_0.5.0                  
 [79] irlba_2.3.5.1                 KernSmooth_2.23-21           
 [81] DBI_1.1.3                     tidyselect_1.2.0             
 [83] bit_4.0.5                     compiler_4.3.0               
 [85] curl_5.0.1                    git2r_0.32.0                 
 [87] xml2_1.3.5                    DelayedArray_0.26.6          
 [89] rtracklayer_1.60.0            lmtest_0.9-40                
 [91] rappdirs_0.3.3                digest_0.6.33                
 [93] goftest_1.2-3                 spatstat.utils_3.0-3         
 [95] rmarkdown_2.23                XVector_0.40.0               
 [97] htmltools_0.5.5               pkgconfig_2.0.3              
 [99] sparseMatrixStats_1.12.2      highr_0.10                   
[101] fastmap_1.1.1                 rlang_1.1.1                  
[103] htmlwidgets_1.6.2             shiny_1.7.4.1                
[105] DelayedMatrixStats_1.22.1     farver_2.1.1                 
[107] jquerylib_0.1.4               zoo_1.8-12                   
[109] jsonlite_1.8.7                BiocParallel_1.34.2          
[111] RCurl_1.98-1.12               GenomeInfoDbData_1.2.10      
[113] patchwork_1.1.2               munsell_0.5.0                
[115] Rcpp_1.0.11                   reticulate_1.30              
[117] stringi_1.7.12                zlibbioc_1.46.0              
[119] MASS_7.3-60                   plyr_1.8.8                   
[121] parallel_4.3.0                listenv_0.9.0                
[123] deldir_1.0-9                  Biostrings_2.68.1            
[125] tensor_1.5                    hms_1.1.3                    
[127] locfit_1.5-9.8                igraph_1.5.0                 
[129] spatstat.geom_3.2-2           ggsignif_0.6.4               
[131] reshape2_1.4.4                biomaRt_2.56.1               
[133] BiocVersion_3.17.1            XML_3.99-0.14                
[135] evaluate_0.21                 BiocManager_1.30.21          
[137] tweenr_2.0.2                  tzdb_0.4.0                   
[139] httpuv_1.6.11                 MatrixModels_0.5-2           
[141] RANN_2.6.1                    polyclip_1.10-4              
[143] future_1.33.0                 scattermore_1.2              
[145] broom_1.0.5                   xtable_1.8-4                 
[147] restfulr_0.0.15               rstatix_0.7.2                
[149] later_1.3.1                   class_7.3-22                 
[151] viridisLite_0.4.2             memoise_2.0.1                
[153] GenomicAlignments_1.36.0      cluster_2.1.4                
[155] workflowr_1.7.0               timechange_0.2.0             
[157] globals_0.16.2                here_1.0.1                   </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
