<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2023-06-22" />

<title>DHT_signal</title>

<script src="site_libs/header-attrs-2.23/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">20131906_HickeyT_JC_NormalBreast</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="description.html">Description</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    QC
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="qc_raw.html">Raw Data</a>
    </li>
    <li>
      <a href="qc_trimmed.html">Trimmed Data</a>
    </li>
    <li>
      <a href="qc_aligned.html">Aligned Data</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="dge_analysis.html">Differential Gene Expression</a>
    </li>
    <li>
      <a href="IDC_ILC.html">IDC/ILC classification</a>
    </li>
    <li>
      <a href="CellDecon_WEHI.html">Cell Type Deconvolution</a>
    </li>
    <li>
      <a href="sSNAPPY.html">sSNAPPY</a>
    </li>
    <li>
      <a href="E2_signal.html">E2 response</a>
    </li>
    <li>
      <a href="DHT_signal.html">DHT response</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">DHT_signal</h1>
<h4 class="date">2023-06-22</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-08-01
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>WenjunLiu_Thesis_Chapter4/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200930code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20200930)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200930code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200930)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongnoversioncontrol">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>Repository version:</strong> no
version control </a>
</p>
</div>
<div id="strongRepositoryversionstrongnoversioncontrol"
class="panel-collapse collapse">
<div class="panel-body">
<p>Tracking code development and connecting the code version to the
results is critical for reproducibility. To start using Git, open the
Terminal and type <code>git init</code> in your project directory.</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
This project is not being versioned with Git. To obtain the full
reproducibility benefits of using workflowr, please see
<code>?wflow_start</code>.
</p>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tidyverse)
library(yaml)
library(scales)
library(pander)
library(AnnotationHub)
library(ggraph)
library(igraph)
library(ensembldb)
library(cowplot)
library(magrittr)
library(sSNAPPY)
library(cqn)
library(DT)
library(ggfortify)
library(org.Hs.eg.db)
library(reactable)
library(corrplot)
library(htmltools)
library(RColorBrewer)
library(msigdbr)
library(ggfortify)
library(pheatmap)
library(ggforce)
library(ggnewscale)
library(concaveman)
library(ggpubr)
library(ggh4x)
library(colorspace)
library(pathview)
library(UpSetR)
source(here::here(&quot;analysis/smallFunctions/make_gsNetwork.R&quot;))</code></pre>
<pre class="r"><code>panderOptions(&quot;table.split.table&quot;, Inf)
panderOptions(&quot;big.mark&quot;, &quot;,&quot;)
theme_set(theme_bw())</code></pre>
<pre class="r"><code>config &lt;- here::here(&quot;config/config.yml&quot;) %&gt;%
  read_yaml()
suffix &lt;- paste0(config$tag)
sp &lt;- config$ref$species %&gt;%
  str_replace(&quot;(^[a-z])[a-z]*_([a-z]+)&quot;, &quot;\\1\\2&quot;) %&gt;%
  str_to_title()</code></pre>
<div id="setup" class="section level1">
<h1>Setup</h1>
<div id="gene-annotations" class="section level2">
<h2>Gene Annotations</h2>
<pre class="r"><code>ah &lt;- AnnotationHub() %&gt;%
  subset(rdataclass == &quot;EnsDb&quot;) %&gt;%
  subset(str_detect(description, as.character(config$ref$release))) %&gt;%
  subset(genome == config$ref$build)
stopifnot(length(ah) == 1)</code></pre>
<pre class="r"><code>ensDb &lt;- ah[[1]]
genesGR &lt;- read_rds(here::here(&quot;output/genesGR.rds&quot;))</code></pre>
</div>
<div id="sample-metadata-and-merged-counts" class="section level2">
<h2>Sample metadata and merged counts</h2>
<p>Sample metadata and merged counts were read in and filtered as in the
dge_analysis.Rmd. DGElists were formed for normal and tumor samples and
<code>cqn</code> was applied for biases introduced by systematic
artefacts.</p>
<pre class="r"><code>init_cellType &lt;- read_delim(here::here(&quot;config/sample_meta.txt&quot;), delim = &quot;\t&quot;)
init_cellType &lt;- init_cellType %&gt;%
    mutate(Stroma = ifelse(str_detect(Dominant_cell_type, regex(&quot;Stroma&quot;, ignore_case = T)), TRUE, FALSE), 
           Epithelial = ifelse(str_detect(Dominant_cell_type, regex(&quot;Epithelial&quot;, ignore_case = T)), TRUE, FALSE), 
           Ducts = ifelse(str_detect(Dominant_cell_type, regex(&quot;ducts&quot;, ignore_case = T)), TRUE, FALSE), 
           Fat = ifelse(str_detect(Dominant_cell_type, regex(&quot;fat&quot;, ignore_case = T)), TRUE, FALSE), 
           patient = str_replace(patient, &quot;TH&quot;, &quot;TH-&quot;)) %&gt;%
    pivot_longer(c(&quot;Stroma&quot;, &quot;Epithelial&quot;, &quot;Ducts&quot;, &quot;Fat&quot;),
                 names_to = &quot;cell_type&quot;, 
                 values_to = &quot;TF&quot;) %&gt;%
    mutate(cell_type = ifelse(TF, cell_type, NA)) %&gt;%
    dplyr::select(-c(&quot;TF&quot;, &quot;Dominant_cell_type&quot;)) %&gt;%
    .[!is.na(.$cell_type),] %&gt;%
    chop(&quot;cell_type&quot;) %&gt;%
    mutate(cell_type = vapply(.$cell_type, function(x){
        paste(x,collapse = &quot;;&quot;)
    }, character(1)))</code></pre>
<pre class="r"><code>samples &lt;- config$samples %&gt;%
  here::here() %&gt;%
  read_tsv() %&gt;%
    left_join(init_cellType) %&gt;%
  mutate(
    Filename = paste0(sample, &quot;.r_1&quot;), 
    condition = ifelse(Tumor, 
                       paste(&quot;Tumor&quot;, treat, sep = &quot;_&quot;), 
                       paste(&quot;Normal&quot;, treat, sep = &quot;_&quot;)), 
    patient = vapply(.$patient, function(x){str_split(x, &quot;-&quot;)[[1]][2]}, character(1)), 
    patient = ifelse(Tumor, 
                       paste(&quot;Tumor&quot;, patient, sep = &quot;-&quot;), 
                       paste(&quot;Normal&quot;,patient, sep = &quot;-&quot;)), 
    desc = paste(patient, treat, sep = &quot; &quot;)
  ) %&gt;%
  dplyr::select(-c(&quot;name&quot;, &quot;sample&quot;)) %&gt;%
  dplyr::rename(name = desc) %&gt;%
  mutate_if(
    function(x){length(unique(x)) &lt; length(x)},
    as.factor
  ) %&gt;%
  mutate(
    treat = relevel(treat, ref = &quot;Veh&quot;)
  )</code></pre>
<pre class="r"><code>mergedSamples &lt;- samples %&gt;%
  group_by(name, patient, treat, Tumor, cell_type, Tissue_type, Age, Diagnosis) %&gt;%
  tally()
tumor_sample &lt;- mergedSamples %&gt;%
  dplyr::filter(Tumor == TRUE) %&gt;%
    droplevels()</code></pre>
<pre class="r"><code>tumor_sample &lt;- tumor_sample %&gt;%
    dplyr::mutate(
        Diagnosis = ifelse(
            str_detect(Diagnosis, &quot;invasive&quot;), 
            &quot;ILC-like&quot;, &quot;IDC-like&quot;
        ),
        Diagnosis = ifelse(
            patient == &quot;Tumor-8&quot;, 
            &quot;IDC-like&quot;, 
            Diagnosis
        ), 
        Diagnosis = as.factor(Diagnosis)
    )</code></pre>
<pre class="r"><code>treat_cols &lt;- c(
  Veh = rgb(0.7, 0.7, 0.7),
  DHT = rgb(0.8, 0.2, 0.2),
  E2 = rgb(0.2, 0.2, 0.8),
  `E2+DHT` = rgb(1, 0.4, 1)
)
tumor_cols &lt;- hcl.colors(
  n = length(unique(samples$Tumor)), 
  palette = &quot;Zissou 1&quot;
  ) %&gt;%
  setNames(unique(samples$Tumor))
patient_cols &lt;- hcl.colors(
  n = length(levels(mergedSamples$patient)), 
  palette = &quot;Spectral&quot;
  ) %&gt;%
  setNames(levels(mergedSamples$patient))
diag_cols &lt;- readRDS(&quot;~/GSE800098/output/diag_cols.rds&quot;)</code></pre>
<pre class="r"><code>treat_shapes &lt;- c(
  Veh = 1,
  DHT = 19,
  E2 = 15,
  `E2+DHT` = 17
)</code></pre>
<pre class="r"><code># counts &lt;- here::here(&quot;data/aligned/counts/counts.out.gz&quot;) %&gt;%
#   gzfile() %&gt;%
#   read_tsv(comment = &quot;#&quot;) %&gt;%
#   dplyr::select(Geneid, ends_with(&quot;bam&quot;)) %&gt;%
#   rename_at(vars(ends_with(&quot;bam&quot;)), dirname) %&gt;%
#   rename_all(basename) %&gt;%
#   column_to_rownames(&quot;Geneid&quot;)</code></pre>
<pre class="r"><code># mergedCounts &lt;- counts %&gt;%
#   rownames_to_column(&quot;gene_id&quot;) %&gt;%
#   pivot_longer(
#     cols = -gene_id,
#     names_to = &quot;Filename&quot;,
#     values_to = &quot;counts&quot;
#   ) %&gt;%
#   left_join(samples, by = &quot;Filename&quot;) %&gt;%
#   group_by(
#     gene_id, name, patient, treat, Tumor) %&gt;%
#   summarise(counts = sum(counts), .groups = &quot;drop&quot;) %&gt;%
#   pivot_wider(
#     id_cols = gene_id,
#     values_from = counts,
#     names_from = name
#   ) %&gt;%
#   column_to_rownames(&quot;gene_id&quot;)
# saveRDS(mergedCounts, here::here(&quot;data/mergedCounts.rds&quot;))
mergedCounts &lt;- readRDS(here::here(&quot;data/mergedCounts.rds&quot;))</code></pre>
<pre><code>           used  (Mb) gc trigger  (Mb) limit (Mb) max used  (Mb)
Ncells 11045701 590.0   16754075 894.8         NA 16754075 894.8
Vcells 26669882 203.5   40741899 310.9     204800 33884902 258.6</code></pre>
<p>The filtered and normalised <code>DGEList</code> was loaded in.</p>
<pre class="r"><code>dge_tumor &lt;- readRDS(here::here(&quot;output/dge_tumor.rds&quot;))</code></pre>
<pre class="r"><code>cqNorma_tumor &lt;- with(
  dge_tumor,
  cqn(
    counts= counts,
    x = genes$gc_content,
    lengths = genes$ave_tx_len
  )
)
dge_tumor$offset &lt;- cqNorma_tumor$glm.offset
logCPM_tumor &lt;- cqNorma_tumor$y + cqNorma_tumor$offset</code></pre>
<pre class="r"><code>pcaPost_tumor &lt;- logCPM_tumor %&gt;%
  t() %&gt;%
  prcomp() </code></pre>
<pre class="r"><code>gsTopology &lt;- retrieve_topology(database = &quot;kegg&quot;, species = &quot;hsapiens&quot;)
logCPM_tumor2 &lt;- logCPM_tumor
rownames(logCPM_tumor2) &lt;- mapIds(ensDb,rownames(logCPM_tumor), &quot;ENTREZID&quot;, keytype = &quot;GENEID&quot;)
logCPM_tumor2 &lt;- logCPM_tumor2[!is.na(rownames(logCPM_tumor2)),]
ssFC_tumor &lt;- sSNAPPY::weight_ss_fc(logCPM_tumor2, tumor_sample, groupBy = &quot;patient&quot;, treatColumn = &quot;treat&quot;, sampleColumn = &quot;name&quot;)
ssFC_tumor &lt;- ssFC_tumor$weighted_logFC/ssFC_tumor$weight
weightedFC_tumor &lt;- sSNAPPY::weight_ss_fc(logCPM_tumor2, tumor_sample, groupBy = &quot;patient&quot;, treatColumn = &quot;treat&quot;, sampleColumn = &quot;name&quot;)
genePertScore_tumor &lt;- raw_gene_pert(weightedFC_tumor$weighted_logFC, gsTopology)</code></pre>
<pre class="r"><code>formatP &lt;- function(p, m = 0.0001){
out &lt;- rep(&quot;&quot;, length(p))
out[p &lt; m] &lt;- sprintf(&quot;%.2e&quot;, p[p&lt;m])
out[p &gt;= m] &lt;- sprintf(&quot;%.4f&quot;, p[p&gt;=m])
out
}</code></pre>
<p>Outputs of <code>sSNAPPY</code> workflow for tumor samples were read
in.</p>
<pre class="r"><code>treat_sig_tumor &lt;- readRDS(here::here(&quot;output/treat_sig_tumor.rds&quot;))
normalised_tumor &lt;- readRDS(here::here(&quot;output/normalisedScores_tumor.rds&quot;))</code></pre>
<pre class="r"><code>normalised_tumor_mat &lt;- normalised_tumor %&gt;%
    dplyr::select(gs_name, sample, robustZ) %&gt;%
    pivot_wider(
        names_from = &quot;sample&quot;, 
        values_from = &quot;robustZ&quot;
    ) %&gt;%
    column_to_rownames(&quot;gs_name&quot;)</code></pre>
<pre class="r"><code>new_diag_col &lt;- qualitative_hcl(2, palette = &quot;Dark 3&quot;) %&gt;%
    set_names(c(&quot;IDC-like&quot;, &quot;ILC-like&quot;))
new_treat_col &lt;- qualitative_hcl(4, palette = &quot;Dynamic&quot;) %&gt;%
    set_names(levels(tumor_sample$treat))</code></pre>
<pre class="r"><code>ssFC_tumor_df &lt;- ssFC_tumor %&gt;%
    as.data.frame() %&gt;%
    rownames_to_column(&quot;entrezid&quot;) %&gt;%
    mutate(
        entrezid = str_remove_all(entrezid, &quot;ENTREZID.&quot;)
    ) %&gt;%
    left_join(
        genesGR %&gt;%
            as.data.frame() %&gt;%
            unnest(entrezid) %&gt;%
            mutate(entrezid = as.character(entrezid)) %&gt;%
            dplyr::select(entrezid, gene_name)
    ) %&gt;%
    pivot_longer(
        cols = contains(&quot;Tumor&quot;), 
        names_to = &quot;name&quot;, 
        values_to = &quot;ssFC&quot;
    ) %&gt;%
    left_join(
        tumor_sample
    ) </code></pre>
<pre class="r"><code>load(system.file(&quot;extdata&quot;, &quot;entrez2name.rda&quot;, package = &quot;sSNAPPY&quot;))</code></pre>
<pre class="r"><code>logCPM_tumor_df &lt;- logCPM_tumor %&gt;%
    as.data.frame() %&gt;%
    rownames_to_column(&quot;gene_id&quot;) %&gt;%
    pivot_longer(
        cols = -&quot;gene_id&quot;, 
        names_to = &quot;name&quot;, 
        values_to = &quot;logCPM&quot;
    ) %&gt;%
    left_join(
        tumor_sample
    ) %&gt;%
    left_join(
        dge_tumor$genes %&gt;%
            dplyr::select(gene_id, gene_name)
    )</code></pre>
<pre class="r"><code>normalised_tumor_df &lt;- normalised_tumor %&gt;%
    dplyr::select(sample, gs_name, robustZ) %&gt;%
    pivot_wider(
        names_from = &quot;gs_name&quot;, 
        values_from = &quot;robustZ&quot;
    )</code></pre>
</div>
</div>
<div id="dht-signature" class="section level1">
<h1>DHT signature</h1>
<pre class="r"><code>treat_tumor_dht &lt;- treat_sig_tumor %&gt;%
    dplyr::filter(
        Comparison %in% c(&quot;DHT_infiltrating&quot;, &quot;DHT_invasive&quot;))
dht_ductal_gs &lt;- treat_sig_tumor %&gt;%
    dplyr::filter(
        Comparison ==  &quot;DHT_infiltrating&quot;, 
        FDR &lt; 0.05) 
dht_ductal_gs &lt;- pull(dht_ductal_gs, `t statistic`) %&gt;%
    set_names(dht_ductal_gs$gs_name)</code></pre>
<p>By applying the sSNAPPY workflow, 9 KEGG pathways were found to be
significantly perturbed in DHT-treated ductal-like PDEs in comparison to
the vehicle-treated PDEs. By performing community detection on the
network of pathways significantly perturbed by DHT in IDC-like PDEs, two
main communities were formed: 1) Cell growth and death; 2) Aging &amp;
Endocrine system &amp; Signaling molecules and interaction.</p>
<pre class="r"><code>set.seed(123)
treat_tumor_dht %&gt;%
    mutate(
        Status = ifelse(`t statistic` &lt; 0, &quot;Inhibited&quot;, &quot;Activated&quot;)) %&gt;%
    dplyr::filter(
        Comparison == &quot;DHT_infiltrating&quot;,
        FDR &lt; 0.05
    ) %&gt;%
    plot_community(
        gsTopology = gsTopology,
        colorBy = &quot;Status&quot;,
        communityMethod = &quot;label_prop&quot;
    ) +
    scale_color_manual(
        values = c(&quot;Inhibited&quot; = &quot;navyblue&quot;, &quot;Activated&quot; = &quot;darkred&quot;),
         name = &quot;Pathway Status&quot;
    ) +
    scale_fill_discrete_divergingx(palette = &quot;Fall&quot;) +
    theme(
        panel.border  = element_blank(),
        legend.key.size = unit(10, &quot;mm&quot;),
        legend.text = element_text(size = 9),
        legend.title = element_text(size = 11),
        legend.margin=margin(0,0,0,0)
    )</code></pre>
<p><img src="figure/DHT_signal.Rmd/unnamed-chunk-11-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ductal_DHT_sample &lt;- tumor_sample %&gt;%
    ungroup() %&gt;%
    dplyr::filter(
        treat == &quot;DHT&quot;, 
        Diagnosis == &quot;IDC-like&quot;
    ) %&gt;%
    pull(name)</code></pre>
<p>The potential drivers behind the perturbation of each pathway were
investigated.</p>
<div id="apoptosis" class="section level2">
<h2>Apoptosis</h2>
<pre class="r"><code>Apop_genePert &lt;-  genePertScore_tumor$`kegg.Apoptosis`
Apop_path &lt;- normalised_tumor_mat %&gt;%
    .[&quot;kegg.Apoptosis&quot;, colnames(.) %in% ductal_DHT_sample]
Apop_gene &lt;- Apop_genePert %&gt;%
        .[, colnames(.) %in% ductal_DHT_sample] %&gt;%
    .[apply(., 1, function(x){sum(sign(x)) == 3}),] %&gt;%
    apply(1, function(y){
            cor(
                y,
                as.vector(t(Apop_path)))}) %&gt;%
        .[!is.na(.)] %&gt;%
        .[. &gt; 0.6] %&gt;%
    names()</code></pre>
<pre class="r"><code>anno_df_Apop &lt;- normalised_tumor_df   %&gt;%
            mutate(
                `Z Range` = cut(
                    `kegg.Apoptosis`, breaks = seq(-1, 1, length.out = 6), include.lowest = TRUE
                )
            ) %&gt;%
            dplyr::select(sample, `Z Range`)
plot_gene_contribution(
    Apop_genePert %&gt;%
        .[rownames(.) %in% Apop_gene, colnames(.) %in% ductal_DHT_sample],
    filterBy = &quot;max.abs&quot;, 
    annotation_df = anno_df_Apop, 
    mapEntrezID = entrez2name, 
    topGene = 10,
    annotation_colors = list(
       
        `Z Range` = setNames(
            colorRampPalette(c(&quot;navyblue&quot;, &quot;white&quot;, &quot;darkred&quot;))(length(levels(anno_df_Apop$`Z Range`))),
            levels(anno_df_Apop$`Z Range`)
        )), 
    color = rev(divergex_hcl(100, palette = &quot;RdBu&quot;)), 
    breaks = seq(-0.00012, 0.00012, length.out = 100)
    ) </code></pre>
<p><img src="figure/DHT_signal.Rmd/unnamed-chunk-14-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Apop_gene_name &lt;- entrez2name %&gt;%
    dplyr::filter(
        entrezid %in% Apop_gene
    ) %&gt;%
    pull(mapTo)
apop_box &lt;-  ssFC_tumor_df %&gt;%
        dplyr::filter(
            gene_name %in% Apop_gene_name, 
            treat == &quot;DHT&quot;
        ) %&gt;%
        mutate(
            gene_name = factor(gene_name, levels = c(
                &quot;NFKBIA&quot;, &quot;TNF&quot;, &quot;ATF4&quot;,  &quot;EIF2AK3&quot;, &quot;FADD&quot;,
                &quot;GADD45G&quot; ,&quot;TNFRSF10A&quot;, &quot;CASP8&quot;, &quot;MAPK10&quot;
            ))) %&gt;%
        ggplot(
            aes(Diagnosis, ssFC, fill = Diagnosis)
        ) +
        geom_boxplot() +
        facet_wrap(~gene_name, 
                   scales = &quot;free_y&quot;) +
        scale_fill_manual(
            values = new_diag_col, 
            name = &quot;Predicted\nsubtype&quot;
        ) +
        geom_hline(yintercept = 0, color = &quot;red&quot;, linetype = &quot;dashed&quot;) +
        theme(
            axis.text.x = element_text(colour = new_diag_col)
        ) +
        labs(
            y = &quot;ssFC&quot;,
            x = &quot;&quot;)
apop_box</code></pre>
<p><img src="figure/DHT_signal.Rmd/unnamed-chunk-15-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="cell-cycle" class="section level2">
<h2>Cell Cycle</h2>
<pre class="r"><code>CC_genePert &lt;-  genePertScore_tumor$`kegg.Cell cycle`
CC_path &lt;- normalised_tumor_mat %&gt;%
    .[&quot;kegg.Cell cycle&quot;, colnames(.) %in% ductal_DHT_sample]
CC_gene &lt;- CC_genePert %&gt;%
        .[, colnames(.) %in% ductal_DHT_sample] %&gt;%
    .[apply(., 1, function(x){sum(sign(x)) == -3}),] %&gt;%
    apply(1, function(y){
            cor(
                y,
                as.vector(t(CC_path)))}) %&gt;%
        .[!is.na(.)] %&gt;%
        .[. &gt; 0.6] %&gt;%
    names()</code></pre>
<pre class="r"><code>anno_df_CC &lt;- normalised_tumor_df   %&gt;%
            mutate(
                `Z Range` = cut(
                    `kegg.Cell cycle`, breaks = seq(-1, 1, length.out = 6), include.lowest = TRUE
                )
            ) %&gt;%
            dplyr::select(sample, `Z Range`)
plot_gene_contribution(
    CC_genePert %&gt;%
        .[rownames(.) %in% CC_gene, colnames(.) %in% ductal_DHT_sample],
    annotation_df = anno_df_CC, 
    mapEntrezID = entrez2name, 
    topGene = length(CC_gene),
    annotation_colors = list(
       
        `Z Range` = setNames(
            colorRampPalette(c(&quot;navyblue&quot;, &quot;white&quot;, &quot;darkred&quot;))(length(levels(anno_df_CC$`Z Range`))),
            levels(anno_df_CC$`Z Range`)
        )), 
    color = rev(divergex_hcl(100, palette = &quot;RdBu&quot;)), 
    breaks = seq(-0.00085, 0.00085, length.out = 100)
    ) </code></pre>
<p><img src="figure/DHT_signal.Rmd/unnamed-chunk-17-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>CC_gene_name &lt;- entrez2name %&gt;%
    dplyr::filter(
        entrezid %in% CC_gene
    ) %&gt;%
    pull(mapTo)
CC_box &lt;- ssFC_tumor_df %&gt;%
    dplyr::filter(
        gene_name %in% c(&quot;PLK1&quot;, &quot;CDK1&quot;), 
        treat == &quot;DHT&quot;
        ) %&gt;%
    ggplot(
        aes(Diagnosis, ssFC, fill = Diagnosis)
    ) +
    geom_boxplot() +
    facet_wrap(~gene_name, 
               scales = &quot;free_y&quot;) +
    scale_fill_manual(
        values = new_diag_col, 
        name = &quot;Predicted\nsubtype&quot;
    ) +
    geom_hline(yintercept = 0, color = &quot;red&quot;, linetype = &quot;dashed&quot;) +
    theme(
        axis.text.x = element_text(colour = new_diag_col)
    ) +
    labs(
        y = &quot;ssFC&quot;,
        x = &quot;&quot;)
CC_box</code></pre>
<p><img src="figure/DHT_signal.Rmd/unnamed-chunk-18-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="p53-signalling-pathway" class="section level2">
<h2>p53 signalling pathway</h2>
<pre class="r"><code>p53_genePert &lt;-  genePertScore_tumor$`kegg.p53 signaling pathway`
p53_path &lt;- normalised_tumor_mat %&gt;%
    .[&quot;kegg.p53 signaling pathway&quot;, colnames(.) %in% ductal_DHT_sample]
p53_gene &lt;- p53_genePert %&gt;%
        .[, colnames(.) %in% ductal_DHT_sample] %&gt;%
    .[apply(., 1, function(x){sum(sign(x)) == -3}),] %&gt;%
    apply(1, function(y){
            cor(
                y,
                as.vector(t(p53_path)))}) %&gt;%
        .[!is.na(.)] %&gt;%
        .[. &gt; 0.6] %&gt;%
    names()</code></pre>
<pre class="r"><code># pathview(gene.data  =  ssFC_tumor %&gt;%
#              .[rownames(.) %in% p53_gene, colnames(.) %in% ductal_DHT_sample] %&gt;%
#              apply(., 1, median) %&gt;%
#              set_names(
#                  str_remove_all(
#                      names(.), &quot;ENTREZID:&quot; 
#                  )
#              ),
#          pathway.id = &quot;04115&quot;,
#          species    = &quot;hsa&quot;,
#          kegg.dir = here::here(&quot;figure/&quot;),
#          low = list(gene =&quot;skyblue&quot;, cpd = &quot;black&quot;),
#          mid = list(gene =&quot;grey&quot;, cpd = &quot;black&quot;),
#          high = list(gene =&quot;red&quot;, cpd = &quot;black&quot;),
#          )</code></pre>
<pre class="r"><code>anno_df_p53 &lt;- normalised_tumor_df   %&gt;%
            mutate(
                `Z Range` = cut(
                    `kegg.Cell cycle`, breaks = seq(-1, 1, length.out = 6), include.lowest = TRUE
                )
            ) %&gt;%
            dplyr::select(sample, `Z Range`)
plot_gene_contribution(
    p53_genePert %&gt;%
        .[rownames(.) %in% p53_gene, colnames(.) %in% ductal_DHT_sample],
    annotation_df = anno_df_p53, 
    mapEntrezID = entrez2name, 
    topGene = length(p53_gene),
    annotation_colors = list(
       
        `Z Range` = setNames(
            colorRampPalette(c(&quot;navyblue&quot;, &quot;white&quot;, &quot;darkred&quot;))(length(levels(anno_df_p53$`Z Range`))),
            levels(anno_df_p53$`Z Range`)
        )), 
    color = rev(divergex_hcl(100, palette = &quot;RdBu&quot;)), 
    breaks = seq(-0.00013, 0.00013, length.out = 100)
    ) </code></pre>
<p><img src="figure/DHT_signal.Rmd/unnamed-chunk-21-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p53_gene_name &lt;- entrez2name %&gt;%
    dplyr::filter(
        entrezid %in% p53_gene
    ) %&gt;%
    pull(mapTo)
p53_box &lt;- ssFC_tumor_df %&gt;%
    dplyr::filter(
        gene_name %in% p53_gene_name, 
        treat == &quot;DHT&quot;
        ) %&gt;%
    mutate(
        gene_name = factor(gene_name, levels = c(&quot;CDK1&quot;, &quot;CDK4&quot;, &quot;GTSE1&quot;, &quot;CCND2&quot;, &quot;CCNB1&quot;, &quot;RRM2&quot;, &quot;CYCS&quot;, &quot;CASP3&quot;))) %&gt;%
    ggplot(
        aes(Diagnosis, ssFC, fill = Diagnosis)
    ) +
    geom_boxplot() +
    facet_wrap(~gene_name, 
               scales = &quot;free_y&quot;) +
    scale_fill_manual(
        values = new_diag_col, 
        name = &quot;Predicted\nsubtype&quot;
    ) +
    geom_hline(yintercept = 0, color = &quot;red&quot;, linetype = &quot;dashed&quot;) +
    theme(
        axis.text.x = element_text(colour = new_diag_col)
    ) +
    labs(
        y = &quot;ssFC&quot;,
        x = &quot;&quot;)
p53_box</code></pre>
<p><img src="figure/DHT_signal.Rmd/unnamed-chunk-22-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="hedgehog-signaling-pathway" class="section level2">
<h2>Hedgehog signaling pathway</h2>
<pre class="r"><code>HH_genePert &lt;-  genePertScore_tumor$`kegg.Hedgehog signaling pathway`
HH_path &lt;- normalised_tumor_mat %&gt;%
    .[&quot;kegg.Hedgehog signaling pathway&quot;, colnames(.) %in% ductal_DHT_sample]
HH_gene &lt;- HH_genePert %&gt;%
        .[, colnames(.) %in% ductal_DHT_sample] %&gt;%
    .[apply(., 1, function(x){sum(sign(x)) == -3}),] %&gt;%
    apply(1, function(y){
            cor(
                y,
                as.vector(t(HH_path)))}) %&gt;%
        .[!is.na(.)] %&gt;%
        .[. &gt; 0.6] %&gt;%
    names()
HH_gene &lt;- intersect(HH_gene, rownames(weightedFC_tumor$weighted_logFC))</code></pre>
<pre class="r"><code>anno_df_HH &lt;- normalised_tumor_df   %&gt;%
            mutate(
                `Z Range` = cut(
                    `kegg.Hedgehog signaling pathway`, breaks = seq(-1, 1, length.out = 6), include.lowest = TRUE
                )
            ) %&gt;%
            dplyr::select(sample, `Z Range`)
plot_gene_contribution(
    HH_genePert %&gt;%
        .[rownames(.) %in% HH_gene, colnames(.) %in% ductal_DHT_sample],
    annotation_df = anno_df_HH, 
    mapEntrezID = entrez2name, 
    topGene = length(HH_gene),
    annotation_colors = list(
       
        `Z Range` = setNames(
            colorRampPalette(c(&quot;navyblue&quot;, &quot;white&quot;, &quot;darkred&quot;))(length(levels(anno_df_HH$`Z Range`))),
            levels(anno_df_HH$`Z Range`)
        )), 
    color = rev(divergex_hcl(100, palette = &quot;RdBu&quot;)), 
    breaks = seq(-7.800603e-05, 7.800603e-05, length.out = 100)
    ) </code></pre>
<p><img src="figure/DHT_signal.Rmd/unnamed-chunk-24-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>HH_gene_name &lt;- entrez2name %&gt;%
    dplyr::filter(
        entrezid %in% HH_gene
    ) %&gt;%
    pull(mapTo)
HH_box &lt;- ssFC_tumor_df %&gt;%
    dplyr::filter(
        gene_name %in% HH_gene_name, 
        treat == &quot;DHT&quot;
        ) %&gt;%
    ggplot(
        aes(Diagnosis, ssFC, fill = Diagnosis)
    ) +
    geom_boxplot() +
    facet_wrap(~gene_name, 
               scales = &quot;free_y&quot;) +
    scale_fill_manual(
        values = new_diag_col, 
        name = &quot;Predicted\nsubtype&quot;
    ) +
    geom_hline(yintercept = 0, color = &quot;red&quot;, linetype = &quot;dashed&quot;) +
    theme(
        axis.text.x = element_text(colour = new_diag_col)
    ) +
    labs(
        y = &quot;ssFC&quot;,
        x = &quot;&quot;)
HH_box</code></pre>
<p><img src="figure/DHT_signal.Rmd/unnamed-chunk-25-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="hif-signalling-pathway" class="section level2">
<h2>HIF signalling pathway</h2>
<pre class="r"><code>HIF_genePert &lt;-  genePertScore_tumor$`kegg.HIF-1 signaling pathway`
HIF_path &lt;- normalised_tumor_mat %&gt;%
    .[&quot;kegg.HIF-1 signaling pathway&quot;, colnames(.) %in% ductal_DHT_sample]
HIF_gene &lt;- HIF_genePert %&gt;%
        .[, colnames(.) %in% ductal_DHT_sample] %&gt;%
    .[apply(., 1, function(x){sum(sign(x)) == 3}),] %&gt;%
    apply(1, function(y){
            cor(
                y,
                as.vector(t(HIF_path)))}) %&gt;%
        .[!is.na(.)] %&gt;%
        .[. &gt; 0.6] %&gt;%
    names()
HIF_gene &lt;- intersect(HIF_gene, rownames(ssFC_tumor))</code></pre>
<pre class="r"><code>pathview(gene.data  =  ssFC_tumor %&gt;%
             .[rownames(.) %in% HIF_gene, colnames(.) %in% ductal_DHT_sample] %&gt;%
             apply(., 1, median) %&gt;%
             set_names(
                 str_remove_all(
                     names(.), &quot;ENTREZID:&quot; 
                 )
             ),
         pathway.id = &quot;04066&quot;,
         species    = &quot;hsa&quot;,
         kegg.dir = here::here(&quot;figure/&quot;),
         low = list(gene =&quot;skyblue&quot;, cpd = &quot;black&quot;),
         mid = list(gene =&quot;grey&quot;, cpd = &quot;black&quot;),
         high = list(gene =&quot;red&quot;, cpd = &quot;black&quot;),
         )</code></pre>
<pre class="r"><code>anno_df_HIF &lt;- normalised_tumor_df   %&gt;%
            mutate(
                `Z Range` = cut(
                    `kegg.HIF-1 signaling pathway`, breaks = seq(-1, 1, length.out = 6), include.lowest = TRUE
                )
            ) %&gt;%
            dplyr::select(sample, `Z Range`)
plot_gene_contribution(
    HIF_genePert %&gt;%
        .[rownames(.) %in% HIF_gene, colnames(.) %in% ductal_DHT_sample],
    filterBy = &quot;mean&quot;, 
    annotation_df = anno_df_HIF, 
    mapEntrezID = entrez2name, 
    topGene = 20,
    annotation_colors = list(
        `Z Range` = setNames(
            colorRampPalette(c(&quot;navyblue&quot;, &quot;white&quot;, &quot;darkred&quot;))(length(levels(anno_df_HIF$`Z Range`))),
            levels(anno_df_HIF$`Z Range`)
        )), 
    clustering_method = &quot;average&quot;,
    color = rev(divergex_hcl(100, palette = &quot;RdBu&quot;)), 
    cutree_rows = 2,
    breaks = seq(-0.00025, 0.00025, length.out = 100)
    ) </code></pre>
<p><img src="figure/DHT_signal.Rmd/unnamed-chunk-28-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>HIF_gene_name &lt;- entrez2name %&gt;%
    dplyr::filter(
        entrezid %in% HIF_gene
    ) %&gt;%
    pull(mapTo)
HIF_box &lt;- ssFC_tumor_df %&gt;%
    dplyr::filter(
        gene_name %in% c(&quot;HIF1A&quot;, &quot;SLC2A1&quot;, &quot;MTOR&quot; , &quot;RPS6&quot;), 
        treat == &quot;DHT&quot;
        ) %&gt;%
    ggplot(
        aes(Diagnosis, ssFC, fill = Diagnosis)
    ) +
    geom_boxplot() +
    facet_wrap(~gene_name, 
               scales = &quot;free_y&quot;) +
    scale_fill_manual(
        values = new_diag_col, 
        name = &quot;Predicted\nsubtype&quot;
    ) +
    geom_hline(yintercept = 0, color = &quot;red&quot;, linetype = &quot;dashed&quot;) +
    theme(
        axis.text.x = element_text(colour = new_diag_col)
    ) +
    labs(
        y = &quot;ssFC&quot;,
        x = &quot;&quot;)
HIF_box</code></pre>
<p><img src="figure/DHT_signal.Rmd/unnamed-chunk-29-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="lipid-and-atherosclerosis" class="section level2">
<h2>Lipid and atherosclerosis</h2>
<pre class="r"><code>gsTopology %&gt;%
    .[names(.) %in% c(
        &quot;kegg.Apoptosis&quot;, &quot;kegg.Lipid and atherosclerosis&quot;
    )] %&gt;%
    lapply(rownames) %&gt;%
    fromList() %&gt;%
    upset(
        sets = colnames(.),
        nintersects = NA
    )</code></pre>
<p><img src="figure/DHT_signal.Rmd/unnamed-chunk-30-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Lipid_genePert &lt;-  genePertScore_tumor$`kegg.Lipid and atherosclerosis`
Lipid_path &lt;- normalised_tumor_mat %&gt;%
    .[&quot;kegg.Lipid and atherosclerosis&quot;, colnames(.) %in% ductal_DHT_sample]
Lipid_gene &lt;- Lipid_genePert %&gt;%
        .[, colnames(.) %in% ductal_DHT_sample] %&gt;%
    .[apply(., 1, function(x){sum(sign(x)) == 3}),] %&gt;%
    apply(1, function(y){
            cor(
                y,
                as.vector(t(Lipid_path)))}) %&gt;%
        .[!is.na(.)] %&gt;%
        .[. &gt; 0.6] %&gt;%
    names()
Lipid_gene &lt;- intersect(Lipid_gene, rownames(weightedFC_tumor$weighted_logFC))</code></pre>
<pre class="r"><code>anno_df_Lipid &lt;- normalised_tumor_df   %&gt;%
            mutate(
                `Z Range` = cut(
                    `kegg.Lipid and atherosclerosis`, breaks = seq(-1, 1, length.out = 6), include.lowest = TRUE
                )
            ) %&gt;%
            dplyr::select(sample, `Z Range`)
plot_gene_contribution(
    Lipid_genePert %&gt;%
        .[rownames(.) %in% Lipid_gene, colnames(.) %in% ductal_DHT_sample],
    filterBy = &quot;max.abs&quot;, 
    annotation_df = anno_df_Lipid, 
    mapEntrezID = entrez2name, 
    topGene = 13,
    annotation_colors = list(
       
        `Z Range` = setNames(
            colorRampPalette(c(&quot;navyblue&quot;, &quot;white&quot;, &quot;darkred&quot;))(length(levels(anno_df_Lipid$`Z Range`))),
            levels(anno_df_Lipid$`Z Range`)
        )), 
    color = rev(divergex_hcl(100, palette = &quot;RdBu&quot;)), 
    cutree_rows = 2,
    breaks = seq(-0.00012, 0.00012, length.out = 100)
    ) </code></pre>
<p><img src="figure/DHT_signal.Rmd/unnamed-chunk-32-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Lipid_gene_name &lt;- entrez2name %&gt;%
    dplyr::filter(
        entrezid %in% Lipid_gene
    ) %&gt;%
    pull(mapTo)
Lipid_box &lt;-  ssFC_tumor_df %&gt;%
        dplyr::filter(
            gene_name %in% c(&quot;HSPA6&quot;, &quot;HSPA1A&quot;, &quot;HSPA1B&quot;, &quot;MAPK1&quot;, &quot;TIRAP&quot;),
            treat == &quot;DHT&quot;
        ) %&gt;%
        ggplot(
            aes(Diagnosis, ssFC, fill = Diagnosis)
        ) +
        geom_boxplot() +
        facet_wrap(~gene_name, 
                   scales = &quot;free_y&quot;) +
        scale_fill_manual(
            values = new_diag_col, 
            name = &quot;Predicted\nsubtype&quot;
        ) +
        geom_hline(yintercept = 0, color = &quot;red&quot;, linetype = &quot;dashed&quot;) +
        theme(
            axis.text.x = element_text(colour = new_diag_col)
        ) +
        labs(
            y = &quot;ssFC&quot;,
            x = &quot;&quot;)
Lipid_box</code></pre>
<p><img src="figure/DHT_signal.Rmd/unnamed-chunk-33-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="compare-to-known-ar-signature" class="section level1">
<h1>Compare to known AR signature</h1>
<div id="ar-signatures" class="section level2">
<h2>AR signatures</h2>
<pre class="r"><code>AR_sig &lt;- read.csv(here::here(&quot;data/AR_sig_down.csv&quot;)) %&gt;%
    mutate(AR_sig = &quot;Down&quot;) %&gt;%
    rbind(
        read.csv(here::here(&quot;data/AR_sig_up.csv&quot;)) %&gt;%
            mutate(AR_sig = &quot;Up&quot;)
    )
# AR_sig %&gt;%
#     dplyr::filter(
#         str_detect(gene_name, &quot;SLC&quot;)
#         # gene_name == &quot;SLC2A1&quot;
#         )</code></pre>
<p>A list of 142 AR signatures that were previously detected in
androgen-treated cell line and xenograft models were read in.</p>
<pre class="r"><code>kg &lt;- sapply(gsTopology, rownames) %&gt;%
    do.call(cbind, .) %&gt;%
    as.data.frame() %&gt;%
    pivot_longer(
        cols = everything(),
        names_to = &quot;gs_name&quot;,
        values_to = &quot;entrezid&quot;
    ) %&gt;%
    left_join(
        entrez2name %&gt;%
            dplyr::rename(gene_name = mapTo)
    ) %&gt;%
    left_join(genesGR %&gt;%
                  as.data.frame() %&gt;%
                  dplyr::select(gene_name, gene_id)) %&gt;%
    unique() %&gt;%
    dplyr::select(-entrezid)</code></pre>
<pre class="r"><code>ductal_DHT_fc &lt;- ssFC_tumor %&gt;%
    .[, colnames(.) %in% ductal_DHT_sample] %&gt;%
    apply(1, median)
ductal_DHT_fcDir &lt;- ifelse(ductal_DHT_fc &lt; 0, &quot;Down&quot;, &quot;Up&quot;) %&gt;%
    enframe(
        name = &quot;entrezid&quot;, 
        value = &quot;Direction&quot;
    ) %&gt;%
    left_join(
        entrez2name %&gt;%
            dplyr::rename(gene_name = mapTo)
    )
ar_sig_overlap &lt;- kg %&gt;%
    dplyr::filter(
        gs_name %in% names(dht_ductal_gs)) %&gt;%
    mutate(gs_name = str_remove_all(gs_name, &quot;kegg.&quot;)) %&gt;%
    left_join(
        AR_sig
    ) %&gt;%
    dplyr::select(-gene_id) %&gt;%
    drop_na() %&gt;%
    left_join(ductal_DHT_fcDir) %&gt;%
    unique() %&gt;%
    dplyr::filter(AR_sig == Direction) %&gt;%
    mutate_all(as.factor)</code></pre>
<p>Out of the 142 previously detected AR signatures, 7 of them were also
implicated in pathways significantly perturbed by DHT in ductal-like
tumours and had directional concordant changes.</p>
<p>The distribution of those 7 genes’ single-sample logFCs in all
DHT-treated PDEs are:</p>
<pre class="r"><code>col &lt;- AR_sig %&gt;%
    mutate(color = ifelse(
        AR_sig == &quot;Down&quot;,&quot;lightskyblue&quot;, &quot;pink&quot;
    )) %&gt;%
    dplyr::filter(gene_name %in% ar_sig_overlap$gene_name) %&gt;%
    .[match(levels(ar_sig_overlap$gene_name), .$gene_name),] %&gt;%
    pull(color)
strips &lt;- strip_themed(
    background_x = elem_list_rect(
        fill = adjustcolor(col, alpha.f = 0.5)
    )
)
ar_sig_box &lt;- ssFC_tumor_df %&gt;%
    dplyr::filter(
        gene_name %in% ar_sig_overlap$gene_name,
        treat == &quot;DHT&quot;
        ) %&gt;%
    mutate(Diagnosis = ifelse(
        str_detect(Diagnosis, &quot;invasive&quot;), &quot;ILC-like&quot;, &quot;IDC-like&quot;)) %&gt;%
    ggplot(
        aes(Diagnosis, ssFC, fill = Diagnosis)
    ) +
    geom_boxplot() +
    facet_wrap2(.~gene_name, 
               scales = &quot;free_y&quot;, 
               strip = strips) +
    scale_fill_manual(
        values = new_diag_col, 
        name = &quot;Predicted\nsubtype&quot;
    ) +
    scale_alpha_manual(name = &quot;Subtype&quot;, values = c(.3, 1), 
                       guide = FALSE) +
    geom_hline(yintercept = 0, color = &quot;red&quot;, linetype = &quot;dashed&quot;) +
    theme(
        strip.text = element_text(size = 18), 
        axis.text.x = element_text(size = 15, 
                                   colour = new_diag_col)
    ) +
    xlab(&quot;Predicted Subtype&quot;) 
ar_sig_box</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/DHT_signal.Rmd/unnamed-chunk-37-1.png" alt="*Single-sample logFCs of DHT-treated samples for genes that were involved in significant pathway perturbation among IDC-like tumours and were previoulsy report AR signatures. Background colors of the gene name strips indicate the directions of changes upon DHT treatment reported previously, where blue means AR-repressed.*" width="1152" />
<p class="caption">
<em>Single-sample logFCs of DHT-treated samples for genes that were
involved in significant pathway perturbation among IDC-like tumours and
were previoulsy report AR signatures. Background colors of the gene name
strips indicate the directions of changes upon DHT treatment reported
previously, where blue means AR-repressed.</em>
</p>
</div>
</div>
<div id="vasiliou-et-al.-ar-responses" class="section level2">
<h2>Vasiliou et al. AR responses</h2>
<pre class="r"><code>vasiliou_ar &lt;- read_csv(here::here(&quot;data/Vasiliou_etal_AR_sig.csv&quot;)) %&gt;%
    mutate(Visiliou_dir = ifelse(logFC &lt; 0, &quot;Down&quot;, &quot;Up&quot;))</code></pre>
<p>In the study by <a
href="https://clinicalproteomicsjournal.biomedcentral.com/articles/10.1186/s12014-022-09352-2#Sec2">Vasiliou
et al. (2022)</a>, ER+, PR+ and HER2 overexpression BT-474 cell line
were treated with DHT for 24 hrs before RNA-sequencing. 225
differentially expressed genes were found. The BT-474 cell line was also
derived from an IDC tumour.</p>
<pre class="r"><code>vasiliou_ar_overlap &lt;- vasiliou_ar %&gt;%
    dplyr::select(gene_name, Visiliou_dir) %&gt;%
    left_join(AR_sig) %&gt;%
    drop_na() %&gt;%
    mutate(Concordant = ifelse(
        Visiliou_dir == AR_sig, TRUE, FALSE
    ))</code></pre>
<p>15 of those DEGs were also in the list AR signatures. The directions
of changes reported in the two studies were all concordant. 3 of the
concordant AR responses were down-regulated genes and 12 were
up-regulated.</p>
<pre class="r"><code>vasiliou_pde_overlap  &lt;- kg %&gt;%
    dplyr::filter(
        gs_name %in% names(dht_ductal_gs)) %&gt;%
    mutate(gs_name = str_remove_all(gs_name, &quot;kegg.&quot;)) %&gt;%
    left_join(
        vasiliou_ar
    ) %&gt;%
    dplyr::select(gs_name, gene_name, Visiliou_dir) %&gt;%
    drop_na() %&gt;%
    left_join(ductal_DHT_fcDir) %&gt;%
    unique() %&gt;%
    dplyr::filter(Visiliou_dir == Direction) %&gt;%
    mutate_all(as.factor)</code></pre>
<p>Out of the 15 DEGs reported by Vasiliou et al., 6 of them were also
implicated in pathways significantly perturbed by DHT in ductal-like
tumours and had directional concordant changes.</p>
<p>The distribution of those 6 genes’ single-sample logFCs in all
DHT-treated PDEs are:</p>
<pre class="r"><code>col &lt;- vasiliou_ar %&gt;%
    mutate(color = ifelse(
        Visiliou_dir == &quot;Down&quot;,&quot;lightskyblue&quot;, &quot;pink&quot;
    )) %&gt;%
    dplyr::filter(gene_name %in% vasiliou_pde_overlap$gene_name) %&gt;%
    .[match(levels(vasiliou_pde_overlap$gene_name), .$gene_name),] %&gt;%
    pull(color)
strips &lt;- strip_themed(
    background_x = elem_list_rect(
        fill = adjustcolor(col, alpha.f = 0.5)
    )
)
vasiliou_box &lt;- ssFC_tumor_df %&gt;%
    dplyr::filter(
        gene_name %in% vasiliou_pde_overlap$gene_name,
        treat == &quot;DHT&quot;
        ) %&gt;%
    mutate(Diagnosis = ifelse(
        str_detect(Diagnosis, &quot;invasive&quot;), &quot;ILC-like&quot;, &quot;IDC-like&quot;)) %&gt;%
    ggplot(
        aes(Diagnosis, ssFC, fill = Diagnosis)
    ) +
    geom_boxplot() +
    facet_wrap2(.~gene_name, 
               scales = &quot;free_y&quot;, 
               strip = strips) +
    scale_fill_manual(
        values = new_diag_col, 
        name = &quot;Predicted\nsubtype&quot;
    ) +
    scale_alpha_manual(name = &quot;Subtype&quot;, values = c(.3, 1), 
                       guide = FALSE) +
    geom_hline(yintercept = 0, color = &quot;red&quot;, linetype = &quot;dashed&quot;) +
    theme(
        strip.text = element_text(size = 18), 
        axis.text.x = element_text(size = 15, 
                                   colour = new_diag_col)
    ) +
    xlab(&quot;Predicted Subtype&quot;) 
vasiliou_box</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/DHT_signal.Rmd/unnamed-chunk-41-1.png" alt="*Single-sample logFCs of DHT-treated samples for genes that were involved in significant pathway perturbation among IDC-like tumours and were previoulsy report to be AR-regulated genes by Vasiliou et al.. Background colors of the gene name strips indicate the directions of changes upon DHT treatment reported previously, where blue means AR-repressed.*" width="768" />
<p class="caption">
<em>Single-sample logFCs of DHT-treated samples for genes that were
involved in significant pathway perturbation among IDC-like tumours and
were previoulsy report to be AR-regulated genes by Vasiliou et al..
Background colors of the gene name strips indicate the directions of
changes upon DHT treatment reported previously, where blue means
AR-repressed.</em>
</p>
</div>
</div>
</div>
<div id="thesis-figures" class="section level1">
<h1>Thesis figures</h1>
<!-- ## network -->
<!-- ```{r} -->
<!-- set.seed(123) -->
<!-- network_dht <- treat_tumor_dht %>% -->
<!--     mutate( -->
<!--         Status = ifelse(`t statistic` < 0, "Inhibited", "Activated")) %>% -->
<!--     dplyr::filter( -->
<!--         Comparison == "DHT_infiltrating", -->
<!--         FDR < 0.05 -->
<!--     ) %>% -->
<!--     plot_community( -->
<!--         gsTopology = gsTopology, -->
<!--         colorBy = "Status", -->
<!--         communityMethod = "label_prop",  -->
<!--         lb_size = 4,  -->
<!--         label.fontsize = 14 -->
<!--     ) + -->
<!--     scale_color_manual( -->
<!--         values = c("Inhibited" = "navyblue", "Activated" = "darkred"), -->
<!--          name = "Pathway Status" -->
<!--     ) + -->
<!--     scale_fill_discrete_divergingx(palette = "Fall") + -->
<!--     guides(fill = FALSE) + -->
<!--     theme( -->
<!--         panel.border  = element_blank(), -->
<!--         legend.key.size = unit(10, "mm"), -->
<!--         legend.text = element_text(size = 12), -->
<!--         legend.title = element_text(size = 14), -->
<!--         legend.margin=margin(0,0,0,0) -->
<!--     ) -->
<!-- png( -->
<!--     "/Users/wenjunliu/PhD_thesis/Images/chapter_04/network_dht.png", -->
<!--     width = 250, height =200, units='mm', res = 300 -->
<!-- ) -->
<!-- network_dht -->
<!-- dev.off() -->
<!-- ``` -->
<!-- ## Apoptosis -->
<!-- ```{r} -->
<!-- p1 <- plot_gene_contribution( -->
<!--     Apop_genePert %>% -->
<!--         .[rownames(.) %in% Apop_gene, colnames(.) %in% ductal_DHT_sample], -->
<!--     filterBy = "max.abs", -->
<!--     annotation_df = anno_df_Apop, -->
<!--     mapEntrezID = entrez2name, -->
<!--     topGene = 10, -->
<!--     annotation_colors = list( -->
<!--         `Z Range` = setNames( -->
<!--             colorRampPalette(c("navyblue", "white", "darkred"))(length(levels(anno_df_Apop$`Z Range`))), -->
<!--             levels(anno_df_Apop$`Z Range`) -->
<!--         )), -->
<!--     color = rev(divergex_hcl(100, palette = "RdBu")), -->
<!--     breaks = seq(-0.00012, 0.00012, length.out = 100) -->
<!--     ) %>% -->
<!--     as.ggplot() -->
<!-- p2 <- apop_box -->
<!-- apop_DHT <- (( p1 | p2) + -->
<!--                   plot_layout(widths = c(0.2, 0.2))) + -->
<!--     plot_annotation(tag_levels = "A")  & -->
<!--     theme( -->
<!--         plot.tag = element_text(size = 22, face = "bold" ), -->
<!--         text = element_text(size = 14) -->
<!--     ) -->
<!-- png( -->
<!--     "/Users/wenjunliu/PhD_thesis/Images/chapter_04/apop_DHT.png", -->
<!--     width = 400, height =230, units='mm', res = 300 -->
<!-- ) -->
<!-- apop_DHT -->
<!-- dev.off() -->
<!-- ``` -->
<!-- ## Cell Cyle -->
<!-- ```{r} -->
<!-- # p1 <- plot_gene_contribution( -->
<!-- #     CC_genePert %>% -->
<!-- #         .[rownames(.) %in% CC_gene, colnames(.) %in% ductal_DHT_sample], -->
<!-- #     annotation_df = anno_df_CC, -->
<!-- #     mapEntrezID = entrez2name, -->
<!-- #     topGene = length(CC_gene), -->
<!-- #     annotation_colors = list( -->
<!-- #  -->
<!-- #         `Z Range` = setNames( -->
<!-- #             colorRampPalette(c("navyblue", "white", "darkred"))(length(levels(anno_df_CC$`Z Range`))), -->
<!-- #             levels(anno_df_CC$`Z Range`) -->
<!-- #         )), -->
<!-- #     color = rev(divergex_hcl(100, palette = "RdBu")), -->
<!-- #     breaks = seq(-0.00085, 0.00085, length.out = 100),  -->
<!-- #     cutree_rows = 2, -->
<!-- #     ) %>% -->
<!-- #     as.ggplot() -->
<!-- # p2 <- CC_box -->
<!-- # CC_DHT <- (( p1 | p2) + -->
<!-- #                   plot_layout(widths  = c(0.4, 0.2))) + -->
<!-- #     plot_annotation(tag_levels = "A")  & -->
<!-- #     theme( -->
<!-- #         plot.tag = element_text(size = 22, face = "bold" ), -->
<!-- #         text = element_text(size = 14) -->
<!-- #     ) -->
<!-- # png( -->
<!-- #     "/Users/wenjunliu/PhD_thesis/Images/chapter_04/CC_DHT.png", -->
<!-- #     width = 300, height =180, units='mm', res = 300 -->
<!-- # ) -->
<!-- # CC_DHT -->
<!-- # dev.off() -->
<!-- ``` -->
<!-- ## p53 signalling -->
<!-- ```{r} -->
<!-- # p1 <- plot_gene_contribution( -->
<!-- #     p53_genePert %>% -->
<!-- #         .[rownames(.) %in% p53_gene, colnames(.) %in% ductal_DHT_sample], -->
<!-- #     annotation_df = anno_df_p53,  -->
<!-- #     mapEntrezID = entrez2name,  -->
<!-- #     topGene = length(p53_gene), -->
<!-- #     annotation_colors = list( -->
<!-- #         -->
<!-- #         `Z Range` = setNames( -->
<!-- #             colorRampPalette(c("navyblue", "white", "darkred"))(length(levels(anno_df_p53$`Z Range`))), -->
<!-- #             levels(anno_df_p53$`Z Range`) -->
<!-- #         )),  -->
<!-- #     color = rev(divergex_hcl(100, palette = "RdBu")),  -->
<!-- #     breaks = seq(-0.00013, 0.00013, length.out = 100) -->
<!-- #     ) %>% -->
<!-- #     as.ggplot() -->
<!-- # p2 <- p53_box -->
<!-- # p53_DHT <- (( p1 | p2) + -->
<!-- #                   plot_layout(widths = c(0.2, 0.2))) + -->
<!-- #     plot_annotation(tag_levels = "A")  & -->
<!-- #     theme( -->
<!-- #         plot.tag = element_text(size = 22, face = "bold" ), -->
<!-- #         text = element_text(size = 14) -->
<!-- #     ) -->
<!-- # png( -->
<!-- #     "/Users/wenjunliu/PhD_thesis/Images/chapter_04/p53_DHT.png", -->
<!-- #     width = 320, height =150, units='mm', res = 300 -->
<!-- # ) -->
<!-- # p53_DHT -->
<!-- # dev.off() -->
<!-- ``` -->
<!-- ## Hedgehog signalling  -->
<!-- ```{r} -->
<!-- # p1 <- plot_gene_contribution( -->
<!-- #     HH_genePert %>% -->
<!-- #         .[rownames(.) %in% HH_gene, colnames(.) %in% ductal_DHT_sample], -->
<!-- #     annotation_df = anno_df_HH, -->
<!-- #     mapEntrezID = entrez2name, -->
<!-- #     topGene = length(HH_gene), -->
<!-- #     annotation_colors = list( -->
<!-- #  -->
<!-- #         `Z Range` = setNames( -->
<!-- #             colorRampPalette(c("navyblue", "white", "darkred"))(length(levels(anno_df_HH$`Z Range`))), -->
<!-- #             levels(anno_df_HH$`Z Range`) -->
<!-- #         )), -->
<!-- #     color = rev(divergex_hcl(100, palette = "RdBu")), -->
<!-- #     breaks = seq(-7.800603e-05, 7.800603e-05, length.out = 100) -->
<!-- #     ) %>% -->
<!-- #     as.ggplot() -->
<!-- # p2 <- HH_box -->
<!-- # HH_DHT <- (( p1 | p2) + -->
<!-- #                   plot_layout(widths = c(0.3, 0.2))) + -->
<!-- #     plot_annotation(tag_levels = "A")  & -->
<!-- #     theme( -->
<!-- #         plot.tag = element_text(size = 22, face = "bold" ), -->
<!-- #         text = element_text(size = 14) -->
<!-- #     ) -->
<!-- # png( -->
<!-- #     "/Users/wenjunliu/PhD_thesis/Images/chapter_04/HH_DHT.png", -->
<!-- #     width = 320, height =180, units='mm', res = 300 -->
<!-- # ) -->
<!-- # HH_DHT -->
<!-- # dev.off() -->
<!-- ``` -->
<!-- ```{r} -->
<!-- # p1 <- plot_gene_contribution( -->
<!-- #     HIF_genePert %>% -->
<!-- #         .[rownames(.) %in% HIF_gene, colnames(.) %in% ductal_DHT_sample], -->
<!-- #     filterBy = "mean", -->
<!-- #     annotation_df = anno_df_WNT, -->
<!-- #     mapEntrezID = entrez2name, -->
<!-- #     topGene = 20, -->
<!-- #     annotation_colors = list( -->
<!-- #         `Z Range` = setNames( -->
<!-- #             colorRampPalette(c("navyblue", "white", "darkred"))(length(levels(anno_df_HIF$`Z Range`))), -->
<!-- #             levels(anno_df_HIF$`Z Range`) -->
<!-- #         )), -->
<!-- #     clustering_method = "average", -->
<!-- #     color = rev(divergex_hcl(100, palette = "RdBu")), -->
<!-- #     cutree_rows = 2, -->
<!-- #     breaks = seq(-0.00025, 0.00025, length.out = 100) -->
<!-- #     ) %>% -->
<!-- #     as.ggplot() -->
<!-- # p2 <- HIF_box -->
<!-- # HIF_DHT <- (( p1 | p2) + -->
<!-- #                   plot_layout(widths = c(0.3, 0.2))) + -->
<!-- #     plot_annotation(tag_levels = "A")  & -->
<!-- #     theme( -->
<!-- #         plot.tag = element_text(size = 22, face = "bold" ), -->
<!-- #         text = element_text(size = 14) -->
<!-- #     ) -->
<!-- # png( -->
<!-- #     "/Users/wenjunliu/PhD_thesis/Images/chapter_04/HIF_DHT.png", -->
<!-- #     width = 320, height =180, units='mm', res = 300 -->
<!-- # ) -->
<!-- # HIF_DHT -->
<!-- # dev.off() -->
<!-- ``` -->
<!-- ## Lipid and atherosclerosis -->
<!-- ```{r} -->
<!-- # p1 <- plot_gene_contribution( -->
<!-- #     Lipid_genePert %>% -->
<!-- #         .[rownames(.) %in% Lipid_gene, colnames(.) %in% ductal_DHT_sample], -->
<!-- #     filterBy = "max.abs",  -->
<!-- #     annotation_df = anno_df_Lipid,  -->
<!-- #     mapEntrezID = entrez2name,  -->
<!-- #     topGene = 13, -->
<!-- #     annotation_colors = list( -->
<!-- #         -->
<!-- #         `Z Range` = setNames( -->
<!-- #             colorRampPalette(c("navyblue", "white", "darkred"))(length(levels(anno_df_Lipid$`Z Range`))), -->
<!-- #             levels(anno_df_Lipid$`Z Range`) -->
<!-- #         )),  -->
<!-- #     color = rev(divergex_hcl(100, palette = "RdBu")),  -->
<!-- #     cutree_rows = 2, -->
<!-- #     breaks = seq(-0.00012, 0.00012, length.out = 100) -->
<!-- #     ) %>% -->
<!-- #     as.ggplot() -->
<!-- # p2 <- Lipid_box -->
<!-- # Lipid_DHT <- (( p1 | p2) + -->
<!-- #                   plot_layout(widths = c(0.25, 0.2))) + -->
<!-- #     plot_annotation(tag_levels = "A")  & -->
<!-- #     theme( -->
<!-- #         plot.tag = element_text(size = 22, face = "bold" ), -->
<!-- #         text = element_text(size = 14) -->
<!-- #     ) -->
<!-- # png( -->
<!-- #     "/Users/wenjunliu/PhD_thesis/Images/chapter_04/Lipid_DHT.png", -->
<!-- #     width = 330, height =180, units='mm', res = 300 -->
<!-- # ) -->
<!-- # Lipid_DHT -->
<!-- # dev.off() -->
<!-- ``` -->
<!-- ```{r} -->
<!-- lipid_apop_overlap <- gsTopology %>% -->
<!--     .[names(.) %in% c( -->
<!--         "kegg.Apoptosis", "kegg.Lipid and atherosclerosis" -->
<!--     )] %>% -->
<!--     lapply(rownames) %>% -->
<!--     fromList() %>% -->
<!--     set_colnames(str_remove_all(colnames(.), "kegg.")) %>% -->
<!--     upset( -->
<!--         sets = colnames(.), -->
<!--         nintersects = NA,  -->
<!--         queries = list( -->
<!--           list(query = intersects, -->
<!--                params = list("Apoptosis",  "Lipid and atherosclerosis"), -->
<!--                color = "orange", -->
<!--                active = T)) -->
<!--     ) -->
<!-- # png( -->
<!-- #     "/Users/wenjunliu/PhD_thesis/Images/chapter_04/lipid_apop_overlap.png", -->
<!-- #     width = 200, height =100, units='mm', res = 300 -->
<!-- # ) -->
<!-- # lipid_apop_overlap -->
<!-- # dev.off() -->
<!-- ``` -->
<!-- ## Public AR signature -->
<!-- ```{r} -->
<!-- # public_AR <- (ar_sig_box / vasiliou_box -->
<!-- #           ) + -->
<!-- #     plot_layout(guides = "collect",  -->
<!-- #                 heights = c(3,2)) + -->
<!-- #     plot_annotation( -->
<!-- #         tag_levels = "A" -->
<!-- #     ) & -->
<!-- #     theme( -->
<!-- #         plot.tag = element_text(size = 35, face = "bold" ) -->
<!-- #     ) -->
<!-- # png( -->
<!-- #     "/Users/wenjunliu/PhD_thesis/Images/chapter_04/public_AR.png", -->
<!-- #     width = 300, height =370, units='mm', res = 300 -->
<!-- # ) -->
<!-- # public_AR -->
<!-- # dev.off() -->
<!-- ``` -->
<!-- ## Fig 5 -->
<!-- ```{r} -->
<!-- # png( -->
<!-- #     "/Users/wenjunliu/PhD_thesis/Images/chapter_04/cell_line_meta.png", -->
<!-- #     width = 700, height =650, units='mm', res = 300 -->
<!-- # ) -->
<!-- # hp <- DHT_4data %>% -->
<!-- #     mutate( -->
<!-- #         gs_name = str_remove_all(gs_name, "kegg."), -->
<!-- #         rankStat = -sign(`t statistic`) * log10(`P-value`)) %>% -->
<!-- #     dplyr::filter(gs_name %in% dht_ductual_final) %>% -->
<!-- #     dplyr::select(gs_name, Type, `t statistic`) %>% -->
<!-- #     mutate( -->
<!-- #         Type = ifelse(Type == "DHT_infiltrating", "Ductal-like PDE", Type), -->
<!-- #         Type = ifelse(Type == "DHT_invasive", "Lobular-like PDE", Type) -->
<!-- #         # gs_name = str_wrap(gs_name, 25) -->
<!-- #     ) %>% -->
<!-- #     pivot_wider( -->
<!-- #         names_from = "Type",  -->
<!-- #         values_from = "t statistic" -->
<!-- #     ) %>% -->
<!-- #     mutate(gs_name = str_wrap(gs_name, 25)) %>% -->
<!-- #     column_to_rownames("gs_name") %>% -->
<!-- #     #dplyr::select(-DHT_invasive) %>% -->
<!-- #     ComplexHeatmap::Heatmap( -->
<!-- #         col = rev(divergex_hcl(100, palette = "RdBu")),  -->
<!-- #         row_split = 2,  -->
<!-- #         column_split = 2, -->
<!-- #         width = unit(400, "mm"),  -->
<!-- #         height = unit(400, "mm"), -->
<!-- #         heatmap_legend_param = list( -->
<!-- #             title = "T-statistic", -->
<!-- #             direction = "horizontal", -->
<!-- #             labels_gp = gpar(fontsize = 35),  -->
<!-- #             grid_height = unit(2, "cm") -->
<!-- #         ),  -->
<!-- #         row_names_gp = gpar(fontsize = 40), -->
<!-- #         column_names_gp = gpar(fontface = c("plain","plain", "plain","plain", "bold","bold"), fontsize = 40), -->
<!-- #         row_title=NULL,  -->
<!-- #         column_title = NULL -->
<!-- #          -->
<!-- #     ) -->
<!-- # ComplexHeatmap::draw(hp, -->
<!-- #                      heatmap_legend_side = "top",  -->
<!-- #                      legend_title_gp = gpar(fontsize = 45, fontface = "bold")) -->
<!-- #  -->
<!-- # # dev.off() -->
<!-- ``` -->
<!-- ## Combination treatment -->
<!-- ```{r} -->
<!-- ductal_2treat %>% -->
<!--     dplyr::filter(group == "DHT & E2+DHT") %>% -->
<!--     mutate(gs_name = str_remove_all(gs_name, "kegg.")) %>% -->
<!--     ggplot( -->
<!--         aes(Comparison, gs_name, fill = `t statistic`) -->
<!--     ) + -->
<!--     geom_tile() + -->
<!--     scale_fill_distiller(palette = "RdBu",  -->
<!--                          limit = c(-5, 5), -->
<!--                          name = "T-statistic") + -->
<!--     labs( x= "",  y = "") -->
<!-- ``` -->
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21)
Platform: x86_64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.3.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Australia/Adelaide
tzcode source: internal

attached base packages:
[1] splines   stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] UpSetR_1.4.0            pathview_1.40.0         colorspace_2.1-0       
 [4] ggh4x_0.2.4             ggpubr_0.6.0            concaveman_1.1.0       
 [7] ggnewscale_0.4.9        ggforce_0.4.1           pheatmap_1.0.12        
[10] msigdbr_7.5.1           RColorBrewer_1.1-3      htmltools_0.5.5        
[13] corrplot_0.92           reactable_0.4.4         org.Hs.eg.db_3.17.0    
[16] ggfortify_0.4.16        DT_0.28                 cqn_1.46.0             
[19] quantreg_5.95           SparseM_1.81            preprocessCore_1.62.1  
[22] nor1mix_1.3-0           mclust_6.0.0            sSNAPPY_1.5.1          
[25] magrittr_2.0.3          cowplot_1.1.1           ensembldb_2.24.0       
[28] AnnotationFilter_1.24.0 GenomicFeatures_1.52.1  AnnotationDbi_1.62.2   
[31] Biobase_2.60.0          GenomicRanges_1.52.0    GenomeInfoDb_1.36.1    
[34] IRanges_2.34.1          S4Vectors_0.38.1        igraph_1.5.0           
[37] ggraph_2.1.0            AnnotationHub_3.8.0     BiocFileCache_2.8.0    
[40] dbplyr_2.3.3            BiocGenerics_0.46.0     pander_0.6.5           
[43] scales_1.2.1            yaml_2.3.7              lubridate_1.9.2        
[46] forcats_1.0.0           stringr_1.5.0           dplyr_1.1.2            
[49] purrr_1.0.1             readr_2.1.4             tidyr_1.3.0            
[52] tibble_3.2.1            ggplot2_3.4.2           tidyverse_2.0.0        

loaded via a namespace (and not attached):
  [1] later_1.3.1                   BiocIO_1.10.0                
  [3] bitops_1.0-7                  filelock_1.0.2               
  [5] polyclip_1.10-4               graph_1.78.0                 
  [7] XML_3.99-0.14                 lifecycle_1.0.3              
  [9] rstatix_0.7.2                 edgeR_3.42.4                 
 [11] rprojroot_2.0.3               vroom_1.6.3                  
 [13] lattice_0.21-8                MASS_7.3-60                  
 [15] backports_1.4.1               limma_3.56.2                 
 [17] sass_0.4.6                    rmarkdown_2.23               
 [19] jquerylib_0.1.4               httpuv_1.6.11                
 [21] DBI_1.1.3                     abind_1.4-5                  
 [23] zlibbioc_1.46.0               RCurl_1.98-1.12              
 [25] tweenr_2.0.2                  rappdirs_0.3.3               
 [27] git2r_0.32.0                  GenomeInfoDbData_1.2.10      
 [29] ggrepel_0.9.3                 MatrixModels_0.5-2           
 [31] codetools_0.2-19              DelayedArray_0.26.6          
 [33] xml2_1.3.5                    tidyselect_1.2.0             
 [35] farver_2.1.1                  viridis_0.6.4                
 [37] matrixStats_1.0.0             GenomicAlignments_1.36.0     
 [39] jsonlite_1.8.7                ellipsis_0.3.2               
 [41] tidygraph_1.2.3               survival_3.5-5               
 [43] systemfonts_1.0.4             tools_4.3.0                  
 [45] progress_1.2.2                Rcpp_1.0.11                  
 [47] glue_1.6.2                    gridExtra_2.3                
 [49] here_1.0.1                    xfun_0.39                    
 [51] MatrixGenerics_1.12.2         withr_2.5.0                  
 [53] BiocManager_1.30.21           fastmap_1.1.1                
 [55] fansi_1.0.4                   digest_0.6.33                
 [57] timechange_0.2.0              R6_2.5.1                     
 [59] mime_0.12                     gtools_3.9.4                 
 [61] biomaRt_2.56.1                RSQLite_2.3.1                
 [63] utf8_1.2.3                    generics_0.1.3               
 [65] rtracklayer_1.60.0            prettyunits_1.1.1            
 [67] graphlayouts_1.0.0            httr_1.4.6                   
 [69] htmlwidgets_1.6.2             S4Arrays_1.0.4               
 [71] graphite_1.46.0               pkgconfig_2.0.3              
 [73] gtable_0.3.3                  blob_1.2.4                   
 [75] workflowr_1.7.0               XVector_0.40.0               
 [77] carData_3.0-5                 ProtGenerics_1.32.0          
 [79] png_0.1-8                     knitr_1.43                   
 [81] rstudioapi_0.15.0             tzdb_0.4.0                   
 [83] reshape2_1.4.4                rjson_0.2.21                 
 [85] curl_5.0.1                    cachem_1.0.8                 
 [87] BiocVersion_3.17.1            parallel_4.3.0               
 [89] restfulr_0.0.15               pillar_1.9.0                 
 [91] grid_4.3.0                    vctrs_0.6.3                  
 [93] promises_1.2.0.1              car_3.1-2                    
 [95] xtable_1.8-4                  Rgraphviz_2.44.0             
 [97] KEGGgraph_1.60.0              evaluate_0.21                
 [99] cli_3.6.1                     locfit_1.5-9.8               
[101] compiler_4.3.0                Rsamtools_2.16.0             
[103] rlang_1.1.1                   crayon_1.5.2                 
[105] ggsignif_0.6.4                labeling_0.4.2               
[107] plyr_1.8.8                    fs_1.6.2                     
[109] stringi_1.7.12                viridisLite_0.4.2            
[111] BiocParallel_1.34.2           babelgene_22.9               
[113] munsell_0.5.0                 Biostrings_2.68.1            
[115] lazyeval_0.2.2                Matrix_1.6-0                 
[117] hms_1.1.3                     bit64_4.0.5                  
[119] KEGGREST_1.40.0               shiny_1.7.4.1                
[121] highr_0.10                    SummarizedExperiment_1.30.2  
[123] interactiveDisplayBase_1.38.0 broom_1.0.5                  
[125] memoise_2.0.1                 bslib_0.5.0                  
[127] bit_4.0.5                    </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
