<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Wenjun Liu Dame Roma Mitchell Cancer Research Laboratories Adelaide Medical School University of Adelaide" />


<title>IDC_ILC</title>

<script src="site_libs/header-attrs-2.23/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.28/datatables.js"></script>
<link href="site_libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script>
<link href="site_libs/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet" />
<script src="site_libs/nouislider-7.0.10/jquery.nouislider.min.js"></script>
<link href="site_libs/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet" />
<script src="site_libs/selectize-0.12.0/selectize.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">20131906_HickeyT_JC_NormalBreast</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="description.html">Description</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    QC
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="qc_raw.html">Raw Data</a>
    </li>
    <li>
      <a href="qc_trimmed.html">Trimmed Data</a>
    </li>
    <li>
      <a href="qc_aligned.html">Aligned Data</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="dge_analysis.html">Differential Gene Expression</a>
    </li>
    <li>
      <a href="IDC_ILC.html">IDC/ILC classification</a>
    </li>
    <li>
      <a href="CellDecon_WEHI.html">Cell Type Deconvolution</a>
    </li>
    <li>
      <a href="sSNAPPY.html">sSNAPPY</a>
    </li>
    <li>
      <a href="E2_signal.html">E2 response</a>
    </li>
    <li>
      <a href="DHT_signal.html">DHT response</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">IDC_ILC</h1>
<h4 class="author">Wenjun Liu<br>Dame Roma Mitchell Cancer Research
Laboratories<br>Adelaide Medical School<br>University of Adelaide</h4>
<h4 class="date">19 September, 2023</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-09-19
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>WenjunLiu_Thesis_Chapter4/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200930code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20200930)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200930code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200930)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomWenjunLiuWenjunLiuThesisChapter4tree772af9f5f126a8b55968d3e50ceb2edda0c03a1dtargetblank772af9fa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/Wenjun-Liu/WenjunLiu_Thesis_Chapter4/tree/772af9f5f126a8b55968d3e50ceb2edda0c03a1d" target="_blank">772af9f</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomWenjunLiuWenjunLiuThesisChapter4tree772af9f5f126a8b55968d3e50ceb2edda0c03a1dtargetblank772af9fa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/Wenjun-Liu/WenjunLiu_Thesis_Chapter4/tree/772af9f5f126a8b55968d3e50ceb2edda0c03a1d" target="_blank">772af9f</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store
    Ignored:    data/aligned/
    Ignored:    data/knott_scRNA.rds
    Ignored:    data/knott_scRNA.rds.zip
    Ignored:    data/raw/
    Ignored:    data/scRNA_ref/
    Ignored:    data/trimmed/
    Ignored:    output/permutedscore_tumor.rds
    Ignored:    output/tcga_breast.rds

Untracked files:
    Untracked:  .DS_Store
    Untracked:  data/CellLine_sSNAPPY.rds
    Untracked:  data/EMT_marker.txt
    Untracked:  hsa04066.pathview.png

Unstaged changes:
    Modified:   analysis/DHT_signal.Rmd
    Modified:   analysis/IDC_ILC.Rmd
    Modified:   analysis/hsa04066.pathview.png

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/IDC_ILC.Rmd</code>) and HTML
(<code>docs/IDC_ILC.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Wenjun-Liu/WenjunLiu_Thesis_Chapter4/blob/772af9f5f126a8b55968d3e50ceb2edda0c03a1d/analysis/IDC_ILC.Rmd" target="_blank">772af9f</a>
</td>
<td>
Wenjun-Liu
</td>
<td>
2023-08-02
</td>
<td>
fixed the x-axis order in HIF-DHT hp
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Wenjun-Liu/WenjunLiu_Thesis_Chapter4/blob/29c6ff2cc27e1163cf921f298b80af1445747833/analysis/IDC_ILC.Rmd" target="_blank">29c6ff2</a>
</td>
<td>
Wenjun-Liu
</td>
<td>
2023-08-02
</td>
<td>
fixxed up AUROC plot
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Wenjun-Liu/WenjunLiu_Thesis_Chapter4/blob/4b1209781ee751444ce0c6a132ac20b5edfee771/analysis/IDC_ILC.Rmd" target="_blank">4b12097</a>
</td>
<td>
Wenjun-Liu
</td>
<td>
2023-08-02
</td>
<td>
tidy up all Rmd for thesis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/Wenjun-Liu/WenjunLiu_Thesis_Chapter4/4b1209781ee751444ce0c6a132ac20b5edfee771/docs/IDC_ILC.html" target="_blank">4b12097</a>
</td>
<td>
Wenjun-Liu
</td>
<td>
2023-08-02
</td>
<td>
tidy up all Rmd for thesis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tidyverse)
library(yaml)
library(scales)
library(pander)
library(AnnotationHub)
library(ggraph)
library(igraph)
library(ensembldb)
library(cowplot)
library(magrittr)
library(sSNAPPY)
library(cqn)
library(DT)
library(multcomp)
library(rvest)
library(ggfortify)
library(xml2)
library(org.Hs.eg.db)
library(lme4)
library(colorspace)
library(lmerTest)
library(reactable)
library(corrplot)
library(scatterpie)
library(htmltools)
library(RColorBrewer)
library(msigdbr)
library(UpSetR)
library(glue)
library(ggfortify)
library(pheatmap)
library(parallel)
library(ggplotify)
library(edgeR)
library(plotly)
library(ggforce)
library(ggnewscale)
library(concaveman)
library(WGCNA)
library(ggforce)
library(ggrepel)
# library(ISOpureR)
library(mixOmics)
library(ExperimentHub)
library(singscore)
library(grid)
library(goseq)
source(here::here(&quot;analysis/smallFunctions/make_gsNetwork.R&quot;))</code></pre>
<pre class="r"><code>panderOptions(&quot;table.split.table&quot;, Inf)
panderOptions(&quot;big.mark&quot;, &quot;,&quot;)
theme_set(theme_bw())</code></pre>
<pre class="r"><code>config &lt;- here::here(&quot;config/config.yml&quot;) %&gt;%
  read_yaml()
suffix &lt;- paste0(config$tag)
sp &lt;- config$ref$species %&gt;%
  str_replace(&quot;(^[a-z])[a-z]*_([a-z]+)&quot;, &quot;\\1\\2&quot;) %&gt;%
  str_to_title()</code></pre>
<div id="setup" class="section level1">
<h1>Setup</h1>
<div id="gene-annotations" class="section level2">
<h2>Gene Annotations</h2>
<pre class="r"><code>ah &lt;- AnnotationHub() %&gt;%
  subset(rdataclass == &quot;EnsDb&quot;) %&gt;%
  subset(str_detect(description, as.character(config$ref$release))) %&gt;%
  subset(genome == config$ref$build)
stopifnot(length(ah) == 1)</code></pre>
<pre class="r"><code>ensDb &lt;- ah[[1]]
genesGR &lt;- genes(ensDb)
transGR &lt;- transcripts(ensDb)</code></pre>
<pre class="r"><code>mcols(transGR) &lt;- mcols(transGR) %&gt;%
  cbind(
    transcriptLengths(ensDb)[rownames(.), c(&quot;nexon&quot;, &quot;tx_len&quot;)]
  )</code></pre>
<pre class="r"><code>mcols(genesGR) &lt;- mcols(genesGR) %&gt;%
  as.data.frame() %&gt;%
  dplyr::select(
    gene_id, gene_name, gene_biotype, entrezid
  ) %&gt;%
  left_join(
    mcols(transGR) %&gt;%
      as.data.frame() %&gt;%
      mutate(
        tx_support_level = case_when(
          is.na(tx_support_level) ~ 1L, 
          TRUE ~ tx_support_level
        )
      ) %&gt;%
      group_by(gene_id) %&gt;%
      summarise(
        n_tx = n(),
        longest_tx = max(tx_len),
        ave_tx_len = mean(tx_len),
        gc_content = sum(tx_len*gc_content) / sum(tx_len)
      ) %&gt;%
      mutate(
        bin_length = cut(
          x = ave_tx_len,
          labels = seq_len(10),
          breaks = quantile(ave_tx_len, probs = seq(0, 1, length.out = 11)),
          include.lowest = TRUE
        ),
        bin_gc = cut(
          x = gc_content,
          labels = seq_len(10),
          breaks = quantile(gc_content, probs = seq(0, 1, length.out = 11)),
          include.lowest = TRUE
        ),
        bin = paste(bin_gc, bin_length, sep = &quot;_&quot;)
      ),
    by = &quot;gene_id&quot;
  ) %&gt;%
  set_rownames(.$gene_id) %&gt;%
  as(&quot;DataFrame&quot;)</code></pre>
</div>
<div id="sample-metadata-and-merged-counts" class="section level2">
<h2>Sample metadata and merged counts</h2>
<p>Sample metadata and merged counts for tumour samples were read in and
filtered as in the <a href="dge_analysis.html">dge analysis</a>. A
<code>DGElist</code> was formed and <code>cqn</code> was applied for
biases introduced by systematic artefacts.</p>
<pre class="r"><code>init_cellType &lt;- read_delim(here::here(&quot;config/sample_meta.txt&quot;), delim = &quot;\t&quot;)
init_cellType &lt;- init_cellType %&gt;%
    mutate(Stroma = ifelse(str_detect(Dominant_cell_type, regex(&quot;Stroma&quot;, ignore_case = T)), TRUE, FALSE), 
           Epithelial = ifelse(str_detect(Dominant_cell_type, regex(&quot;Epithelial&quot;, ignore_case = T)), TRUE, FALSE), 
           Ducts = ifelse(str_detect(Dominant_cell_type, regex(&quot;ducts&quot;, ignore_case = T)), TRUE, FALSE), 
           Fat = ifelse(str_detect(Dominant_cell_type, regex(&quot;fat&quot;, ignore_case = T)), TRUE, FALSE), 
           patient = str_replace(patient, &quot;TH&quot;, &quot;TH-&quot;)) %&gt;%
    pivot_longer(c(&quot;Stroma&quot;, &quot;Epithelial&quot;, &quot;Ducts&quot;, &quot;Fat&quot;),
                 names_to = &quot;cell_type&quot;, 
                 values_to = &quot;TF&quot;) %&gt;%
    mutate(cell_type = ifelse(TF, cell_type, NA)) %&gt;%
    dplyr::select(-c(&quot;TF&quot;, &quot;Dominant_cell_type&quot;)) %&gt;%
    .[!is.na(.$cell_type),] %&gt;%
    chop(&quot;cell_type&quot;) %&gt;%
    mutate(cell_type = vapply(.$cell_type, function(x){
        paste(x,collapse = &quot;;&quot;)
    }, character(1)))</code></pre>
<pre class="r"><code>samples &lt;- config$samples %&gt;%
  here::here() %&gt;%
  read_tsv() %&gt;%
    left_join(init_cellType) %&gt;%
  mutate(
    Filename = paste0(sample, &quot;.r_1&quot;), 
    condition = ifelse(Tumor, 
                       paste(&quot;Tumor&quot;, treat, sep = &quot;_&quot;), 
                       paste(&quot;Normal&quot;, treat, sep = &quot;_&quot;)), 
    patient = vapply(.$patient, function(x){str_split(x, &quot;-&quot;)[[1]][2]}, character(1)), 
    patient = ifelse(Tumor, 
                       paste(&quot;Tumor&quot;, patient, sep = &quot;-&quot;), 
                       paste(&quot;Normal&quot;,patient, sep = &quot;-&quot;)), 
    desc = paste(patient, treat, sep = &quot; &quot;)
  ) %&gt;%
  dplyr::select(-c(&quot;name&quot;, &quot;sample&quot;)) %&gt;%
  dplyr::rename(name = desc) %&gt;%
  mutate_if(
    function(x){length(unique(x)) &lt; length(x)},
    as.factor
  ) %&gt;%
  mutate(
    treat = relevel(treat, ref = &quot;Veh&quot;)
  )</code></pre>
<pre class="r"><code>mergedSamples &lt;- samples %&gt;%
  group_by(name, patient, treat, Tumor, cell_type, Tissue_type, Age, Diagnosis) %&gt;%
  tally()
tumor_sample &lt;- mergedSamples %&gt;%
  dplyr::filter(Tumor == TRUE) %&gt;%
    droplevels()
diag_cols &lt;- readRDS(&quot;~/GSE800098/output/diag_cols.rds&quot;)</code></pre>
<pre class="r"><code>treat_cols &lt;- c(
  Veh = rgb(0.7, 0.7, 0.7),
  DHT = rgb(0.8, 0.2, 0.2),
  E2 = rgb(0.2, 0.2, 0.8),
  `E2+DHT` = rgb(1, 0.4, 1)
)
tumor_cols &lt;- hcl.colors(
  n = length(unique(samples$Tumor)), 
  palette = &quot;Zissou 1&quot;
  ) %&gt;%
  setNames(unique(samples$Tumor))
patient_cols &lt;- hcl.colors(
  n = length(levels(tumor_sample$patient)), 
  palette = &quot;Spectral&quot;
  ) %&gt;%
  setNames(levels(tumor_sample$patient))</code></pre>
<pre class="r"><code>treat_shapes &lt;- c(
  Veh = 1,
  DHT = 19,
  E2 = 15,
  `E2+DHT` = 17
)</code></pre>
<pre class="r"><code># counts &lt;- here::here(&quot;data/aligned/counts/counts.out.gz&quot;) %&gt;%
#   gzfile() %&gt;%
#   read_tsv(comment = &quot;#&quot;) %&gt;%
#   dplyr::select(Geneid, ends_with(&quot;bam&quot;)) %&gt;%
#   rename_at(vars(ends_with(&quot;bam&quot;)), dirname) %&gt;%
#   rename_all(basename) %&gt;%
#   column_to_rownames(&quot;Geneid&quot;)</code></pre>
<pre class="r"><code># mergedCounts &lt;- counts %&gt;%
#   rownames_to_column(&quot;gene_id&quot;) %&gt;%
#   pivot_longer(
#     cols = -gene_id,
#     names_to = &quot;Filename&quot;,
#     values_to = &quot;counts&quot;
#   ) %&gt;%
#   left_join(samples, by = &quot;Filename&quot;) %&gt;%
#   group_by(
#     gene_id, name, patient, treat, Tumor) %&gt;%
#   summarise(counts = sum(counts), .groups = &quot;drop&quot;) %&gt;%
#   pivot_wider(
#     id_cols = gene_id,
#     values_from = counts,
#     names_from = name
#   ) %&gt;%
#   column_to_rownames(&quot;gene_id&quot;)
# saveRDS(mergedCounts, here::here(&quot;data/mergedCounts.rds&quot;))
mergedCounts &lt;- readRDS(here::here(&quot;data/mergedCounts.rds&quot;))</code></pre>
<pre class="r"><code>rm(samples)
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger   (Mb) limit (Mb) max used   (Mb)
Ncells 12022827 642.1   18782531 1003.1         NA 18782531 1003.1
Vcells 31693152 241.8   54016728  412.2     204800 53994622  412.0</code></pre>
<pre class="r"><code>minCPM &lt;- 1.5
minSamples_tumor &lt;- 8
genes2Keep_tumor &lt;- mergedCounts[,colnames(mergedCounts) %in% tumor_sample$name] %&gt;%
  edgeR::cpm() %&gt;%
  is_greater_than(minCPM) %&gt;%
  rowSums() %&gt;%
  is_weakly_greater_than(minSamples_tumor)</code></pre>
<pre class="r"><code>dge_tumor &lt;- readRDS(here::here(&quot;output/dge_tumor.rds&quot;))</code></pre>
<pre class="r"><code>cqNorma_tumor &lt;- with(
  dge_tumor,
  cqn(
    counts= counts,
    x = genes$gc_content,
    lengths = genes$ave_tx_len
  )
)
dge_tumor$offset &lt;- cqNorma_tumor$glm.offset
logCPM_tumor &lt;- cqNorma_tumor$y + cqNorma_tumor$offset</code></pre>
<pre class="r"><code>pcaPost_tumor &lt;- logCPM_tumor %&gt;%
  t() %&gt;%
  prcomp() </code></pre>
<pre class="r"><code>tpm_func &lt;- function(counts,len) {
    x &lt;- counts/len
    return(t(t(x)*1e6/colSums(x)))
}
GeneLength &lt;- genesGR %&gt;%
    as_tibble() %&gt;%
    dplyr::select(gene_name, ave_tx_len) 
tumor_count &lt;- mergedCounts[,colnames(mergedCounts) %in% tumor_sample$name]
tumor_count &lt;-tumor_count %&gt;%
    rownames_to_column(&quot;gene_id&quot;) %&gt;%
    left_join(
        genesGR %&gt;%
            as.data.frame() %&gt;%
            dplyr::select(gene_id, gene_name)
    ) %&gt;%
    .[!duplicated(.$gene_name),] %&gt;%
    dplyr::select(-gene_id)
rownames(tumor_count) &lt;- tumor_count$gene_name
tumor_count &lt;- tumor_count %&gt;%
    dplyr::select(-gene_name)</code></pre>
<pre class="r"><code>new_diag_col &lt;- qualitative_hcl(2, palette = &quot;Dark 3&quot;) %&gt;%
    set_names(c(&quot;IDC-like&quot;, &quot;ILC-like&quot;))
# tumor_sample &lt;- tumor_sample %&gt;%
#     mutate(
#         Diagnosis = ifelse(
#             str_detect(Diagnosis, &quot;invasive&quot;), &quot;ILC-like&quot;, &quot;IDC-like&quot;
#         )
#     )</code></pre>
</div>
</div>
<div id="exppression-of-ilc-marker-genes" class="section level1">
<h1>Exppression of ILC marker genes</h1>
<p>By correlating the first three PCA components of all samples with
corresponding clinical information, a strong confounding effect of
pathological diagnosis was revealed. In the clinical records of tumours
used in this study, tumours were either designated <em>invasive
carcinoma of no special type</em> or <em>infiltrating carcinoma of no
special type</em>, which pathologically is considered to be the same
entity. An independent pathological analysis of hematoxylin and eosin
(H&amp;E) staining sections of representative uncultured tumour
fragments confirmed they were ductal carcinomas with no distinguishing
features, ruling out the possibility that some were lobular carcinomas
which had been misclassified as ductal carcinomas. However, the
different labels indicated that subtle phenotypic distinctions between
the two groups of tumours were observed by the pathologist, and it is
known that clinically tumours of no special type often have mixed ductal
and lobular features.</p>
<p>So we examined the expression of a few genes that are known to be
distinct between IDC and ILC tumours.</p>
<pre class="r"><code>geneOfInterest &lt;- dge_tumor$genes %&gt;%
    dplyr::filter(gene_name %in% c(&quot;CDH1&quot;,  &quot;FOXA1&quot;, &quot;PTEN&quot;)) %&gt;%
    pull(gene_id)</code></pre>
<p>The ILC hallmarks that were identified in The Cancer Genome Atlas
(TCGA) breast tumour RNA-seq dataset (lack of E-cadherin <em>CDH1</em>,
loss of <em>PTEN</em> and elevated <em>FOXA1</em> expression) were also
observed in our cohort of primary tumours that were labelled to be
<em>invasive carcinoma of no special type</em>. Therefore, we
hypothesized that transcriptionally, the group of samples labelled as
<em>invasive carcinoma of no special type</em> by the pathologist may be
more “lobular-like” and the other group is more “ductal-like”, whilst
still being with the larger category of ductal carcinomas.</p>
<pre class="r"><code>logCPM_tumor %&gt;%
    .[rownames(.) %in% geneOfInterest, ] %&gt;%
    # .[, str_detect(colnames(.), &quot;Tumor-2&quot;, TRUE)] %&gt;%
    as.data.frame() %&gt;%
    rownames_to_column(&quot;gene_id&quot;) %&gt;%
    pivot_longer(
        cols = -&quot;gene_id&quot;, 
        names_to = &quot;name&quot;, 
        values_to = &quot;logCPM&quot;
    ) %&gt;%
    left_join(tumor_sample) %&gt;%
    left_join(dge_tumor$genes) %&gt;%
 
    ggplot(
        aes(Diagnosis, logCPM, color = Diagnosis)
    ) +
    geom_boxplot() +
    # ggpubr::stat_compare_means(method = &quot;t.test&quot;,
    #                    paired = FALSE,
    #                    label.x = 1.5,
    #                    label.y = 8.5,
    #                    size = 4,
    #                    aes(label = paste0(&quot;p-value = &quot;, ..p.format..))
    # ) +
    geom_point() +
    scale_color_manual(
        values = set_names(new_diag_col, 
                           c(&quot;infiltrating carcinoma of no special type&quot;, 
                             &quot;invasive carcinoma of no special type&quot;))) +
    scale_x_discrete(labels = function(x) str_wrap(x, width = 20)) +
    geom_label_repel(
        aes(label = name),
        data = . %&gt;%
            dplyr::filter(patient == &quot;Tumor-2&quot;, 
                          gene_name == &quot;CDH1&quot;),
        size = 3) +
    facet_wrap(~gene_name)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/IDC_ILC.Rmd/unnamed-chunk-4-1.png" alt="*Expressions of ILC marker genes in tumours that were labelled differently by the pathologist.*" width="1344" />
<p class="caption">
<em>Expressions of ILC marker genes in tumours that were labelled
differently by the pathologist.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Wenjun-Liu/WenjunLiu_Thesis_Chapter4/blob/4b1209781ee751444ce0c6a132ac20b5edfee771/docs/figure/IDC_ILC.Rmd/unnamed-chunk-4-1.png" target="_blank">4b12097</a>
</td>
<td>
Wenjun-Liu
</td>
<td>
2023-08-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="tcga-data" class="section level1">
<h1>TCGA data</h1>
<p>In an attempt to further explore whether the two groups of PDE
tumours can be distinguished by relative levels of ductal versus lobular
features, a sparse partial least squares – discriminant analysis
(sPLS-DA) classifier was trained and tested using mixOmics., using TCGA
female primary breast tumour cohorts.</p>
<p>Breast tumors TCGA data was retrieved from <strong>GEO</strong>
GSE62944 using <code>ExperimentalHub()</code></p>
<pre class="r"><code>eh = ExperimentHub()
query(eh , &quot;GSE62944&quot;)</code></pre>
<pre><code>ExperimentHub with 3 records
# snapshotDate(): 2023-04-24
# $dataprovider: GEO
# $species: Homo sapiens
# $rdataclass: SummarizedExperiment, ExpressionSet
# additional mcols(): taxonomyid, genome, description,
#   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,
#   rdatapath, sourceurl, sourcetype 
# retrieve records with, e.g., &#39;object[[&quot;EH1&quot;]]&#39; 

           title                                                              
  EH1    | RNA-Sequencing and clinical data for 7706 tumor samples from The...
  EH1043 | RNA-Sequencing and clinical data for 9246 tumor samples from The...
  EH1044 | RNA-Sequencing and clinical data for 741 normal samples from The...</code></pre>
<pre class="r"><code>tcga_data &lt;- eh[[&quot;EH1&quot;]]
breast_data &lt;-  tcga_data[, which(phenoData(tcga_data)$CancerType==&quot;BRCA&quot;)]
breast_data &lt;-  breast_data[, which(phenoData(breast_data)$gender ==&quot;FEMALE&quot;)]
breast_data &lt;-  breast_data[, which(phenoData(breast_data)$histological_type %in% c(&quot;Infiltrating Ductal Carcinoma&quot;, &quot;Infiltrating Lobular Carcinoma&quot;))]</code></pre>
<div id="pca" class="section level2">
<h2>PCA</h2>
<pre class="r"><code>proteinGene &lt;- genesGR %&gt;%
    as.data.frame() %&gt;%
    # group_by(gene_name) %&gt;%
    # filter(n() == 1) %&gt;%
    # ungroup() %&gt;%
    dplyr::filter(gene_biotype == &quot;protein_coding&quot;) %&gt;%
    pull(gene_name)
breast_data &lt;-  breast_data[ which(featureNames(breast_data)%in% proteinGene),]</code></pre>
<p>PCA was firstly performed on all primary tumor tissues derived from
female donors that were either IDC or ILC. To reduce running time, only
the top 5k genes with the highest average counts were included.</p>
<pre class="r"><code>top5000_bothGroup &lt;- exprs(breast_data) %&gt;%
    rowMeans() %&gt;%
    sort(decreasing = TRUE) %&gt;%
    .[1:5000] %&gt;%
    names()
bothGroup_pca &lt;- breast_data[ which(featureNames(breast_data )%in% top5000_bothGroup),] %&gt;%
    exprs() %&gt;%
    t() %&gt;%
    prcomp() 
 phenoData(breast_data)$histological_type &lt;- ifelse(
      str_detect(phenoData(breast_data)$histological_type, &quot;Ductal&quot;), 
      &quot;IDC&quot;, &quot;ILC&quot;
 )</code></pre>
<pre class="r"><code>bothGroup_import &lt;- percent(summary(bothGroup_pca)$importance[&quot;Proportion of Variance&quot;, ], 0.1) %&gt;%
    paste(names(.), ., sep = &quot;: &quot;)
p1 &lt;- bothGroup_pca$x[, 1:5] %&gt;%
    as.data.frame() %&gt;%
    rownames_to_column(&quot;sample&quot;) %&gt;%
    mutate(
       Diagnosis = phenoData(breast_data)$histological_type) %&gt;%
    ggplot(
        aes(PC1, PC2, color =Diagnosis)
    ) +
    geom_point() +
    labs(
        x = bothGroup_import[1], 
        y = bothGroup_import[2]
    ) +
    scale_color_manual(values = diag_cols)
p2 &lt;-bothGroup_pca$x[, 1:5] %&gt;%
    as.data.frame() %&gt;%
    rownames_to_column(&quot;sample&quot;) %&gt;%
    mutate(
       Diagnosis = phenoData(breast_data)$histological_type) %&gt;%
    ggplot(
        aes(PC3, PC4, color =Diagnosis)
    ) +
    geom_point() +
    labs(
        x = bothGroup_import[3], 
        y = bothGroup_import[4]
    ) +
    scale_color_manual(values = diag_cols)
plot_grid(
    plot_grid(
    p1 + theme(legend.position = &quot;none&quot;),
    p2 + theme(legend.position = &quot;none&quot;),
    nrow = 1), 
    get_legend(p1), 
    nrow = 1, 
    rel_widths = c(7.5,1))</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/IDC_ILC.Rmd/unnamed-chunk-8-1.png" alt="*PCA plots of all IDC and ILC samples. No clear separation between the two groups could be observed.*" width="1344" />
<p class="caption">
<em>PCA plots of all IDC and ILC samples. No clear separation between
the two groups could be observed.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Wenjun-Liu/WenjunLiu_Thesis_Chapter4/blob/4b1209781ee751444ce0c6a132ac20b5edfee771/docs/figure/IDC_ILC.Rmd/unnamed-chunk-8-1.png" target="_blank">4b12097</a>
</td>
<td>
Wenjun-Liu
</td>
<td>
2023-08-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>IDC &lt;- breast_data[ which(featureNames(breast_data) %in% rownames(tumor_count)), which(phenoData(breast_data)$histological_type == &quot;IDC&quot;)]
ILC &lt;- breast_data[ which(featureNames(breast_data) %in% rownames(tumor_count)), which(phenoData(breast_data)$histological_type == &quot;ILC&quot;)]</code></pre>
<p>There were 728 IDC and 164 ILC samples in the dataset.50 IDCs and 50
ILCs were randomly sampled to form the training dataset while the
remaining data formed the test dataset.</p>
<pre class="r"><code>set.seed(123)
IDC_train_sample &lt;- sample(phenoData(IDC)$bcr_patient_barcode, 50)
IDC_train &lt;- IDC[, which(phenoData(IDC)$bcr_patient_barcode %in% IDC_train_sample)]
IDC_test &lt;- IDC[, which(!phenoData(IDC)$bcr_patient_barcode %in% IDC_train_sample)]
ILC_train_sample &lt;- sample(phenoData(ILC)$bcr_patient_barcode, 50)
ILC_train &lt;- ILC[, which(phenoData(ILC)$bcr_patient_barcode %in% ILC_train_sample)]
ILC_test &lt;- ILC[, which(!phenoData(ILC)$bcr_patient_barcode %in% ILC_train_sample)]</code></pre>
</div>
</div>
<div id="spls-da" class="section level1">
<h1>sPLS-DA</h1>
<div id="normalise-to-z-score" class="section level2">
<h2>Normalise to z-score</h2>
<p>To make sure the expression data is comparable across different
studies, they will all be converted to logCPM first before converting to
z-scores.</p>
<pre class="r"><code>combined_train &lt;- cbind(exprs(IDC_train), exprs(ILC_train)[match(rownames(exprs(ILC_train)), rownames(exprs(IDC_train))), ]) %&gt;%
    .[!duplicated(rownames(.)), ] %&gt;%
     edgeR::cpm(log = TRUE)</code></pre>
<pre class="r"><code>combined_train_mean &lt;- apply(combined_train, 1, mean)
combined_train_sd &lt;- apply(combined_train, 1, sd)
combined_train &lt;- sapply(rownames(combined_train), function(x){
    (combined_train[x,] - combined_train_mean[x])/combined_train_sd[x]
})</code></pre>
<p>To start analysis, a basic MINT PLS-DA model was formed, in which all
features were used to construct an arbitrarily chosen number of
components.</p>
<pre class="r"><code>train_type &lt;- rep(c(&quot;IDC&quot;, &quot;ILC&quot;), times = c(ncol(exprs(IDC_train)), ncol(exprs(ILC_train))))
# train_study &lt;- rep(c(&quot;tcga&quot;, &quot;gse80098&quot;), times = c((ncol(exprs(IDC_train)) + ncol(exprs(ILC_train)))))
basic.plsda.model &lt;- splsda(
    X = combined_train, 
    Y = train_type, 
    ncomp = 10)</code></pre>
<p>The number of components was set to 10 for the initial inspection.
There is a large degree of overlap between ILC &amp; IDC on the first
two components.</p>
<pre class="r"><code>plotIndiv(basic.plsda.model, comp = c(1,2), ellipse = TRUE,  ind.names = FALSE) </code></pre>
<p><img src="figure/IDC_ILC.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Wenjun-Liu/WenjunLiu_Thesis_Chapter4/blob/4b1209781ee751444ce0c6a132ac20b5edfee771/docs/figure/IDC_ILC.Rmd/unnamed-chunk-14-1.png" target="_blank">4b12097</a>
</td>
<td>
Wenjun-Liu
</td>
<td>
2023-08-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>and the 3rd and 4th components.</p>
<pre class="r"><code>plotIndiv(basic.plsda.model, comp = c(3,4), ellipse = TRUE,  ind.names = FALSE) </code></pre>
<p><img src="figure/IDC_ILC.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Wenjun-Liu/WenjunLiu_Thesis_Chapter4/blob/4b1209781ee751444ce0c6a132ac20b5edfee771/docs/figure/IDC_ILC.Rmd/unnamed-chunk-15-1.png" target="_blank">4b12097</a>
</td>
<td>
Wenjun-Liu
</td>
<td>
2023-08-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="tune-parameter" class="section level2">
<h2>Tune parameter</h2>
<p><code>perf()</code> function was used to choose the optimal number of
components.</p>
<pre class="r"><code># set.seed(30)
# Perf.plsda &lt;- perf(basic.plsda.model, validation = &quot;Mfold&quot;, folds = 3,
#                    progressBar = FALSE, nrepeat = 50, cpus = 8)
# saveRDS(Perf.plsda, here::here(&quot;output/Perf_plsda.rds&quot;))
Perf.plsda &lt;- readRDS(here::here(&quot;output/Perf_plsda.rds&quot;))
matplot(Perf.plsda$error.rate$BER, type = &#39;l&#39;, lty = 1,
        col = color.mixo(1:3),
        main = &#39;Balanced Error rate&#39;)
legend(&#39;topright&#39;,
       c(&#39;max.dist&#39;, &#39;centroids.dist&#39;, &#39;mahalanobis.dist&#39;),
       lty = 1,
       col = color.mixo(5:7))</code></pre>
<p><img src="figure/IDC_ILC.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-1">
Past versions of unnamed-chunk-16-1.png
</button>
</p>
<div id="fig-unnamed-chunk-16-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Wenjun-Liu/WenjunLiu_Thesis_Chapter4/blob/4b1209781ee751444ce0c6a132ac20b5edfee771/docs/figure/IDC_ILC.Rmd/unnamed-chunk-16-1.png" target="_blank">4b12097</a>
</td>
<td>
Wenjun-Liu
</td>
<td>
2023-08-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Based on the graph above, 4 components gives the lowest balanced
error rate.</p>
<p><code>tune.splsda()</code> function was then used to tune the number
of features for each component. The <code>ncomp</code> parameter was set
to 6 because there can be disagreement between the optimal number of
components chosen by the two functions.</p>
<pre class="r"><code># set.seed(30)
# tune.splsda.srbct &lt;- tune.splsda(
#     X = combined_train,
#     Y = train_type,
#     ncomp = 6,
#     validation = &#39;Mfold&#39;,
#     folds = 3, dist = &#39;max.dist&#39;, progressBar = FALSE,
#     measure = &quot;BER&quot;, test.keepX = c(5:20,
#                                      seq(20, 300, 5)
#                                     ),
#     nrepeat = 50,
#     )
# saveRDS(tune.splsda.srbct, here::here(&quot;output/tune_splsda_srbct.rds&quot;))
tune.splsda.srbct &lt;- readRDS(here::here(&quot;output/tune_splsda_srbct.rds&quot;)) </code></pre>
<pre class="r"><code>ncomp &lt;- tune.splsda.srbct$choice.ncomp$ncomp # optimal number of components based on t-tests on the error rate
select.keepX &lt;- tune.splsda.srbct$choice.keepX[1:2] </code></pre>
<p>The optimal number of components decided by this method was 2, and
the numbers of features were 19, 247.</p>
<p>The final model was plotted and there is now a better separation
between the two groups on 1st component, but there is still a certain
degree of overlap.</p>
<pre class="r"><code>splsda.final &lt;- splsda(
    X = combined_train, 
    Y = train_type,  
    ncomp = ncomp, 
    keepX = select.keepX 
)
plotIndiv(splsda.final, ind.names = FALSE, legend=TRUE,
          ellipse = TRUE,  title=&quot;SPLS-DA, Final result&quot;, 
          comp = c(1, 2))</code></pre>
<p><img src="figure/IDC_ILC.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-19-1">
Past versions of unnamed-chunk-19-1.png
</button>
</p>
<div id="fig-unnamed-chunk-19-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Wenjun-Liu/WenjunLiu_Thesis_Chapter4/blob/4b1209781ee751444ce0c6a132ac20b5edfee771/docs/figure/IDC_ILC.Rmd/unnamed-chunk-19-1.png" target="_blank">4b12097</a>
</td>
<td>
Wenjun-Liu
</td>
<td>
2023-08-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The decision boundaries decided by this final model were visualised.
New data will be projected on this prediction background.</p>
<pre class="r"><code>background &lt;- background.predict(splsda.final, comp.predicted=2,
                                 dist = &quot;max.dist&quot;) 
plotIndiv(splsda.final, comp = 1:2, group = train_type,
          ind.names = FALSE, title = &quot;Maximum distance&quot;,
          legend = TRUE,  background = background)</code></pre>
<p><img src="figure/IDC_ILC.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-20-1">
Past versions of unnamed-chunk-20-1.png
</button>
</p>
<div id="fig-unnamed-chunk-20-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Wenjun-Liu/WenjunLiu_Thesis_Chapter4/blob/4b1209781ee751444ce0c6a132ac20b5edfee771/docs/figure/IDC_ILC.Rmd/unnamed-chunk-20-1.png" target="_blank">4b12097</a>
</td>
<td>
Wenjun-Liu
</td>
<td>
2023-08-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="comp1-features" class="section level3">
<h3>Comp1 features</h3>
<p>Features selected for the 1st component were:</p>
<pre class="r"><code>plotLoadings(splsda.final, contrib = &#39;max&#39;, method = &#39;mean&#39;, comp = 1, 
             legend.color = diag_cols[c(&quot;IDC&quot;, &quot;ILC&quot;)])</code></pre>
<p><img src="figure/IDC_ILC.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-21-1">
Past versions of unnamed-chunk-21-1.png
</button>
</p>
<div id="fig-unnamed-chunk-21-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Wenjun-Liu/WenjunLiu_Thesis_Chapter4/blob/4b1209781ee751444ce0c6a132ac20b5edfee771/docs/figure/IDC_ILC.Rmd/unnamed-chunk-21-1.png" target="_blank">4b12097</a>
</td>
<td>
Wenjun-Liu
</td>
<td>
2023-08-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>comp1_gene &lt;- splsda.final$loadings$X %&gt;%
    as.data.frame() %&gt;% 
    rownames_to_column(&quot;gene_name&quot;) %&gt;%
    dplyr::filter(comp1 != 0) %&gt;%
    pull(gene_name)
combined_train_df &lt;- t(combined_train) %&gt;%
    as.data.frame() %&gt;%
    rownames_to_column(&quot;gene_name&quot;) %&gt;%
    pivot_longer(
        cols = -&quot;gene_name&quot;,
        names_to = &quot;sample&quot;, 
                 values_to = &quot;normalised expression&quot;) %&gt;%
    left_join(
        data.frame(
            type = train_type, 
            sample = rownames(combined_train))
    )</code></pre>
<pre class="r"><code>IDC_highGene &lt;- combined_train_df %&gt;%
    dplyr　::filter(gene_name %in% comp1_gene) %&gt;%
    split(f = .$gene_name) %&gt;%
    lapply(group_by, type) %&gt;%
    lapply(summarize, median = median(`normalised expression`)) %&gt;%
    sapply(function(x){x[1,2] &gt; x[2,2]}) %&gt;%
    enframe(
        name = &quot;gene_name&quot;, 
        value = &quot;high_in_IDC&quot;
    ) %&gt;%
    mutate(
        color = ifelse(high_in_IDC, &quot;firebrick&quot;, &quot;midnightblue&quot;)
    )</code></pre>
</div>
<div id="comp2-features" class="section level3">
<h3>Comp2 features</h3>
<pre class="r"><code>comp2_gene &lt;- splsda.final$loadings$X %&gt;%
    as.data.frame() %&gt;% 
    rownames_to_column(&quot;gene_name&quot;) %&gt;%
    dplyr::filter(comp2 != 0) %&gt;%
    pull(gene_name)
IDC_highGene_comp2 &lt;- combined_train_df %&gt;%
    dplyr　::filter(gene_name %in% comp2_gene) %&gt;%
    split(f = .$gene_name) %&gt;%
    lapply(group_by, type) %&gt;%
    lapply(summarize,  median = median(`normalised expression`)) %&gt;%
    sapply(function(x){x[1,2] &gt; x[2,2]}) %&gt;%
    enframe(
        name = &quot;gene_name&quot;, 
        value = &quot;high_in_IDC&quot;
    ) %&gt;%
    mutate(
        color = ifelse(high_in_IDC, &quot;firebrick&quot;, &quot;midnightblue&quot;)
    )</code></pre>
<p>A total of 247 component 2 features were selected by the
classifier.</p>
</div>
<div id="tf-features-selected" class="section level3">
<h3>TF features selected</h3>
<p>A vector of all known transciption factors was created by summarizing
6 TFT datasets: TRED, ITFP, ENCODE, Neph2012, TRRUST and
Marbach2016.</p>
<pre class="r"><code># tftargets package summarizes 6 TFT datasets 
# library(RCurl)
# download.file(
#   url = &quot;https://raw.githubusercontent.com/slowkow/tftargets/master/data/tftargets.rda&quot;,
#   destfile = here::here(&quot;data/tftargets.rda&quot;),
#   method = &quot;curl&quot;
# )
TFT_ls &lt;- admisc::listRDA(here::here(&quot;data/tftargets.rda&quot;))
all_TF &lt;- lapply(TFT_ls, names) %&gt;%
    Reduce(union, .) %&gt;%
    unique()</code></pre>
<pre class="r"><code>IDC_highGene_comp2_tf &lt;- IDC_highGene_comp2 %&gt;%
    mutate(
        TF = ifelse(gene_name %in% all_TF, TRUE, FALSE)
    ) %&gt;%
    dplyr::filter(TF)</code></pre>
<p>50 of the component 2 features were transcription factors:</p>
<pre class="r"><code>IDC_highGene_comp2_tf %&gt;%
    mutate(
      `Group with Higher Expression` = ifelse(
          high_in_IDC, &quot;IDC&quot;, &quot;ILC&quot;
      )
    ) %&gt;%
    dplyr::select(
        Gene = gene_name, 
        `Group with Higher Expression`
    ) %&gt;%
    mutate_all(as.factor) %&gt;%
    datatable(
        filter = &quot;top&quot;
    )</code></pre>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-4e1770d66a86dcce9699" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-4e1770d66a86dcce9699">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td><\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;AHCTF1&quot;,&quot;ARID4B&quot;,&quot;ASH1L&quot;,&quot;ATF1&quot;,&quot;ATF2&quot;,&quot;BCLAF1&quot;,&quot;BDP1&quot;,&quot;CAND1&quot;,&quot;CCM2&quot;,&quot;CORO7&quot;,&quot;CPEB3&quot;,&quot;DDX3X&quot;,&quot;EP300&quot;,&quot;ETV3&quot;,&quot;FBXL3&quot;,&quot;FBXO28&quot;,&quot;FUBP3&quot;,&quot;GBX1&quot;,&quot;GSTA4&quot;,&quot;INTS4&quot;,&quot;IRF6&quot;,&quot;ITGB1BP1&quot;,&quot;KDM5B&quot;,&quot;LCOR&quot;,&quot;LGALS8&quot;,&quot;MATR3&quot;,&quot;MPHOSPH8&quot;,&quot;MSC&quot;,&quot;NKX2-2&quot;,&quot;NUFIP1&quot;,&quot;PER2&quot;,&quot;PIR&quot;,&quot;POGK&quot;,&quot;PPP2R2A&quot;,&quot;PSPC1&quot;,&quot;RBBP5&quot;,&quot;RBM26&quot;,&quot;RNF169&quot;,&quot;RSF1&quot;,&quot;SMG7&quot;,&quot;SNAI1&quot;,&quot;SP3&quot;,&quot;TBX2&quot;,&quot;WDR36&quot;,&quot;YTHDC1&quot;,&quot;ZBTB33&quot;,&quot;ZNF281&quot;,&quot;ZNF317&quot;,&quot;ZNF350&quot;,&quot;ZNF551&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;IDC&quot;,&quot;ILC&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n<\/tr>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50"],["AHCTF1","ARID4B","ASH1L","ATF1","ATF2","BCLAF1","BDP1","CAND1","CCM2","CORO7","CPEB3","DDX3X","EP300","ETV3","FBXL3","FBXO28","FUBP3","GBX1","GSTA4","INTS4","IRF6","ITGB1BP1","KDM5B","LCOR","LGALS8","MATR3","MPHOSPH8","MSC","NKX2-2","NUFIP1","PER2","PIR","POGK","PPP2R2A","PSPC1","RBBP5","RBM26","RNF169","RSF1","SMG7","SNAI1","SP3","TBX2","WDR36","YTHDC1","ZBTB33","ZNF281","ZNF317","ZNF350","ZNF551"],["IDC","ILC","ILC","ILC","ILC","ILC","ILC","IDC","ILC","ILC","ILC","IDC","IDC","ILC","ILC","IDC","ILC","IDC","IDC","ILC","ILC","ILC","IDC","ILC","ILC","ILC","ILC","ILC","ILC","ILC","ILC","IDC","IDC","ILC","ILC","IDC","ILC","IDC","IDC","IDC","IDC","IDC","ILC","ILC","ILC","IDC","ILC","ILC","ILC","ILC"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Gene<\/th>\n      <th>Group with Higher Expression<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
<p>They are potentially implicated in mediating the transcriptional
difference between the two tumour groups.</p>
</div>
</div>
<div id="test-model" class="section level2">
<h2>Test model</h2>
<p>The fitted model was used to predict the diagnostic type of test
data, which contained 17293 and 17293 samples.</p>
<pre class="r"><code>combined_test &lt;- cbind(exprs(IDC_test), exprs(ILC_test)[match(rownames(exprs(ILC_test)), rownames(exprs(IDC_test))), ])  %&gt;%
    .[!duplicated(rownames(.)), ] %&gt;%
    edgeR::cpm(log = TRUE)
combined_test_mean &lt;- apply(combined_test, 1, mean)
combined_test_sd &lt;- apply(combined_test, 1, sd)
combined_test &lt;- sapply(rownames(combined_test), function(x){
    (combined_test[x,] - combined_test_mean[x])/combined_test_sd[x]
}) </code></pre>
<p>Condfusion matrix with the predicted labels and the true lables
were:</p>
<pre class="r"><code>test_type &lt;- rep(c(&quot;IDC&quot;, &quot;ILC&quot;), times = c(ncol(exprs(IDC_test)), ncol(exprs(ILC_test))))
splsda.test &lt;- predict(splsda.final,
                       combined_test, 
                       dist = &quot;all&quot;)</code></pre>
<p>The ROC curve of the testing was:</p>
<pre class="r"><code>auroc(splsda.final, newdata = combined_test, outcome.test = test_type, roc.comp = 2, print = FALSE,
     legend.title = &quot;Subtype&quot;)</code></pre>
<p><img src="figure/IDC_ILC.Rmd/unnamed-chunk-30-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-30-1">
Past versions of unnamed-chunk-30-1.png
</button>
</p>
<div id="fig-unnamed-chunk-30-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Wenjun-Liu/WenjunLiu_Thesis_Chapter4/blob/4b1209781ee751444ce0c6a132ac20b5edfee771/docs/figure/IDC_ILC.Rmd/unnamed-chunk-30-1.png" target="_blank">4b12097</a>
</td>
<td>
Wenjun-Liu
</td>
<td>
2023-08-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="prediction" class="section level2">
<h2>Prediction</h2>
<p>The trained and tested model was used to predict the subtype of all
the vehicle PDEs.</p>
<pre class="r"><code>cpm_tumor &lt;-  tumor_count %&gt;%
    edgeR::cpm(log = TRUE) %&gt;%
     .[, str_detect(colnames(.), &quot;Veh&quot;)] %&gt;%
    .[match(colnames(splsda.final$X), rownames(.)),]
cpm_tumor_mean &lt;- apply(cpm_tumor, 1, mean)
cpm_tumor_sd &lt;- apply(cpm_tumor, 1, sd)
cpm_tumor &lt;- sapply(rownames(cpm_tumor), function(x){
    (cpm_tumor[x,] - cpm_tumor_mean[x])/cpm_tumor_sd[x]
}) </code></pre>
<p>The prediction results were:</p>
<pre class="r"><code>splsda.final &lt;- splsda(
    X = combined_train, 
    Y = train_type,  
    ncomp = ncomp, 
    keepX = c(19, 311) 
)
splsda.pred &lt;- predict(splsda.final,
                       cpm_tumor , 
                       dist = &quot;centroids.dist&quot;)
splsda.pred$class$centroids.dist %&gt;%
    as.data.frame() %&gt;%
    rownames_to_column(&quot;name&quot;) %&gt;%
    left_join(dge_tumor$samples) %&gt;%
    mutate(
        Diagnosis = ifelse(
            str_detect(Diagnosis, &quot;invasive&quot;), &quot;ILC-like&quot;, &quot;IDC-like&quot;
        )
    ) %&gt;%
    dplyr::select(patient, &quot;comp2&quot;, 
                  Diagnosis,  
                  # Total_Epithelial
                  ) %&gt;%
    unique() %&gt;%
    mutate(Alignment = ifelse(
        (str_detect(Diagnosis, &quot;ILC&quot;) &amp; comp2 == &quot;ILC&quot; | 
            str_detect(Diagnosis, &quot;IDC&quot;) &amp; comp2 == &quot;IDC&quot;), 
        &quot;\u2714 Yes&quot;, &quot;\u2716 No&quot;
    )) %&gt;%
    dplyr::rename(
        Patient = patient,
        `Manual Prediction` = Diagnosis, 
        `sPLS-DA Prediction` = comp2
    ) %&gt;%
    mutate_all(as.factor) %&gt;%
    datatable(
        filter = &quot;top&quot;, 
        height = 700,
    ) %&gt;%
    formatStyle(
        c( &quot;sPLS-DA Prediction&quot;),
        color = styleEqual(c(&quot;ILC&quot;, &quot;IDC&quot;), c(&quot;#00AD9A&quot;, &quot;#E16A86&quot;))
    ) %&gt;%
    formatStyle(
        &quot;Manual Prediction&quot;,
        color = styleEqual(c(&quot;ILC-like&quot;, &quot;IDC-like&quot;), c(&quot;#00AD9A&quot;, &quot;#E16A86&quot;))
    ) %&gt;%
    formatStyle(
        &#39;Alignment&#39;,
            color = styleEqual(c( &quot;\u2714 Yes&quot;, &quot;\u2716 No&quot;), c(&quot;#008000&quot;, &quot;#e00000&quot;))
    )</code></pre>
<div id="htmlwidget-65edfe9f869aa904450f" style="width:100%;height:700px;" class="datatables html-widget "></div>
<script type="application/json" data-for="htmlwidget-65edfe9f869aa904450f">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td><\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;Tumor-1&quot;,&quot;Tumor-2&quot;,&quot;Tumor-3&quot;,&quot;Tumor-4&quot;,&quot;Tumor-5&quot;,&quot;Tumor-6&quot;,&quot;Tumor-7&quot;,&quot;Tumor-8&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;IDC&quot;,&quot;ILC&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;IDC-like&quot;,&quot;ILC-like&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;✔ Yes&quot;,&quot;✖ No&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n<\/tr>","data":[["1","2","3","4","5","6","7","8"],["Tumor-1","Tumor-2","Tumor-3","Tumor-4","Tumor-5","Tumor-6","Tumor-7","Tumor-8"],["ILC","IDC","IDC","ILC","IDC","ILC","ILC","IDC"],["ILC-like","IDC-like","IDC-like","ILC-like","IDC-like","ILC-like","ILC-like","ILC-like"],["✔ Yes","✔ Yes","✔ Yes","✔ Yes","✔ Yes","✔ Yes","✔ Yes","✖ No"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Patient<\/th>\n      <th>sPLS-DA Prediction<\/th>\n      <th>Manual Prediction<\/th>\n      <th>Alignment<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[2]; $(this.api().cell(row, 2).node()).css({'color':value == \"ILC\" ? \"#00AD9A\" : value == \"IDC\" ? \"#E16A86\" : null});\nvar value=data[3]; $(this.api().cell(row, 3).node()).css({'color':value == \"ILC-like\" ? \"#00AD9A\" : value == \"IDC-like\" ? \"#E16A86\" : null});\nvar value=data[4]; $(this.api().cell(row, 4).node()).css({'color':value == \"✔ Yes\" ? \"#008000\" : value == \"✖ No\" ? \"#e00000\" : null});\n}"},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":["options.rowCallback"],"jsHooks":[]}</script>
<p>The classifier’s prediction results and the prediction based on the
pathologist’s label and marker gene expressions aligned for all but one
sample Tumor-8.</p>
<pre class="r"><code>pred_result &lt;- splsda.pred$class$centroids.dist %&gt;%
    as.data.frame() %&gt;%
    rownames_to_column(&quot;name&quot;) %&gt;%
    left_join(dge_tumor$samples) %&gt;%
    mutate(Diagnosis = ifelse(
        str_detect(Diagnosis, &quot;invasive&quot;), &quot;ILC&quot;, &quot;IDC&quot;
    )) %&gt;%
    dplyr::select(
        patient, 
        `sPLS-DA prediction` = &quot;comp2&quot;, 
        `Prediction by\nmarker genes` = Diagnosis) %&gt;%
    unique() %&gt;%
    mutate(align = ifelse(
        `sPLS-DA prediction` == `Prediction by\nmarker genes`, 
        TRUE, FALSE
    )) %&gt;%
    pivot_longer(
        cols = -c(&quot;patient&quot;, &quot;align&quot;), 
        names_to = &quot;Type&quot;, 
        values_to = &quot;Subtype&quot;
    ) </code></pre>
<p>When a PCA was performed on the 8 treatment-naive samples, tumour-8
was found to be more transcriptionally similar to the rest of samples
that are labelled as <em>infiltrating carcinoma of no special
type</em>.</p>
<pre class="r"><code>pcaPost_tumor_veh &lt;- logCPM_tumor %&gt;%
    .[, str_detect(colnames(.), &quot;Veh&quot;)] %&gt;%
    t() %&gt;%
    prcomp() 
pcaPost_tumor_veh %&gt;%
    autoplot(
        data = dge_tumor$samples %&gt;%
            dplyr::filter(treat == &quot;Veh&quot;), 
        x = 1, y = 2,
        colour = &quot;Diagnosis&quot;, 
        shape = &quot;treat&quot;,
        size = 5
    ) +
    scale_shape_manual(values = treat_shapes) +
    guides(shape = FALSE) +
    scale_color_manual(values = c(
        &quot;infiltrating carcinoma of no special type&quot; = &quot;#E16A86&quot;,
            &quot;invasive carcinoma of no special type&quot; = &quot;#00AD9A&quot;
    ), 
    name = &quot;Pathologist\nLabel&quot;) +
    geom_label_repel(
        data = . %&gt;%
            dplyr::filter(patient == &quot;Tumor-8&quot;),
        aes(label = patient),
        size = 4) +
    labs(
        colour = &quot;Diagnosis&quot;,
        shape = &quot;Treatment&quot;, 
        x = &quot;PC1&quot;, 
        y = &quot;PC2&quot;
    ) +
    theme(
        legend.position = &quot;none&quot;
    )</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/IDC_ILC.Rmd/unnamed-chunk-34-1.png" alt="*PCA plot of only the untreated samples. The sample with disconcordant labels between the manual prediction and the ML classification is Tumour-8.*" width="480" />
<p class="caption">
<em>PCA plot of only the untreated samples. The sample with
disconcordant labels between the manual prediction and the ML
classification is Tumour-8.</em>
</p>
</div>
<p>Based on the classifier results and the PCA plot, we classified 4 PDE
tumours (Tumour-1, 4, 6, and 7 ) as lobular-like and 4 (Tumour-2, 3, 5,
and 8) as ductal-like in subsequent analyses.</p>
<pre class="r"><code>tumor_sample &lt;- tumor_sample %&gt;%
    dplyr::mutate(
            Diagnosis = ifelse(
                str_detect(Diagnosis, &quot;invasive&quot;), &quot;ILC-like&quot;, &quot;IDC-like&quot;
            ),
            Diagnosis = ifelse(patient == &quot;Tumor-8&quot;, &quot;IDC-like&quot;, Diagnosis)
    )</code></pre>
</div>
</div>
<div id="classifer-validation-using-cell-line-data"
class="section level1">
<h1>Classifer validation using cell line data</h1>
<p>To further validate the sPLS-DA classifier, RNA-Seq expression
profiling of 81 breast cancer cell lines (without replicates) were
retrieved from project <strong>SRP064259</strong> using
<code>recount3</code>.</p>
<p>The <a
href="https://www.cell.com/cell/pdfExtended/S0092-8674(15)01624-4">original
study</a> didn’t provide information about whether a cell line was a
lobular or ductal line, but a full list of current ILC and ILC-like cell
lines available were found n <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8582475/"><em>Atlas
of Lobular Breast Cancer Models: Challenges and Strategic
Directions</em></a>.</p>
<p>Cell lines with known histology information that were used to
validate the classifier were:</p>
<pre class="r"><code>cell_meta_his &lt;- cell_meta %&gt;%
    # dplyr::filter(str_detect(cell_line, &quot;MCF7&quot;))
    mutate(
        Histology = case_when(
            cell_line %in% c(&quot;MDAMB134VI&quot;, &quot;SUM44&quot;, &quot;MDAMB330&quot;) ~ &quot;ILC&quot;,
            cell_line %in% c(&quot;MDAMB453&quot;, &quot;MDAMB468&quot;, &quot;CAMA1&quot;, &quot;SKBR3&quot;,
                             &quot;EVSAT&quot;, &quot;CAL148&quot;, &quot;ZR7530&quot;, &quot;HCC2218&quot;, &quot;600MPE&quot;,
                             &quot;BT549&quot;) ~ &quot;ILC-like&quot;,
            cell_line %in% c(&quot;MCF7&quot;, &quot;T47D&quot;, &quot;MDAMB231&quot;) ~ &quot;IDC&quot;
        )
    ) %&gt;%
    dplyr::filter(
        !is.na(Histology)
    )
cell_meta_his %&gt;%
    dplyr::select(cell_line, Histology) %&gt;%
    mutate_all(as.factor) %&gt;%
    datatable(filter = &quot;top&quot;)</code></pre>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-171f14364aa3ebedfbfc" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-171f14364aa3ebedfbfc">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td><\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;600MPE&quot;,&quot;BT549&quot;,&quot;CAL148&quot;,&quot;CAMA1&quot;,&quot;EVSAT&quot;,&quot;HCC2218&quot;,&quot;MCF7&quot;,&quot;MDAMB134VI&quot;,&quot;MDAMB231&quot;,&quot;MDAMB330&quot;,&quot;MDAMB453&quot;,&quot;MDAMB468&quot;,&quot;SKBR3&quot;,&quot;SUM44&quot;,&quot;T47D&quot;,&quot;ZR7530&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;IDC&quot;,&quot;ILC&quot;,&quot;ILC-like&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n<\/tr>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16"],["HCC2218","MDAMB134VI","600MPE","BT549","CAL148","CAMA1","EVSAT","MCF7","MDAMB231","MDAMB330","MDAMB453","MDAMB468","SKBR3","SUM44","T47D","ZR7530"],["ILC-like","ILC","ILC-like","ILC-like","ILC-like","ILC-like","ILC-like","IDC","IDC","ILC","ILC-like","ILC-like","ILC-like","ILC","IDC","ILC-like"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>cell_line<\/th>\n      <th>Histology<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
<div id="qc" class="section level2">
<h2>QC</h2>
<pre class="r"><code>public_CellLine_count &lt;- assays(public_CellLine)$counts %&gt;%
    .[, colnames(.) %in% cell_meta_his$sample]
rownames(public_CellLine_count) &lt;- vapply(rownames(public_CellLine_count), function(x){
    str_split(x, &quot;\\.&quot;)[[1]][1]
}, character(1)) %&gt;%
    unname()
minCPM &lt;- 1.5
minSamples &lt;- 3
genes2Keep &lt;- public_CellLine_count  %&gt;%
  edgeR::cpm() %&gt;%
  is_greater_than(minCPM) %&gt;%
  rowSums() %&gt;%
  is_weakly_greater_than(minSamples)</code></pre>
<p>The criteria for a gene to be considered as detected, <span
class="math inline">\(&gt;\)</span> 1.5 counts per million (CPM) were
required to observed in <span class="math inline">\(\geq\)</span> 3
samples.</p>
<p>Of the 63,856 genes contained in the annotation for this release,
44,881 genes were removed as failing this criteria for detection,
leaving 18,975 genes for downstream analysis.</p>
<pre class="r"><code>p1 &lt;- public_CellLine_count %&gt;%
  edgeR::cpm(log = TRUE) %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;gene_id&quot;) %&gt;%
  as_tibble() %&gt;%
  pivot_longer(
    cols = -&quot;gene_id&quot;,
    names_to = &quot;sample&quot;,
    values_to = &quot;logCPM&quot;
  ) %&gt;%
  left_join(cell_meta_his) %&gt;%
  ggplot(aes(logCPM, stat(density), colour = Histology, linetype = Histology)) +
  geom_density() +
  scale_colour_manual(values= c(diag_cols[c(&quot;IDC&quot;, &quot;ILC&quot;)], &quot;ILC-like&quot; = &quot;darkgreen&quot;) ) +
  labs(
    y = &quot;Density&quot;
  )
p2 &lt;- public_CellLine_count[genes2Keep,] %&gt;%
edgeR::cpm(log = TRUE) %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;gene_id&quot;) %&gt;%
  as_tibble() %&gt;%
  pivot_longer(
    cols = -&quot;gene_id&quot;,
    names_to = &quot;sample&quot;,
    values_to = &quot;logCPM&quot;
  ) %&gt;%
  left_join(cell_meta_his) %&gt;%
  ggplot(aes(logCPM, stat(density), colour = Histology, linetype = Histology)) +
  geom_density() +
  scale_colour_manual(values= c(diag_cols[c(&quot;IDC&quot;, &quot;ILC&quot;)], &quot;ILC-like&quot; = &quot;darkgreen&quot;) ) +
  labs(
    y = &quot;Density&quot;
  )
plot_grid(p1, p2)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/IDC_ILC.Rmd/plotDensities-1.png" alt="*Distributions of logCPM values on cell line raw counts, A) before and B) after filtering.*" width="672" />
<p class="caption">
<em>Distributions of logCPM values on cell line raw counts, A) before
and B) after filtering.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plotDensities-1">
Past versions of plotDensities-1.png
</button>
</p>
<div id="fig-plotDensities-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Wenjun-Liu/WenjunLiu_Thesis_Chapter4/blob/4b1209781ee751444ce0c6a132ac20b5edfee771/docs/figure/IDC_ILC.Rmd/plotDensities-1.png" target="_blank">4b12097</a>
</td>
<td>
Wenjun-Liu
</td>
<td>
2023-08-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>A <code>DGEList</code> was created for those selected cell lines
after filtration.</p>
<pre class="r"><code>dge_CellLine &lt;- public_CellLine_count[genes2Keep,] %&gt;%
  DGEList(
    samples = cell_meta_his %&gt;%
      as.data.frame() %&gt;%
      set_rownames(.$sample) %&gt;%
      .[colnames(public_CellLine_count),],
    group = cell_meta_his %&gt;%
      as.data.frame() %&gt;%
      set_rownames(.$sample) %&gt;%
      .[colnames(public_CellLine_count),] %&gt;%
      pull(Histology),
    genes = genesGR %&gt;%
        as.data.frame() %&gt;%
        .[rownames(public_CellLine_count[genes2Keep,] ),]
  ) %&gt;%
  calcNormFactors()</code></pre>
<pre class="r"><code>dge_CellLine$samples %&gt;%
    ggplot(aes(sample, lib.size, fill = Histology)) +
    geom_col() +
    geom_hline(yintercept = 1e7, linetype = 2) +
    scale_y_continuous(
        labels = comma, expand = expansion(c(0, 0.05))
    ) +
    scale_fill_manual(values=  c(diag_cols[c(&quot;IDC&quot;, &quot;ILC&quot;)], &quot;ILC-like&quot; = &quot;darkgreen&quot;)) +
    labs(x = &quot;Sample Name&quot;, y = &quot;Library Size&quot;)  +
    theme(legend.position = &quot;top&quot;,
          axis.text.x = element_text(angle = 45, vjust = 0.5))</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/IDC_ILC.Rmd/plotLibSizes-1.png" alt="*Library sizes of all samples after removal of undetectable genes. The common-use minimum library size of 10 million reads is shown as a dashed line.*" width="1536" />
<p class="caption">
<em>Library sizes of all samples after removal of undetectable genes.
The common-use minimum library size of 10 million reads is shown as a
dashed line.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-plotLibSizes-1">
Past versions of plotLibSizes-1.png
</button>
</p>
<div id="fig-plotLibSizes-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Wenjun-Liu/WenjunLiu_Thesis_Chapter4/blob/4b1209781ee751444ce0c6a132ac20b5edfee771/docs/figure/IDC_ILC.Rmd/plotLibSizes-1.png" target="_blank">4b12097</a>
</td>
<td>
Wenjun-Liu
</td>
<td>
2023-08-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Conditional quantile normalisation was performed, and the PCA was
performed post-normalisation.</p>
<pre class="r"><code>dge_CellLine &lt;- dge_CellLine[!is.na( dge_CellLine$genes$ave_tx_len), ]
cqNorma_CellLine &lt;-cqn(
    counts=  dge_CellLine$counts,
    x =  dge_CellLine$genes$gc_content,
    lengths =  dge_CellLine$genes$ave_tx_len

)
dge_CellLine$offset &lt;- cqNorma_CellLine$glm.offset
logCPM_CellLine &lt;- cqNorma_CellLine$y + cqNorma_CellLine$offset
pcaPost_CellLine &lt;- logCPM_CellLine %&gt;%
  t() %&gt;%
  prcomp()</code></pre>
<p>The first PCA components were correlated with receptor subtype and
histology subtype.</p>
<pre class="r"><code>pcaPost_CellLine$x %&gt;%
    as.data.frame() %&gt;%
    rownames_to_column(&quot;sample&quot;) %&gt;%
    left_join(cell_meta_his) %&gt;%
    dplyr::select(PC1, PC2, PC3, PC4, PC5, `Receptor subtype` = subtype_three_receptor, Histology) %&gt;%
    mutate_if(is.character, as.factor) %&gt;%
    mutate_if(is.factor, as.numeric) %&gt;%
    cor() %&gt;%
    corrplot(
        type = &quot;lower&quot;,
        diag = FALSE,
        addCoef.col = &quot;black&quot;,
        addCoefasPercent = TRUE
    )</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/IDC_ILC.Rmd/unnamed-chunk-41-1.png" alt="*Correlatoins between the first PCA components among the 16 cell lines selected and receptor subtype and histology subtype.*" width="480" />
<p class="caption">
<em>Correlatoins between the first PCA components among the 16 cell
lines selected and receptor subtype and histology subtype.</em>
</p>
</div>
<pre class="r"><code>p1 &lt;- pcaPost_CellLine %&gt;%
    autoplot(
        data = cell_meta_his,
        x = 1, y = 2,
        color = &quot;subtype_three_receptor&quot;,
        shape = &quot;subtype_three_receptor&quot;,
        size = 4
    )
p2 &lt;- pcaPost_CellLine %&gt;%
    autoplot(
        data = cell_meta_his,
        x = 2, y = 3,
        color = &quot;Histology&quot;,
        shape = &quot;subtype_three_receptor&quot;,
        size = 4
    ) +
    scale_color_manual(values=  c(diag_cols[c(&quot;IDC&quot;, &quot;ILC&quot;)], &quot;ILC-like&quot; = &quot;darkgreen&quot;))
plot_grid(p1, p2,
          labels = LETTERS[1:2])</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/IDC_ILC.Rmd/unnamed-chunk-42-1.png" alt="*PCA plots of the 16 selected cell lines, colored by (A) receptor subtypes, and (B) histology subtypes. The PC1 direction was driven by receptor subtype while the PC2 direction was more driven by histologfy subtypes*" width="1536" />
<p class="caption">
<em>PCA plots of the 16 selected cell lines, colored by (A) receptor
subtypes, and (B) histology subtypes. The PC1 direction was driven by
receptor subtype while the PC2 direction was more driven by histologfy
subtypes</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-42-1">
Past versions of unnamed-chunk-42-1.png
</button>
</p>
<div id="fig-unnamed-chunk-42-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Wenjun-Liu/WenjunLiu_Thesis_Chapter4/blob/4b1209781ee751444ce0c6a132ac20b5edfee771/docs/figure/IDC_ILC.Rmd/unnamed-chunk-42-1.png" target="_blank">4b12097</a>
</td>
<td>
Wenjun-Liu
</td>
<td>
2023-08-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="filtration" class="section level2">
<h2>Filtration</h2>
<pre class="r"><code>cell_meta_his &lt;- cell_meta_his %&gt;%
    dplyr::filter(subtype_three_receptor == &quot;ER&quot;)
dge_CellLine &lt;- dge_CellLine[, dge_CellLine$samples$sample %in% cell_meta_his$sample]
logCPM_CellLine &lt;- dge_CellLine$counts %&gt;%
    cpm(log =TRUE)
pcaPost_CellLine &lt;- logCPM_CellLine %&gt;%
  t() %&gt;%
  prcomp()</code></pre>
<p>To simplify the analysis,only ER-positive cell lines were kept,
leaving the following cell lines in the analysis:</p>
<pre class="r"><code>cell_meta_his %&gt;%
    dplyr::select(cell_line, Histology) %&gt;%
    chop(&quot;cell_line&quot;) %&gt;%
    mutate(
        cell_line = vapply(.$cell_line, function(x)paste(x, collapse = &quot;; &quot;),
                           character(1))
    ) %&gt;%
    pander(
    )</code></pre>
<table style="width:49%;">
<colgroup>
<col width="16%" />
<col width="31%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">Histology</th>
<th align="center">cell_line</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">ILC</td>
<td align="center">MDAMB134VI; SUM44</td>
</tr>
<tr class="even">
<td align="center">ILC-like</td>
<td align="center">600MPE; CAMA1; EVSAT</td>
</tr>
<tr class="odd">
<td align="center">IDC</td>
<td align="center">MCF7; T47D</td>
</tr>
</tbody>
</table>
<pre class="r"><code>p1 &lt;- pcaPost_CellLine %&gt;%
    autoplot(
        data = cell_meta_his,
        x = 1, y = 2,
        color = &quot;Histology&quot;,
        size = 4
    ) +
    geom_label_repel(
        aes(label = cell_line)
    ) +
    scale_color_manual(values=  c(diag_cols[c(&quot;IDC&quot;, &quot;ILC&quot;)], &quot;ILC-like&quot; = &quot;darkgreen&quot;))
p2 &lt;- pcaPost_CellLine %&gt;%
    autoplot(
        data = cell_meta_his,
        x = 3, y = 4,
        color = &quot;Histology&quot;,
        size = 4
    ) +
    geom_label_repel(
        aes(label = cell_line)
    ) +
    scale_color_manual(values=  c(diag_cols[c(&quot;IDC&quot;, &quot;ILC&quot;)], &quot;ILC-like&quot; = &quot;darkgreen&quot;))
plot_grid(p1, p2,
          labels = LETTERS[1:2])</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/IDC_ILC.Rmd/unnamed-chunk-45-1.png" alt="*PCA plots of the 7 selected cell lines on (A) PC1 &amp; PC2 directions, and (B) PC3 &amp; PC4 direction. The 2 ILC-like cell lines CAMA1 and 600MPE seem to be closer to IDC cell lines on the PC2 direction, while the other ILC-like cell line EVSAT was an outlier on both PC1 and PC4 directions. *" width="1536" />
<p class="caption">
<em>PCA plots of the 7 selected cell lines on (A) PC1 &amp; PC2
directions, and (B) PC3 &amp; PC4 direction. The 2 ILC-like cell lines
CAMA1 and 600MPE seem to be closer to IDC cell lines on the PC2
direction, while the other ILC-like cell line EVSAT was an outlier on
both PC1 and PC4 directions. </em>
</p>
</div>
</div>
<div id="marker-expression" class="section level2">
<h2>Marker Expression</h2>
<p>The expressions of lobular hallmark CDH1 werevisualised as a boxplot.
As expected, expressions were the highest in the IDC cell lines and
lowest in the ILC cell lines. It’s interesting to observe that the
ILC-like model EVSAT had CDH1 expression that were as high as the median
of the two IDC lines.</p>
<pre class="r"><code>cdh1_cell &lt;- logCPM_CellLine %&gt;%
    as.data.frame() %&gt;%
    rownames_to_column(&quot;gene_id&quot;) %&gt;%
    pivot_longer(
        cols = -&quot;gene_id&quot;,
        names_to = &quot;sample&quot;,
        values_to = &quot;logCPM&quot;
    ) %&gt;%
    left_join(cell_meta_his) %&gt;%
    left_join(dge_CellLine$genes) %&gt;%
    dplyr::filter(gene_name %in% c(&quot;CDH1&quot;)) %&gt;%
    ggplot(
        aes(Histology, logCPM, color = Histology)
    ) +
    geom_boxplot() +
    scale_color_manual(
        values = c(diag_cols[c(&quot;IDC&quot;, &quot;ILC&quot;)], &quot;ILC-like&quot; = &quot;darkgreen&quot;)) +
    theme(
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank()
    )+
    xlab(&quot;&quot;) +
    geom_point(aes(fill = Histology)) +
    # facet_wrap(~gene_name) +
    geom_label_repel(
        data = . %&gt;%
            dplyr::filter(
                Histology == &quot;ILC-like&quot;, 
                gene_name %in% c(&quot;CDH1&quot;, &quot;PTEN&quot;)), 
        aes(label = cell_line)
    ) +
    ggtitle(&quot;CDH1 expression&quot;)
cdh1_cell</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/IDC_ILC.Rmd/unnamed-chunk-46-1.png" alt="*LogCPM expressions of ILC marker CDH1 in the 7 cell lines, colored by histology subtypes.*" width="480" />
<p class="caption">
<em>LogCPM expressions of ILC marker CDH1 in the 7 cell lines, colored
by histology subtypes.</em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-46-1">
Past versions of unnamed-chunk-46-1.png
</button>
</p>
<div id="fig-unnamed-chunk-46-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Wenjun-Liu/WenjunLiu_Thesis_Chapter4/blob/4b1209781ee751444ce0c6a132ac20b5edfee771/docs/figure/IDC_ILC.Rmd/unnamed-chunk-46-1.png" target="_blank">4b12097</a>
</td>
<td>
Wenjun-Liu
</td>
<td>
2023-08-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="spls-da-prediction" class="section level2">
<h2>sPLS-DA prediction</h2>
<p>To apply the sPLS-DA for prediction, the logCPM matrix was normalised
to z-scores.</p>
<pre class="r"><code>logCPM_CellLine2 &lt;- logCPM_CellLine
rownames(logCPM_CellLine2) &lt;- unname(mapIds(ensDb, rownames(logCPM_CellLine2), &quot;GENENAME&quot;, keytype = &quot;GENEID&quot; ))
logCPM_CellLine2 &lt;- logCPM_CellLine2 %&gt;%
    .[match(colnames(splsda.final$X), rownames(.)),]
logCPM_CellLine2 &lt;- logCPM_CellLine2[!is.na(rownames(logCPM_CellLine2)),]
missingGene &lt;- setdiff(colnames(splsda.final$X), rownames(logCPM_CellLine2))
temp &lt;- matrix(0, ncol = ncol(logCPM_CellLine2), nrow = length(missingGene))
rownames(temp) &lt;- missingGene
logCPM_CellLine2 &lt;- rbind(logCPM_CellLine2, temp)
cpm_mean_CellLine &lt;- apply(logCPM_CellLine2, 1, mean)
cpm_sd_CellLine &lt;- apply(logCPM_CellLine2, 1, sd)
cpm_scaled_CellLine &lt;- sapply(rownames(logCPM_CellLine2), function(x){
    (logCPM_CellLine2[x,] - cpm_mean_CellLine[x])/cpm_sd_CellLine[x]
})</code></pre>
<p>Prediction results of the 7 cell lines were:</p>
<pre class="r"><code>splsda.pred_CellLine &lt;- predict(splsda.final,
                                cpm_scaled_CellLine %&gt;%
                                    .[,match(colnames(splsda.final$X), colnames(.))],
                                dist = &quot;all&quot;)
splsda.pred_CellLine$class$mahalanobis.dist %&gt;%
    as.data.frame() %&gt;%
    rownames_to_column(&quot;sample&quot;) %&gt;%
    left_join(cell_meta_his) %&gt;%
    dplyr::select(cell_line, 
                  `sPLS-DA prediction` = &quot;comp2&quot;, 
                  Histology) %&gt;%
    unique() %&gt;%
    mutate(
        Alignment = if_else(
          Histology == &quot;ILC-like&quot; &amp; `sPLS-DA prediction` == &quot;IDC&quot;, &quot;\u2716 No&quot;,  &quot;\u2714 Yes&quot;, )
        ) %&gt;%
    mutate_all(as.factor) %&gt;%
    datatable(
        filter = &quot;top&quot;,
        height = 700,
    ) %&gt;%
    formatStyle(
        c( &quot;sPLS-DA prediction&quot;),
        color = styleEqual(c(&quot;ILC&quot;, &quot;IDC&quot;), c(&quot;#00AD9A&quot;, &quot;#E16A86&quot;))
    ) %&gt;%
    formatStyle(
        &quot;Histology&quot;,
        color = styleEqual(names(c(diag_cols[c(&quot;IDC&quot;, &quot;ILC&quot;)], &quot;ILC-like&quot; = &quot;darkgreen&quot;)), c(diag_cols[c(&quot;IDC&quot;, &quot;ILC&quot;)], &quot;ILC-like&quot; = &quot;darkgreen&quot;))
    ) %&gt;%
    formatStyle(
        &#39;Alignment&#39;,
            color = styleEqual(c( &quot;\u2714 Yes&quot;, &quot;\u2716 No&quot;), c(&quot;#008000&quot;, &quot;#e00000&quot;))
    )</code></pre>
<div id="htmlwidget-52d82635d988c6e12f4c" style="width:100%;height:700px;" class="datatables html-widget "></div>
<script type="application/json" data-for="htmlwidget-52d82635d988c6e12f4c">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td><\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;600MPE&quot;,&quot;CAMA1&quot;,&quot;EVSAT&quot;,&quot;MCF7&quot;,&quot;MDAMB134VI&quot;,&quot;SUM44&quot;,&quot;T47D&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;IDC&quot;,&quot;ILC&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;IDC&quot;,&quot;ILC&quot;,&quot;ILC-like&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;✔ Yes&quot;,&quot;✖ No&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n<\/tr>","data":[["1","2","3","4","5","6","7"],["MDAMB134VI","600MPE","CAMA1","EVSAT","MCF7","SUM44","T47D"],["ILC","ILC","ILC","IDC","IDC","ILC","IDC"],["ILC","ILC-like","ILC-like","ILC-like","IDC","ILC","IDC"],["✔ Yes","✔ Yes","✔ Yes","✖ No","✔ Yes","✔ Yes","✔ Yes"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>cell_line<\/th>\n      <th>sPLS-DA prediction<\/th>\n      <th>Histology<\/th>\n      <th>Alignment<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[2]; $(this.api().cell(row, 2).node()).css({'color':value == \"ILC\" ? \"#00AD9A\" : value == \"IDC\" ? \"#E16A86\" : null});\nvar value=data[3]; $(this.api().cell(row, 3).node()).css({'color':value == \"IDC\" ? \"#E16A86\" : value == \"ILC\" ? \"#00AD9A\" : value == \"ILC-like\" ? \"darkgreen\" : null});\nvar value=data[4]; $(this.api().cell(row, 4).node()).css({'color':value == \"✔ Yes\" ? \"#008000\" : value == \"✖ No\" ? \"#e00000\" : null});\n}"},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":["options.rowCallback"],"jsHooks":[]}</script>
<p>All the ILC and IDC models were classified corrrectly by the
classifier, while only one of the three ILC-like line EVSAT, which was
already identified as an outlier in the PCA plots, was predicted to be
IDC.</p>
</div>
</div>
<div id="score-ilc-gene-signatures" class="section level1 tabset">
<h1 class="tabset">Score ILC gene signatures</h1>
<p>In <a
href="https://pubmed.ncbi.nlm.nih.gov/30228172/"><em>Comprehensive
Phenotypic Characterization of Human Invasive Lobular Carcinoma Cell
Lines in 2D and 3D Cultures</em></a>, ILC vs IDC comparisons were
carried out between cell ines and TCGA tumours. 14 consistent up and 17
consistent down signatures were found. The list of 31 sigantures were
loaded in, and <code>singscore</code> was used to score those signatures
in all untreated samples.</p>
<p>The signatures used here were:</p>
<pre class="r"><code>ILC_sig &lt;- read_delim(&quot;~/20131906_HickeyT_JC_NormalBreast/data/ILC_signature.txt&quot;)
ILC_sig %&gt;%
  dplyr::select(
    `Gene name` = gene_name, 
    `Type of sig` = type
  ) %&gt;%
  mutate_all(as.factor) %&gt;%
  datatable(filter = &quot;top&quot;)%&gt;%
  formatStyle(
    &#39;Type of sig&#39;,
    color = styleEqual(c(&quot;ILC_high&quot;, &quot;ILC_low&quot;), c( &quot;#e00000&quot;, &quot;blue&quot;))
  )</code></pre>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-52b7f36e26c300a71d5d" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-52b7f36e26c300a71d5d">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td><\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;APRT&quot;,&quot;BBOX1&quot;,&quot;C16orf45&quot;,&quot;CCR2&quot;,&quot;CDH1&quot;,&quot;CLEC3B&quot;,&quot;CPEB1&quot;,&quot;CPXM2&quot;,&quot;CX3CR1&quot;,&quot;CYP27B1&quot;,&quot;DPP3&quot;,&quot;EEF1A2&quot;,&quot;GJB2&quot;,&quot;HIST1H3H&quot;,&quot;ISG15&quot;,&quot;KCNF1&quot;,&quot;NR3C2&quot;,&quot;PCDHGB4&quot;,&quot;PDE4B&quot;,&quot;PDE7B&quot;,&quot;PGR&quot;,&quot;PLOD2&quot;,&quot;PPFIBP2&quot;,&quot;PTGIS&quot;,&quot;PXDN&quot;,&quot;RASSF6&quot;,&quot;SULF1&quot;,&quot;TFAP2B&quot;,&quot;TGFBI&quot;,&quot;TSPAN7&quot;,&quot;WDR72&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;ILC_high&quot;,&quot;ILC_low&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n<\/tr>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31"],["PPFIBP2","RASSF6","CPEB1","PDE4B","CCR2","NR3C2","PDE7B","CX3CR1","CPXM2","PTGIS","CLEC3B","TSPAN7","BBOX1","TFAP2B","CDH1","APRT","DPP3","PXDN","C16orf45","TGFBI","PCDHGB4","PGR","CYP27B1","GJB2","ISG15","HIST1H3H","PLOD2","SULF1","EEF1A2","KCNF1","WDR72"],["ILC_high","ILC_high","ILC_high","ILC_high","ILC_high","ILC_high","ILC_high","ILC_high","ILC_high","ILC_high","ILC_high","ILC_high","ILC_high","ILC_high","ILC_low","ILC_low","ILC_low","ILC_low","ILC_low","ILC_low","ILC_low","ILC_low","ILC_low","ILC_low","ILC_low","ILC_low","ILC_low","ILC_low","ILC_low","ILC_low","ILC_low"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Gene name<\/th>\n      <th>Type of sig<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true,"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[2]; $(this.api().cell(row, 2).node()).css({'color':value == \"ILC_high\" ? \"#e00000\" : value == \"ILC_low\" ? \"blue\" : null});\n}"},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":["options.rowCallback"],"jsHooks":[]}</script>
<p>Interestingly, among this list of 31 signatures, only CDH1 was also
selected as a feature by the sPLS-DA.</p>
<pre class="r"><code># IDC_highGene %&gt;%
#     mutate(comp = &quot;comp1&quot;) %&gt;%
#     rbind(
#         IDC_highGene_comp2 %&gt;%
#             mutate(comp = &quot;comp2&quot;)
#     ) %&gt;%
#     dplyr::filter(gene_name %in% ILC_sig$gene_name)</code></pre>
<pre class="r"><code>ILC_sig_ls &lt;- ILC_sig %&gt;%
  left_join(genesGR %&gt;%
              as.data.frame() %&gt;%
                dplyr::filter(gene_biotype == &quot;protein_coding&quot;, 
                              gene_id != &quot;ENSG00000277445&quot;)) %&gt;%
    mutate(
        gene_id = ifelse(gene_name == &quot;HIST1H3H&quot;, &quot;ENSG00000278828&quot;, gene_id)
    ) %&gt;%
  drop_na() %&gt;%
  split(f = .$type) %&gt;%
  lapply(pull, gene_id)</code></pre>
<p>To apply <code>singscore</code>, the raw counts for both the
untreated PDEs and the cell lines were converted to <code>tpm</code>
values to avoid the confounding effect of library sizes.</p>
<pre class="r"><code>GeneLength &lt;- genesGR %&gt;%
    as_tibble() %&gt;%
    dplyr::select(gene_id, ave_tx_len) 
veh_count &lt;- mergedCounts %&gt;%
  .[,str_detect(colnames(.), &quot;Veh&quot;)] %&gt;%
  .[,str_detect(colnames(.), &quot;Tumor&quot;)]
tpm_func &lt;- function(counts,len) {
    x &lt;- counts/len
    return(t(t(x)*1e6/colSums(x)))
}
tpm_veh &lt;- tpm_func(veh_count, GeneLength %&gt;%
                        dplyr::filter(gene_id %in% rownames(veh_count)) %&gt;%
                        .[match(rownames(veh_count), .$gene_id),] %&gt;% 
                        pull(ave_tx_len))
ranked_veh &lt;- rankGenes(tpm_veh)
cell_count &lt;- - dge_CellLine$counts
cell_count &lt;- cell_count %&gt;%
    .[rownames(.) %in% GeneLength$gene_id,]
tpm_cell &lt;- tpm_func(cell_count, GeneLength %&gt;%
                        dplyr::filter(gene_id %in% rownames(cell_count)) %&gt;%
                        .[match(rownames(cell_count), .$gene_id),] %&gt;% 
                        pull(ave_tx_len))
ranked_cell &lt;- rankGenes(tpm_cell)</code></pre>
<p>3 ILC-high signatures ENSG00000121807, ENSG00000168329,
ENSG00000156298 were not detected in this cell line data.</p>
<pre class="r"><code>ILC_score_cell &lt;- simpleScore(
  ranked_cell, 
  upSet = ILC_sig_ls$ILC_high, 
  downSet = ILC_sig_ls$ILC_low
  ) 
ILC_score_veh &lt;- simpleScore(
  ranked_veh, 
  upSet = ILC_sig_ls$ILC_high, 
  downSet = ILC_sig_ls$ILC_low
  ) </code></pre>
<div id="cell-lines" class="section level2">
<h2>Cell lines</h2>
<p>Clear differences in the total ILC scores between the three cell
lines types were observed.</p>
<pre class="r"><code>pred_result_cell &lt;- splsda.pred_CellLine$class$mahalanobis.dist %&gt;%
    as.data.frame() %&gt;%
    rownames_to_column(&quot;sample&quot;) %&gt;%
    left_join(cell_meta_his) %&gt;%
    dplyr::select(cell_line,
                  `sPLS-DA prediction` = &quot;comp2&quot;,
                  `Cell-line type` = Histology)  %&gt;%
    unique() %&gt;%
    mutate(
        align = ifelse(
            `sPLS-DA prediction` == `Cell-line type`,
            TRUE, FALSE),
        align = ifelse(
            `Cell-line type`  == &quot;ILC-like&quot; &amp; `sPLS-DA prediction` == &quot;ILC&quot;,
            TRUE, align
        )) %&gt;%
    pivot_longer(
        cols = -c(&quot;cell_line&quot;, &quot;align&quot;),
        names_to = &quot;Type&quot;,
        values_to = &quot;Subtype&quot;
    )</code></pre>
<pre class="r"><code>ILC_cell_p &lt;- ILC_score_cell %&gt;%
    rownames_to_column(&quot;sample&quot;) %&gt;%
    left_join(cell_meta_his) %&gt;%
    left_join(pred_result_cell) %&gt;%
    dplyr::filter(Type == &quot;Cell-line type&quot;) %&gt;%
    ggplot(
        aes(Subtype, TotalScore, color = Subtype )
    ) +
    geom_boxplot() +
    scale_color_manual(values = c(diag_cols, &quot;ILC-like&quot; = &quot;darkgreen&quot;), 
                       name = &quot;subtype&quot;) +

    labs(
        x = &quot;&quot;, 
        y = &quot;ILC scores&quot;) +
    geom_point(aes(color = Subtype)) +
    geom_label_repel(
        data = . %&gt;%
            dplyr::filter(Subtype == &quot;ILC-like&quot;),
        aes(label = cell_line),
        size = 3
    ) </code></pre>
<pre class="r"><code>plot_grid(
    cdh1_cell +
        theme(
            legend.position = &quot;none&quot;
        ), 
    ILC_cell_p + ggtitle(&quot;ILC score&quot;), 
    rel_widths = c(0.8, 1), 
    labels = LETTERS[1:2]
)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/IDC_ILC.Rmd/unnamed-chunk-56-1.png" alt="*(A) CDH1 expressions and (B) ILC score derived by scoring a list of 31 ILC signatures against the gene ranks within each samples in the 7 cell line models. Clear differences between the three cell line types were observed.*" width="672" />
<p class="caption">
<em>(A) CDH1 expressions and (B) ILC score derived by scoring a list of
31 ILC signatures against the gene ranks within each samples in the 7
cell line models. Clear differences between the three cell line types
were observed.</em>
</p>
</div>
</div>
<div id="pde" class="section level2">
<h2>PDE</h2>
<p>It’s interesting to observe that these marker genes could not
separate the untreated PDEs like they could in the cell lines and Ellis
AI data, indicating that those signatures might have less discriminatory
power in tumours with a mixture of phenotypes.</p>
<pre class="r"><code>ILC_score_veh %&gt;%
    rownames_to_column(&quot;name&quot;) %&gt;%
    dplyr::select(-contains(&quot;Dispersion&quot;)) %&gt;%
    pivot_longer(
        cols = -&quot;name&quot;, 
        names_to = &quot;Score_type&quot;, 
        values_to = &quot;score&quot;
    ) %&gt;%
    left_join(tumor_sample %&gt;%
                  dplyr::select(name, patient, Diagnosis)) %&gt;%
    ggplot(
        aes(Diagnosis, score, color = Diagnosis)
    ) +
    geom_boxplot() +
    scale_color_manual(values = new_diag_col, 
                       name = &quot;Predicted Subtype&quot;) +
    labs(
        x = &quot;&quot;, 
        y = &quot;ILC scores&quot;) +
    geom_point(aes(color = Diagnosis)) +
    facet_wrap(~Score_type
               # , scale = &quot;free_y&quot;
               )</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/IDC_ILC.Rmd/unnamed-chunk-57-1.png" alt="*ILC score derived by scoring a list of 31 ILC signatures against the gene ranks within each samples in the 8 untreated PDE models. No clear difference was observed in any of the score type. *" width="1152" />
<p class="caption">
<em>ILC score derived by scoring a list of 31 ILC signatures against the
gene ranks within each samples in the 8 untreated PDE models. No clear
difference was observed in any of the score type. </em>
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-57-1">
Past versions of unnamed-chunk-57-1.png
</button>
</p>
<div id="fig-unnamed-chunk-57-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/Wenjun-Liu/WenjunLiu_Thesis_Chapter4/blob/4b1209781ee751444ce0c6a132ac20b5edfee771/docs/figure/IDC_ILC.Rmd/unnamed-chunk-57-1.png" target="_blank">4b12097</a>
</td>
<td>
Wenjun-Liu
</td>
<td>
2023-08-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="de-bewteen-ilc-vs-idc" class="section level1">
<h1>DE bewteen ILC vs IDC</h1>
<p>Since the predicted tumour subtype is a known strong confounding
factor in this dataset and clear separation by this factor on the PCA
plots was observed, a DE analysis was performed to compare the
treatment-naive ILC-like tumours against the treatment-naive IDC-like
ones. The tumour <code>DGEList</code> was subsetted to derive a
vehicle-sample only <code>DGEList</code>.</p>
<pre class="r"><code>dge_tumor_veh &lt;- dge_tumor %&gt;%
    .[,str_detect(colnames(.$counts), &quot;Veh&quot;)]</code></pre>
<pre class="r"><code>X &lt;- model.matrix(~  Diagnosis,
                  data = dge_tumor_veh$samples %&gt;%
                      mutate(
                          Diagnosis = as.character(Diagnosis), 
                          Diagnosis = ifelse(
                              patient == &quot;Tumor-8&quot;, &quot;IDC-like&quot;, 
                              Diagnosis
                          ),
                          Diagnosis = as.factor(ifelse(
                              str_detect(Diagnosis, &quot;invasive&quot;), 
                              &quot;ILC-like&quot;, &quot;IDC-like&quot;))
                          )
) %&gt;%
    set_colnames(str_remove_all(colnames(.), &quot;Diagnosis&quot;)) %&gt;%
    .[,colSums(.) != 0]  
dge_tumor_veh &lt;- estimateDisp(dge_tumor_veh, design = X, robust = TRUE)
fit &lt;- glmQLFit(dge_tumor_veh)</code></pre>
<p>None of the gene passed the statistical threshold of FDR &lt;
0.05.</p>
<pre class="r"><code>alpha &lt;- 0.05
topTables_veh &lt;- glmQLFTest(fit, coef = &quot;ILC-like&quot;) %&gt;%
      topTags(n = Inf) %&gt;%
      .[[&quot;table&quot;]] %&gt;%
      as_tibble() %&gt;%
      mutate(
        location = paste0(seqnames, &quot;:&quot;, start, &quot;-&quot;, end, &quot;:&quot;, strand),
        rankingStat = -sign(logFC)*log10(PValue),
        signedRank = rank(rankingStat),
        DE = FDR &lt; alpha
      ) %&gt;%
      dplyr::select(
        gene_id, gene_name, logCPM, logFC, PValue, FDR, 
        location, gene_biotype, entrezid, ave_tx_len, gc_content,
        rankingStat, signedRank, DE
      )</code></pre>
<div id="gsea" class="section level2">
<h2>GSEA</h2>
<p>GSEA was performed on the ranking statistics derived from the DE
analysis.</p>
<pre class="r"><code>gene_ranks &lt;- topTables_veh %&gt;%
  arrange(FDR) %&gt;% 
  distinct(gene_name, .keep_all = TRUE) %&gt;% 
  arrange(desc(rankingStat)) %&gt;% 
  with(
    setNames(rankingStat, gene_name)
  ) </code></pre>
<pre class="r"><code>gsTopology &lt;- retrieve_topology(database = &quot;kegg&quot;, species = &quot;hsapiens&quot;)
gsTokeep &lt;- readRDS(here::here(&quot;output/kg_gsTokeep.rds&quot;))
load(system.file(&quot;extdata&quot;, &quot;entrez2name.rda&quot;, package = &quot;sSNAPPY&quot;))
kg &lt;- sapply(gsTopology, rownames) %&gt;%
    do.call(cbind, .) %&gt;%
    as.data.frame() %&gt;%
    pivot_longer(
        cols = everything(),
        names_to = &quot;gs_name&quot;,
        values_to = &quot;entrezid&quot;
    ) %&gt;%
    dplyr::filter(gs_name %in% gsTokeep) %&gt;%
    left_join(
        entrez2name %&gt;%
            dplyr::rename(gene_name = mapTo)
    ) %&gt;%
    left_join(genesGR %&gt;%
                  as.data.frame() %&gt;%
                  dplyr::select(gene_name, gene_id)) %&gt;%
    unique() %&gt;%
    dplyr::select(-entrezid)
kgByGS &lt;- kg %&gt;%
    split(f = .$gs_name) %&gt;%
    lapply(pull, gene_name)</code></pre>
<pre class="r"><code>library(fgsea)
gsea_res &lt;- fgsea(kgByGS,gene_ranks, eps = 0)
gsea_res_sig &lt;- gsea_res %&gt;%
    dplyr::filter(padj &lt; 0.05) %&gt;%
    arrange(NES) %&gt;%
    mutate(
        pathway = str_remove_all(pathway, &quot;kegg.&quot;), 
        pathway = as.factor(pathway)
    ) 
gsea_res_sig$pathway &lt;- factor(gsea_res_sig$pathway, levels = gsea_res_sig$pathway)</code></pre>
<p>34 pathways were found to be significantly enriched (FDR &lt; 0.05),
out of which 6 were enriched among genes with elevated expressions in
ILC-like tumours and 28 were enriched among genes with decreased
expressions.</p>
<pre class="r"><code>gsea_res_sig %&gt;%
    mutate(
        hjust = ifelse(NES &gt; 0, 1.05, -0.05)
    ) %&gt;%
    ggplot(
        aes(pathway, 
            NES, 
            fill = NES, 
        )) +
    geom_bar(stat=&quot;identity&quot;, position=&quot;stack&quot;) +
    scale_fill_distiller(palette = &quot;RdBu&quot;, 
                          limit = c(-2.5, 2.5),
                          name = &quot;NES&quot;) +
    xlab(&quot;&quot;) +
    geom_text(aes(label = pathway, y  = 0, hjust = hjust)) + 
    coord_flip()  +
    scale_x_discrete(
                     breaks = NULL) +
    theme(
        panel.grid = element_blank(), 
        axis.text.x = element_text(size = 12), 
        axis.text.y = element_text(size = 12), 
        panel.border = element_blank()
    )</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/IDC_ILC.Rmd/unnamed-chunk-62-1.png" alt="*KEGG pathays that were significantly enriched in ILC-like control PDEs, colored by the sign of normalised enrichment score.*" width="768" />
<p class="caption">
<em>KEGG pathays that were significantly enriched in ILC-like control
PDEs, colored by the sign of normalised enrichment score.</em>
</p>
</div>
<pre class="r"><code>logCPM_tumor_df &lt;- logCPM_tumor %&gt;%
    as.data.frame() %&gt;%
    rownames_to_column(&quot;gene_id&quot;) %&gt;%
    pivot_longer(
        cols = -&quot;gene_id&quot;, 
        names_to = &quot;name&quot;, 
        values_to = &quot;logCPM&quot;
    ) %&gt;%
    left_join(
        tumor_sample
    ) %&gt;%
    left_join(
        dge_tumor$genes %&gt;%
            dplyr::select(gene_id, gene_name)
    )</code></pre>
</div>
</div>
<div id="fu-et-al.-ilcidc-signature-genes" class="section level1">
<h1>Fu et al. ILC/IDC signature genes</h1>
<pre class="r"><code>fu_sig &lt;- read_csv(here::here(&quot;data/Fu_ILC_sig.csv&quot;)) %&gt;%
    .[, 1:2]
# IDC_highGene %&gt;%
#     mutate(comp = &quot;comp1&quot;) %&gt;%
#     rbind(
#         IDC_highGene_comp2 %&gt;%
#             mutate(comp = &quot;comp2&quot;)
#     ) %&gt;%
#     dplyr::filter(gene_name %in% fu_sig$gene_name)</code></pre>
<p>In the study by Fu et al. 2017, a set of genes capable of
distinguishing ILC from IDC were selected by applying a shrunken
centroid analysis to three public microarray dataset with clear
histology subtype diagnoses. To refine the gene list, an elastic-net
regularized linear modelling was used with a 10-fold cross validation
step. The final list consists of 46 genes. Those genes and their
coefficients in the predictive model were loaded in.</p>
<p>Again, interestingly, only CDH1 was in both the list of sPLS-DA
features selected and the list generated by Fu et al. The lack of
overlap in features selected using different classification methods was
mentioned in the manuscript for the sPLS-DA method. The author showed
that many clasisifcation methods had similar performance in
classificatio accuracy but selected very distinct features, meaning that
they may be capturing complementary information.</p>
<p>Among this list of ILC/IDC distinguishing features generated by Fu et
al. , CDH1, DPP3 were found to be consistently differentially expressed
between cell ines and TCGA tumours in the study by Tasdemir et
al. 2018.</p>
<p>Although the coefficients for the prediction model and the
classification cut-off were provided, how the authors scaled the RNA-seq
data was unclear, making it hard to implement the model.</p>
</div>
<div id="mesenchymal-marker-expression" class="section level1">
<h1>Mesenchymal marker expression</h1>
<p>Expressions of a few mesenchymal markers were in vehicle-treated
samples were visualised. Interestingly, the median expression of the
mesenchymal marker VIM is higher in the IDC-like tumours than the
ILC-like group. The median expression fo DDR2, a collagen-specific
receptor that reflects adaptation to the altered ECM microenvironment
associated with the EMT, is higher in the ILC-like group than the
IDC-like counterpart.</p>
<p>Expressions of the E-cadherin repressors ZEB1 and ZEB2 were higher in
smaples that were classified as ILC-like, aligning with the relatively
lower expression of E-cad observed in that group.</p>
<pre class="r"><code>mes_marker &lt;- read_csv(here::here(&quot;data/EMT_marker.txt&quot;))
logCPM_tumor %&gt;%
    as.data.frame() %&gt;%
    rownames_to_column(&quot;gene_id&quot;) %&gt;%
    pivot_longer(
        cols = -&quot;gene_id&quot;, 
        names_to = &quot;name&quot;, 
        values_to = &quot;logCPM&quot;
    ) %&gt;%
    left_join(tumor_sample) %&gt;%
    left_join(dge_tumor$genes) %&gt;%
    dplyr::filter(
        gene_name %in% mes_marker$gene_name,
         treat == &quot;Veh&quot;) %&gt;%
    ggplot(
        aes(Diagnosis, logCPM, color = Diagnosis)
    ) +
    geom_boxplot() +
    geom_point() +
    scale_color_manual(
        values = new_diag_col) +
    scale_x_discrete(labels = function(x) str_wrap(x, width = 20)) +
    ggrepel::geom_label_repel(
        aes(label = name),
        data = . %&gt;%
            dplyr::filter(patient == &quot;Tumor-2&quot;, 
                          gene_name == &quot;CDH1&quot;),
        size = 3) +
    facet_wrap(~gene_name, scales = &quot;free_y&quot;)</code></pre>
<p><img src="figure/IDC_ILC.Rmd/unnamed-chunk-65-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="thesis-figure" class="section level1">
<h1>Thesis Figure</h1>
<p>Code used to generate figures for the thesis.</p>
<div id="figure-4.2" class="section level2">
<h2>Figure 4.2</h2>
<pre class="r"><code># tumor_sample &lt;- tumor_sample %&gt;%
#     .[match(rownames(pcaPost_tumor$x), .$name),]
# pcaPost_tumor_df &lt;- pcaPost_tumor$x %&gt;%
#     as.data.frame() %&gt;%
#     rownames_to_column(&quot;name&quot;) %&gt;%
#     left_join(
#         tumor_sample
#     )
# chap4_1a &lt;- pcaPost_tumor_df %&gt;%
#     ggplot(
#         aes(PC1, PC2)
#     ) +
#     geom_point(aes( color = patient, shape = treat), 
#                size = 3) +
#     scale_shape_manual(values = treat_shapes) +
#     labs(
#         colour = &quot;Patient&quot;,
#         fill = &quot;Patient&quot;,
#         shape = &quot;Treatment&quot;
#     ) +
#     geom_mark_ellipse(
#         aes(
#             # label = patient,
#             fill = patient
#         ), 
#         # label.fontsize = 9
#     ) +
#     guides( fill = FALSE,
#             # color = guide_legend(nrow = 2), 
#             # shape = guide_legend(nrow = 2)
#             ) +
#     theme(
#         # legend.position = &quot;top&quot;, 
#         # legend.justification = &quot;right&quot;
#     )
# ```
# 
# ```{r}
# chap4_1b &lt;- pcaPost_tumor$x %&gt;%
#     as.data.frame() %&gt;%
#     rownames_to_column(&quot;name&quot;) %&gt;%
#     left_join(tumor_sample) %&gt;%
#     dplyr::select(PC1, PC2, PC3, patient, treat,  Diagnosis) %&gt;%
#     dplyr::rename(
#         Treatment = treat,
#         Patient = patient,
#         `Pathology Diagnosis` = Diagnosis
#     ) %&gt;%
#     mutate(
#            `Pathology Diagnosis` = as.factor(`Pathology Diagnosis`),
#            across(everything(), as.numeric)) %&gt;%
#     cor() %&gt;%
#     reshape2::melt() %&gt;%
#     dplyr::filter(
#         Var1 != Var2, 
#         # !str_detect(Var1, &quot;PC&quot;) &amp; str_detect(Var2, &quot;PC&quot;)
#         ) %&gt;%
#     .[!duplicated(.$value),] %&gt;%
#     ggplot(
#         aes(x = Var1, y = Var2, fill = value)) +
#     geom_tile() +
#     scale_fill_distiller(palette = &quot;RdBu&quot;, 
#                          limit = c(-1, 1)) +
#     scale_x_discrete(labels = function(x) str_wrap(x, width = 10)) +
#     scale_y_discrete(labels = function(x) str_wrap(x, width = 10)) +
#     labs(x = &quot;&quot;, y = &quot;&quot;, fill = &quot;Correlation&quot;) +
#     theme(
#         panel.background = element_blank() ,
#         panel.grid = element_blank(), 
#         panel.border = element_blank(),
#         # plot.margin = unit(c(0, 0, 0, 0), &quot;mm&quot;),
#         axis.ticks = element_blank(), 
#         legend.position = &quot;top&quot;, 
#         legend.justification = &quot;right&quot;,
#         legend.key.width = unit(0.8, &quot;cm&quot;)
#     ) +
#     geom_text(aes(x = Var1, y = Var2, label = round(value, 2)), color = &quot;black&quot;, 
#                size = 4)
# ```
# 
# ```{r}
# pcaPost_tumor_veh &lt;- logCPM_tumor %&gt;%
#     .[, str_detect(colnames(.), &quot;Veh&quot;)] %&gt;%
#     t() %&gt;%
#     prcomp() 
# chap4_1c &lt;- pcaPost_tumor_veh %&gt;%
#     autoplot(
#         data = dge_tumor$samples %&gt;%
#             dplyr::filter(treat == &quot;Veh&quot;), 
#         x = 1, y = 2,
#         colour = &quot;Diagnosis&quot;, 
#         shape = &quot;treat&quot;,
#         size = 5
#     ) +
#     scale_shape_manual(values = treat_shapes) +
#     guides(shape = FALSE) +
#     scale_color_manual(values = c(
#         &quot;infiltrating carcinoma of no special type&quot; = &quot;#E16A86&quot;,
#             &quot;invasive carcinoma of no special type&quot; = &quot;#00AD9A&quot;
#     ), 
#     name = &quot;Pathologist\nLabel&quot;) +
#     geom_label_repel(
#         data = . %&gt;%
#             dplyr::filter(patient == &quot;Tumor-8&quot;),
#         aes(label = patient),
#         size = 4) +
#     labs(
#         colour = &quot;Diagnosis&quot;,
#         shape = &quot;Treatment&quot;, 
#         x = &quot;PC1&quot;, 
#         y = &quot;PC2&quot;
#     ) +
#     theme(
#         legend.position = &quot;none&quot;
#     )
# ```
# 
# 
# ```{r}
# chap4_1d &lt;- logCPM_tumor %&gt;%
#     as.data.frame() %&gt;%
#     rownames_to_column(&quot;gene_id&quot;) %&gt;%
#     pivot_longer(
#         cols = -&quot;gene_id&quot;, 
#         names_to = &quot;name&quot;, 
#         values_to = &quot;logCPM&quot;
#     ) %&gt;%
#     left_join(dge_tumor$samples) %&gt;%
#     left_join(dge_tumor$genes) %&gt;%
#     dplyr::filter(
#         gene_name %in% c(&quot;CDH1&quot;, &quot;FOXA1&quot; ,&quot;PTEN&quot;), 
#         # treat == &quot;Veh&quot;
#         ) %&gt;%
#     mutate(
#         Diagnosis = str_to_title(Diagnosis),
#         Diagnosis = str_wrap(Diagnosis, width = 20)
#         ) %&gt;%
#     ggplot(
#         aes(Diagnosis, logCPM, fill = Diagnosis)
#     ) +
#     geom_boxplot() +
#     geom_point() +
#     facet_wrap(~gene_name) +
#     scale_fill_discrete_qualitative(name = &quot;Pathologist Label&quot;) +
#     xlab(&quot;&quot;) +
#     theme(
#         panel.grid = element_blank(), 
#         axis.text.x = element_blank(), 
#         axis.ticks.x = element_blank(),
#         # legend.position = &quot;top&quot;
#     )
# ```
# 
# ```{r}
# chap4_1e &lt;- pred_result %&gt;%
#     mutate(
#         Type = ifelse(
#             Type == &quot;sPLS-DA prediction&quot;, &quot;sPLS-DA&quot;, &quot;Pathologist Label&quot;
#         )
#     )  %&gt;%
#     ggplot(
#         aes(
#             str_wrap(Type, 10),  
#             patient, 
#             color = str_wrap(Subtype, 15)
#             )
#     ) +
#     geom_point(
#         size = 4, 
#         # aes(shape = Type)
#     ) +
#     scale_color_manual(
#         values = diag_cols %&gt;%
#             set_names(str_wrap(names(diag_cols), 7)), 
#         name = &quot;Subtype&quot;
#     ) +
#     coord_fixed(0.3) +
#     labs( x = &quot;&quot;, y = &quot;&quot;) +
#     theme(
#         axis.text.y = element_text(
#             color = c( rep(&quot;black&quot;, 7), &quot;red&quot;)
#         )
#     )
# # +
# #     guides(color=guide_legend(ncol=2)) 
# ```
# 
# ```{r}
# library(gridGraphics)
# library(grid)
# 
# # ## grab the scene as a grid object &amp; save it to P1
# # grid.echo()
# # chap4_1f &lt;- grid.grab()
# png(
#      &quot;/Users/wenjunliu/PhD_thesis/draft_figure/chapter_04/loading.png&quot;,
#     width = 150, height =130, units=&#39;mm&#39;, res = 300
# )
# plotLoadings(splsda.final, contrib = &#39;max&#39;, method = &#39;median&#39;, comp = 1,
#              title = &quot;&quot; ,size.name = 1,
#              # legend.title = &quot;Subtype&quot;, size.legend = 2, 
#              legend.color = diag_cols[c(&quot;IDC&quot;, &quot;ILC&quot;)], 
#              legend = FALSE
#              )
# dev.off()
# library(magick)
# chap4_1f  &lt;- image_read(path = &quot;/Users/wenjunliu/PhD_thesis/draft_figure/chapter_04/loading.png&quot;) %&gt;%
#     image_ggplot()
# ```
# 
# ```{r}
# library(patchwork)
# chap4_fig1 &lt;- plot_grid(
#     plot_grid(
#         chap4_1a, chap4_1b, 
#         rel_widths = c(0.6, 0.4), 
#         labels = c(&quot;A&quot;, &quot;B&quot;)
#     ), 
#     plot_grid(
#         chap4_1c &amp;
#             theme(
#                 text = element_text(size = 14)
#             ), chap4_1d &amp;
#             theme(
#                 strip.text  = element_text(size = 12), 
#                 legend.text = element_text(size = 11), 
#                 legend.title = element_text(size = 12), 
#                 plot.margin = unit(c(0, 0, 0.5, 0.5), &quot;cm&quot;)
#             ), 
#         rel_widths = c(0.4, 0.7), 
#         labels = c(&quot;C&quot;, &quot;D&quot;)
#     ), 
#     plot_grid(
#         chap4_1e &amp;
#             theme(
#                 text = element_text(size = 14)
#             ), chap4_1f &amp;
#             theme(
#                 plot.margin = unit(c(-0.6, 0, 0.5, -0.3), &quot;cm&quot;)
#             ), 
#         labels = c(&quot;E&quot;, &quot;F&quot;), 
#         scale = c(1, 1.1), 
#         axis = &quot;l&quot;
#     ), 
#     nrow = 3, 
#     rel_heights = c(0.4, 0.25, 0.3) 
#     )
# png(
#     &quot;/Users/wenjunliu/PhD_thesis/Images/chapter_04/intro_diff.png&quot;,
#     width = 260, height =260, units=&#39;mm&#39;, res = 300
# )
# chap4_fig1
# dev.off()
# ```
# 
# 
# ## Figure 4.3
# 
# ```{r}
# gsea &lt;- gsea_res_sig %&gt;%
#     mutate(
#         hjust = ifelse(NES &gt; 0, 1.05, -0.05)
#     ) %&gt;%
#     ggplot(
#         aes(pathway, 
#             NES, 
#             fill = NES, 
#         )) +
#     geom_bar(stat=&quot;identity&quot;, position=&quot;stack&quot;) +
#     scale_fill_distiller(palette = &quot;RdBu&quot;, 
#                           limit = c(-2.2, 2.2),
#                           name = &quot;NES&quot;) +
#     xlab(&quot;&quot;) +
#     ylab(&quot;&quot;) +
#     geom_text(aes(label = pathway, y  = 0, hjust = hjust) , 
#                   size = 2.5) + 
#     coord_flip()  +
#     scale_x_discrete(
#                      breaks = NULL) +
#     guides(size = FALSE) +
#     theme( 
#         panel.grid = element_blank(), 
#         text = element_text(size = 11),  
#         legend.text = element_text(size = 6), 
#         legend.title = element_text(size = 7),
#         panel.border = element_blank()
#     )
# # png(
# #     &quot;/Users/wenjunliu/PhD_thesis/Images/chapter_04/naive_gsea.png&quot;,
# #     width = 130, height =110, units=&#39;mm&#39;, res = 300
# # )
# # gsea
# # dev.off()
# ```
# 
# ## Supp Fig ROC curve
# ```{r}
# # png(&quot;~/PhD_thesis/Images/chapter_04/ROC.png&quot;,
# #     width = 120, height =80, units=&#39;mm&#39;, res = 300)
# #  auroc(splsda.final, newdata = combined_test, outcome.test = test_type, roc.comp = 2, print = FALSE,
# #     legend.title = &quot;Subtype&quot;)
# # dev.off()
# ```
# 
# ## Supp 4.2
# 
# ```{r}
# sup1a &lt;- splsda.pred_CellLine$class$mahalanobis.dist %&gt;%
#     as.data.frame() %&gt;%
#     rownames_to_column(&quot;sample&quot;) %&gt;%
#     left_join(cell_meta_his) %&gt;%
#     dplyr::select(cell_line, 
#                   `sPLS-DA` = &quot;comp2&quot;, 
#                   `Cell-line` = Histology) %&gt;%
#     dplyr::filter(str_detect(`Cell-line`, &quot;like&quot;, TRUE)) %&gt;%
#     unique() %&gt;%
#     droplevels() %&gt;%
#     pivot_longer(
#         cols = -&quot;cell_line&quot;, 
#         names_to = &quot;Type&quot;, 
#         values_to = &quot;Subtype&quot;
#     ) %&gt;%
#     ggplot(
#         aes(factor(cell_line, levels = c(&quot;MCF7&quot;, &quot;T47D&quot;, 
#                                          &quot;MDAMB134VI&quot;, &quot;SUM44&quot;)), 
#             Type,
#             # str_wrap(Type, 10),  
#             color = Subtype)
#     ) +
#     geom_point(
#         size = 7, 
#         # aes(shape = Type)
#     ) +
#     scale_color_manual(
#         values = c(
#             diag_cols), 
#         name = &quot;Subtype&quot;, 
#         drop = FALSE
#     ) +
#     coord_fixed() +
#     labs( x = &quot;&quot;, y = &quot;&quot;) +
#     # guides(color=guide_legend(ncol=2)) +
#     theme(
#         panel.grid = element_blank()
#     )
# ```
# 
# ```{r}
# pred_result_ellis &lt;- readRDS(here::here(&quot;figure/ellis_pred.rds&quot;))
# ellis_order &lt;- pred_result_ellis %&gt;%
#   dplyr::filter(Type == &quot;Pathologist Label&quot;) %&gt;%
#   mutate(
#     group = paste(Subtype, align,sep = &quot;_&quot;), 
#     group = factor(group, levels = c(&quot;ILC_TRUE&quot;, &quot;IDC_TRUE&quot;, &quot;IDC_FALSE&quot;))) %&gt;%
#   .[order(.$group),] 
# sup1b &lt;- pred_result_ellis %&gt;%
#     mutate(Type = ifelse(
#         str_detect(Type, &quot;prediction&quot;), &quot;sPLS-DA&quot;, &quot;Pathologist&quot;
#     )) %&gt;%
#     ggplot(
#         aes(
#             factor(subject_id, levels = ellis_order$subject_id), 
#             str_wrap(Type,10), 
#             #Type,
#             color = str_wrap(Subtype, 15))
#     ) +
#     geom_point(
#         size = 7, 
#         # aes(shape = Type)
#     ) +
#     scale_color_manual(
#         values = diag_cols %&gt;%
#             set_names(str_wrap(names(diag_cols), 15)), 
#         name = &quot;Subtype&quot;
#     ) +
#     coord_fixed(1.5) +
#     labs( x = &quot;&quot;, y = &quot;&quot;)+
#     theme(
#         legend.box.background = element_rect(colour = &quot;black&quot;)
#     )
# ```
# 
# ```{r}
# sup1c &lt;- ILC_score_cell %&gt;%
#     rownames_to_column(&quot;sample&quot;) %&gt;%
#     left_join(cell_meta_his) %&gt;%
#     left_join(pred_result_cell) %&gt;%
#     dplyr::filter(
#         Type == &quot;Cell-line type&quot;, 
#         str_detect(Subtype, &quot;like&quot;, TRUE)) %&gt;%
#     ggplot(
#         aes(Subtype, TotalScore, color = Subtype )
#     ) +
#     geom_boxplot() +
#     scale_color_manual(values = c(diag_cols, &quot;ILC-like&quot; = &quot;darkgreen&quot;), 
#                        name = &quot;Subtype&quot;) +
# 
#     labs(
#         x = &quot;Cell-line type&quot;, 
#         y = &quot;ILC scores&quot;) +
#     theme(legend.position = &quot;none&quot;)
# ```
# 
# ```{r}
# ILC_score_ellis &lt;- readRDS(here::here(&quot;figure/ellis_ILC_score.rds&quot;))
# sup1d &lt;- ILC_score_ellis %&gt;%
#     dplyr::select(Type, Subtype, TotalScore, subject_id) %&gt;%
#     unique() %&gt;%
#     mutate(Type = ifelse(
#         str_detect(Type, &quot;prediction&quot;), &quot;sPLS-DA&quot;, &quot;Pathologist&quot;
#     )) %&gt;%
#     pivot_wider(
#         names_from = &quot;Type&quot;, 
#         values_from = &quot;Subtype&quot;
#     ) %&gt;%
#     mutate(
#         Type = ifelse(
#             `sPLS-DA` ==  Pathologist, Pathologist, &quot;Discordant&quot;
#         ), 
#         Type = factor(Type, levels = c(&quot;IDC&quot;, &quot;ILC&quot;, &quot;Discordant&quot;))
#     ) %&gt;%
#     ggplot(
#         aes(Type, TotalScore, color =Type )
#     ) +
#     geom_boxplot() +
#     scale_color_manual(values = diag_cols, 
#                        name = &quot;Subtype&quot;) +
#     theme(
#         # axis.text.x = element_blank(), 
#         axis.ticks.x = element_blank(), 
#         legend.position = &quot;none&quot;
#     ) +
#     labs(
#         x = &quot;&quot;, 
#         y = &quot;ILC scores&quot;) +
#     geom_point(aes(color = Type)) 
# ```
# 
# ```{r}
# pred_result_enobosarm &lt;- readRDS(here::here(&quot;figure/enobosarm_pred.rds&quot;))
# enobosarm_order &lt;- pred_result_enobosarm %&gt;%
#   dplyr::filter(Type == &quot;Pathologist Label&quot;) %&gt;%
#   .[order(.$align),] 
# sup1e &lt;- pred_result_enobosarm %&gt;%
#     mutate(Type = ifelse(
#         str_detect(Type, &quot;prediction&quot;), &quot;sPLS-DA&quot;, &quot;Pathologist&quot;
#     )) %&gt;%
#     ggplot(
#         aes(
#             factor(patient, levels = enobosarm_order$patient), 
#             str_wrap(Type, 10),  color = Subtype)
#     ) +
#     geom_point(
#         size = 7, 
#         # aes(shape = Type)
#     ) +
#     scale_color_manual(
#         values = c(diag_cols, &quot;Unknow&quot; = &quot;grey&quot;), 
#         name = &quot;Subtype&quot;
#     ) +
#     coord_fixed() +
#     labs( x = &quot;&quot;, y = &quot;&quot;) +
#     theme(
#         legend.box.background = element_rect(colour = &quot;black&quot;))
# ```
# 
# ```{r}
# ILC_score_enobosarm &lt;- readRDS(here::here(&quot;figure/enobosarm_ILC_score.rds&quot;))
# sup1f &lt;- ILC_score_enobosarm %&gt;%
#     drop_na() %&gt;%
#     dplyr::select(Type, Subtype, TotalScore, patient) %&gt;%
#     unique() %&gt;%
#     mutate(Type = ifelse(
#         str_detect(Type, &quot;prediction&quot;), &quot;sPLS-DA&quot;, &quot;Pathologist&quot;
#     )) %&gt;%
#     pivot_wider(
#         names_from = &quot;Type&quot;, 
#         values_from = &quot;Subtype&quot;
#     ) %&gt;%
#     mutate(
#         Type = ifelse(
#             `sPLS-DA` ==  Pathologist, Pathologist, &quot;Discordant&quot;
#         ), 
#         Type = factor(Type, levels = c(&quot;IDC&quot;, &quot;ILC&quot;, &quot;Discordant&quot;))
#     ) %&gt;%
#     ggplot(
#         aes(Type, TotalScore, color = Type)
#     ) +
#     geom_boxplot() +
#     scale_color_manual(values = diag_cols, 
#                        name = &quot;subtype&quot;) +
#     theme(
#     # axis.text.x = element_blank(), 
#     axis.ticks.x = element_blank(), 
#     legend.position = &quot;none&quot;
#     )+
#     labs(
#         x = &quot;&quot;, 
#         y = &quot;ILC scores&quot;) +
#     geom_point(aes(color = Type))
# ```
# 
# ```{r}
# sup1g &lt;- ILC_score_veh %&gt;%
#     rownames_to_column(&quot;name&quot;) %&gt;%
#     dplyr::select(-contains(&quot;Dispersion&quot;)) %&gt;%
#     pivot_longer(
#         cols = -&quot;name&quot;, 
#         names_to = &quot;Score_type&quot;, 
#         values_to = &quot;score&quot;
#     ) %&gt;%
#     left_join(tumor_sample %&gt;%
#                   dplyr::select(name, patient, Diagnosis)) %&gt;%
#     ggplot(
#         aes(Diagnosis, score, color = Diagnosis)
#     ) +
#     geom_boxplot() +
#     scale_color_manual(values = new_diag_col, 
#                        name = &quot;Predicted Subtype&quot;) +
#     labs(
#         x = &quot;Predicted Subtype&quot;, 
#         y = &quot;ILC scores&quot;) +
#     geom_point(aes(color = Diagnosis)) +
#     facet_wrap(~Score_type
#                # , scale = &quot;free_y&quot;
#                )  +
#     theme(legend.position = &quot;none&quot;)
# ```
# 
# ```{r, fig.height=8, fig.width=10}
# sup1 &lt;- ((((sup1a | sup1b) &amp; theme(axis.text.x = element_text(angle = 45, hjust = 1))) +
#     plot_layout(widths = c(0.3, 0.7))) /
#     (((sup1c + theme(axis.title.y = element_text(vjust = -12))) | plot_spacer() | (sup1d + theme(strip.text = element_text(size = 27))) ) +
#     plot_layout(widths = c(0.3, 0.1, 0.7))) /
#     (((sup1e + theme(axis.text.x = element_text(angle = 45, hjust = 1))) | (sup1f + theme(axis.title.x = element_text( vjust = 7)))) +
#     plot_layout(widths = c(0.6, 0.4))) /
#     (((sup1g + theme(axis.title.y = element_text(vjust = -12), strip.text = element_text(size = 27)))| plot_spacer()) +
#     plot_layout(widths = c(0.9, 0.1)) )) +
#     plot_annotation(
#         tag_levels = &quot;A&quot;
#     ) +
#     plot_layout(
#         guides = &quot;collect&quot;, 
#         heights = c(0.2, 0.3, 0.3, 0.3)
#     ) &amp;
#     theme(
#         plot.tag = element_text(size = 26, face = &quot;bold&quot; ), 
#         text = element_text(size = 24), 
#         legend.key.size = unit(2, &quot;cm&quot;), 
#         legend.box.background = element_rect(colour = &quot;black&quot;, size = 2), 
#         legend.box.spacing = unit(10, &quot;mm&quot;)
#     )
# # png(
# #     &quot;/Users/wenjunliu/PhD_thesis/Images/chapter_04/classifier.png&quot;,
# #     width = 450, height =500, units=&#39;mm&#39;, res = 300
# # )
# # sup1
# # dev.off()</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21)
Platform: x86_64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.3.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Australia/Adelaide
tzcode source: internal

attached base packages:
 [1] grid      parallel  splines   stats4    stats     graphics  grDevices
 [8] utils     datasets  methods   base     

other attached packages:
 [1] fgsea_1.26.0                recount3_1.10.2            
 [3] SummarizedExperiment_1.30.2 MatrixGenerics_1.12.2      
 [5] matrixStats_1.0.0           GSE62944_1.28.1            
 [7] GEOquery_2.68.0             goseq_1.52.0               
 [9] geneLenDataBase_1.36.0      BiasedUrn_2.0.10           
[11] singscore_1.20.0            ExperimentHub_2.8.0        
[13] mixOmics_6.24.0             lattice_0.21-8             
[15] ggrepel_0.9.3               WGCNA_1.72-1               
[17] fastcluster_1.2.3           dynamicTreeCut_1.63-1      
[19] concaveman_1.1.0            ggnewscale_0.4.9           
[21] ggforce_0.4.1               plotly_4.10.2              
[23] edgeR_3.42.4                limma_3.56.2               
[25] ggplotify_0.1.1             pheatmap_1.0.12            
[27] glue_1.6.2                  UpSetR_1.4.0               
[29] msigdbr_7.5.1               RColorBrewer_1.1-3         
[31] htmltools_0.5.5             scatterpie_0.2.1           
[33] corrplot_0.92               reactable_0.4.4            
[35] lmerTest_3.1-3              colorspace_2.1-0           
[37] lme4_1.1-34                 Matrix_1.6-0               
[39] org.Hs.eg.db_3.17.0         xml2_1.3.5                 
[41] ggfortify_0.4.16            rvest_1.0.3                
[43] multcomp_1.4-25             TH.data_1.1-2              
[45] MASS_7.3-60                 survival_3.5-5             
[47] mvtnorm_1.2-2               DT_0.28                    
[49] cqn_1.46.0                  quantreg_5.95              
[51] SparseM_1.81                preprocessCore_1.62.1      
[53] nor1mix_1.3-0               mclust_6.0.0               
[55] sSNAPPY_1.5.1               magrittr_2.0.3             
[57] cowplot_1.1.1               ensembldb_2.24.0           
[59] AnnotationFilter_1.24.0     GenomicFeatures_1.52.1     
[61] AnnotationDbi_1.62.2        Biobase_2.60.0             
[63] GenomicRanges_1.52.0        GenomeInfoDb_1.36.1        
[65] IRanges_2.34.1              S4Vectors_0.38.1           
[67] igraph_1.5.0                ggraph_2.1.0               
[69] AnnotationHub_3.8.0         BiocFileCache_2.8.0        
[71] dbplyr_2.3.3                BiocGenerics_0.46.0        
[73] pander_0.6.5                scales_1.2.1               
[75] yaml_2.3.7                  lubridate_1.9.2            
[77] forcats_1.0.0               stringr_1.5.0              
[79] dplyr_1.1.2                 purrr_1.0.1                
[81] readr_2.1.4                 tidyr_1.3.0                
[83] tibble_3.2.1                ggplot2_3.4.2              
[85] tidyverse_2.0.0            

loaded via a namespace (and not attached):
  [1] fs_1.6.2                      ProtGenerics_1.32.0          
  [3] bitops_1.0-7                  httr_1.4.6                   
  [5] doParallel_1.0.17             numDeriv_2016.8-1.1          
  [7] backports_1.4.1               tools_4.3.0                  
  [9] utf8_1.2.3                    R6_2.5.1                     
 [11] mgcv_1.9-0                    lazyeval_0.2.2               
 [13] withr_2.5.0                   graphite_1.46.0              
 [15] prettyunits_1.1.1             gridExtra_2.3                
 [17] cli_3.6.1                     sandwich_3.0-2               
 [19] labeling_0.4.2                sass_0.4.6                   
 [21] Rsamtools_2.16.0              yulab.utils_0.0.6            
 [23] foreign_0.8-84                R.utils_2.12.2               
 [25] sessioninfo_1.2.2             impute_1.74.1                
 [27] rstudioapi_0.15.0             RSQLite_2.3.1                
 [29] generics_0.1.3                gridGraphics_0.5-1           
 [31] BiocIO_1.10.0                 crosstalk_1.2.0              
 [33] vroom_1.6.3                   gtools_3.9.4                 
 [35] GO.db_3.17.0                  fansi_1.0.4                  
 [37] R.methodsS3_1.8.2             lifecycle_1.0.3              
 [39] whisker_0.4.1                 blob_1.2.4                   
 [41] promises_1.2.0.1              crayon_1.5.2                 
 [43] annotate_1.78.0               KEGGREST_1.40.0              
 [45] pillar_1.9.0                  knitr_1.43                   
 [47] rjson_0.2.21                  boot_1.3-28.1                
 [49] corpcor_1.6.10                admisc_0.33                  
 [51] codetools_0.2-19              fastmatch_1.1-3              
 [53] ggfun_0.1.1                   data.table_1.14.8            
 [55] vctrs_0.6.3                   png_0.1-8                    
 [57] gtable_0.3.3                  cachem_1.0.8                 
 [59] xfun_0.39                     S4Arrays_1.0.4               
 [61] mime_0.12                     tidygraph_1.2.3              
 [63] iterators_1.0.14              statmod_1.5.0                
 [65] interactiveDisplayBase_1.38.0 ellipsis_0.3.2               
 [67] nlme_3.1-162                  bit64_4.0.5                  
 [69] progress_1.2.2                filelock_1.0.2               
 [71] rprojroot_2.0.3               bslib_0.5.0                  
 [73] rpart_4.1.19                  DBI_1.1.3                    
 [75] Hmisc_5.1-0                   nnet_7.3-19                  
 [77] tidyselect_1.2.0              bit_4.0.5                    
 [79] compiler_4.3.0                curl_5.0.1                   
 [81] git2r_0.32.0                  graph_1.78.0                 
 [83] htmlTable_2.4.1               DelayedArray_0.26.6          
 [85] rtracklayer_1.60.0            checkmate_2.2.0              
 [87] rappdirs_0.3.3                digest_0.6.33                
 [89] minqa_1.2.5                   rmarkdown_2.23               
 [91] XVector_0.40.0                base64enc_0.1-3              
 [93] pkgconfig_2.0.3               highr_0.10                   
 [95] fastmap_1.1.1                 rlang_1.1.1                  
 [97] htmlwidgets_1.6.2             shiny_1.7.4.1                
 [99] farver_2.1.1                  jquerylib_0.1.4              
[101] zoo_1.8-12                    jsonlite_1.8.7               
[103] BiocParallel_1.34.2           R.oo_1.25.0                  
[105] RCurl_1.98-1.12               Formula_1.2-5                
[107] GenomeInfoDbData_1.2.10       munsell_0.5.0                
[109] Rcpp_1.0.11                   babelgene_22.9               
[111] viridis_0.6.4                 stringi_1.7.12               
[113] zlibbioc_1.46.0               plyr_1.8.8                   
[115] Biostrings_2.68.1             graphlayouts_1.0.0           
[117] hms_1.1.3                     locfit_1.5-9.8               
[119] reshape2_1.4.4                biomaRt_2.56.1               
[121] BiocVersion_3.17.1            XML_3.99-0.14                
[123] evaluate_0.21                 BiocManager_1.30.21          
[125] nloptr_2.0.3                  tzdb_0.4.0                   
[127] foreach_1.5.2                 tweenr_2.0.2                 
[129] httpuv_1.6.11                 MatrixModels_0.5-2           
[131] polyclip_1.10-4               xtable_1.8-4                 
[133] restfulr_0.0.15               RSpectra_0.16-1              
[135] later_1.3.1                   viridisLite_0.4.2            
[137] rARPACK_0.11-0                ellipse_0.4.5                
[139] memoise_2.0.1                 GenomicAlignments_1.36.0     
[141] cluster_2.1.4                 workflowr_1.7.0              
[143] timechange_0.2.0              here_1.0.1                   
[145] GSEABase_1.62.0              </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
